---
title: "DESeq2_final02"
author: "Gwonjin Lee"
date: "24 Januar 2019"
output: html_document
editor_options: 
  chunk_output_type: console
---

# This RMD file is of population-specific normalization for Cd response in three populations and
normalization across all samples for comparison between population pairs.
Normalized counts are divided by gene length for visualization of gene expression.

## Install DEseq2 & Dependencies


```{r,warning=FALSE, echo=FALSE}

#R.version
#source("http://www.bioconductor.org/BiocManager::install.R")
#?BiocManager::install
#BiocManager::install()
#BiocManager::install("BiocUpgrade")
##??biocUpgrade
#update.packages("BiocManager::install")
#BiocManager::install("limma")
#BiocManager::install("statmod")
#BiocManager::install("DelayedArray")
#BiocManager::install("data.table")
#install.packages("stringi",type="win.binary")
#BiocManager::install("stringi")
#BiocManager::install("DESeq2")
#BiocManager::install("S4Vectors")
#BiocManager::install("apeglm")
#BiocManager::install("genefilter")
#install.packages("gplots")
#install.packages("checkmate")
#install.packages("rlang")
#BiocManager::install("tibble")
#BiocManager::install("colorspace")
#install.packages("purrr")
#BiocManager::install("lazyeval")
#BiocManager::install("tidyr")
#BiocManager::install("httpuv")
#BiocManager::install("haven")
#BiocManager::install("readxl")


if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")


library(data.table)
library(backports)
library(readxl)
library(httpuv)
library(haven)
library(tidyr)
library(Rcpp)
library(purrr)
library(colorspace)
library(checkmate)
library(tibble)
library(S4Vectors)
library(DelayedArray)
library(GenomicRanges)
library(IRanges)
library(limma)
library(statmod)
library(stringi)

library(DESeq2)

library(RColorBrewer)
library(gplots)
library(affy)
library(ggplot2)
library(gtools)
library(ggpubr)
library(genefilter)
library(ggrepel)
library(EnhancedVolcano)
library(magrittr)
library(gridExtra)
library(grid)
library(cowplot)
library(gtable)
library(reshape2)
library(plyr)
library(dplyr)
library(Rmisc)
library(mime)
library(pheatmap)
library(viridis)
library(tidyverse)
library(later)
library(spdep)

library(zip)
library(car)
library(FSA)
require(scales)
require(stats)
library(PMCMR)
library(stats)

#library(agricolae)


```

## Load data

```{r,warning=FALSE, echo=FALSE}

setwd("D:/PhD/Project_3_RNAseq/DESeq2/Counts")

# Load and merge multiple countdata 

filelist = list.files(pattern = "*.txt") # Make a file list from all count text files

filelist <- mixedsort(filelist) # Sort the numbers: 1 to 72

datafr = do.call(cbind,lapply(filelist,function(fn)read.table(fn,header=FALSE, sep="\t")[,2])) #merge all count files
genes <- read.delim("P1_count.txt", head = FALSE)[,1] # for the first column
genes2 <- as.character(genes)

count0 = cbind(genes2, datafr) #merge with gene name
countdata <- count0[,-1]
rownames(countdata) <- count0[,1] 
colnames(countdata) <- c(1:72) # add column names
colnames(countdata)[1:72] <- paste("P", colnames(countdata[,c(1:72)]), sep = "") # add a letter P to the column names just to avoid confusion
mode(countdata)
mode(countdata) = "numeric" #convert the class of a matrix from character to numeric
class(countdata)

#statsPerSample <- data.frame(t(apply(countdata, 2, summary)))
#head(statsPerSample)

# Define a function to draw a scatter plot for a pair of variables (samples) with density colors
#plotFun <- function(x,y){ 
#  dns <- densCols(x,y); 
#  points(x,y, col=dns, pch=".", panel.first=grid());  
#  abline(a=0, b=1, col="brown")
#  }


# Plot the scatter plot for a few pairs of variables selected at random
#set.seed(123) # forces the random number generator to produce fixed results
#pairs(log2(countdata[,sample(ncol(countdata),5)] + 1), 
#      panel=plotFun, lower.panel = NULL)

# Load the pre-made coldata to represent experimental designs
coldata <- read.delim("D:/PhD/Project_3_RNAseq/DESeq2/Counts/coldata/All_coldata2.txt", header = TRUE)
rownames(coldata) <- coldata$ID
colnames(datafr) <- coldata$ID


```

## Calculation of FPKM
```{r}
# 1) Load and Merge

#gff <- read.table('C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/Mapping/Genome/Aha_kent_genome(Gemmifera)/annotation.gff ', header = FALSE, sep = '\t')

gtf <- read.table('C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Aha_kent_annotation.gtf', header = FALSE, sep = '\t') # Load a annotation (gtf) file
colnames(gtf) <- c("scaffold", "source", "feature", "start", "end", "score", "strand","frame" ,"ID")
gtf$ID <- gsub("gene_id ", "", gtf$ID) 
head(gtf)               
# Then merge with the Counts dataframe using the column "ID" 
count_fpkm <- data.frame(count0)
colnames(count_fpkm)<- c("ID",1:72) # add column names
head(count_fpkm)
colnames(count_fpkm)[2:73] <- paste("P", colnames(count_fpkm[,c(2:73)]), sep = "") 

FPKM_df = merge(gtf, count_fpkm, by = "ID")
FPKM_df$length <- with(FPKM_df, end - start)
head(FPKM_df)

write.table(as.data.frame(FPKM_df), "C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/FPKM_df.txt", sep = "\t", row.names = FALSE)

# 2) Calculate
  # FPKM=(Counts/(gene size/1000))/(sum mapped fragments/1 million)

#Load the FPKM data calculated for each sample and each gene in excel
FPKM_df_calculated <- read.delim("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/FPKM_df.txt")

# 3) Plot FPKM (vs counts)
hist(as.matrix(log2(FPKM_df_calculated[,83:154] + 1)), col="grey15", border="white",
     main="Log2-transformed FPKM per gene", xlab="log2(FPKM+1)", ylab="Number of genes", 
     las=1, cex.axis=0.7, xlim = c(0,10))

plot(log2(FPKM_df_calculated$P1+1), log2(FPKM_df_calculated$FPKM_P1+1), xlab = "log2(FPKM+1) ", ylab = "log2(count+1)", cex = .7, main = "P1")
plot(log2(FPKM_df_calculated$P38+1), log2(FPKM_df_calculated$FPKM_P38+1), xlab = "log2(FPKM+1) ", ylab = "log2(count+1)", cex = .7, main = "P38")

# 4) Explore
# Create a histogram of expression (FPKM) using a log-transformation.
D_P1 = density((log10(FPKM_df_calculated$FPKM_P1+1)))
plot(D_P1, main = "FPKM_P1", xlab = "log2(FPKM + 1)")

D_P38 = density((log10(FPKM_df_calculated$FPKM_P38+1)))
plot(D_P38, main = "FPKM_P38", xlab = "log2(FPKM + 1)")

hist(log2(FPKM_df_calculated$FPKM_P1+1), main = "FPKM_P1", xlab = "log2(FPKM + 1)")

```

##Make data frames with grouping and run the DESeq2
```{r}
#Make a data frame form a matrix
dds <- DESeqDataSetFromMatrix(
countData = countdata,
colData = coldata,
design = ~ Type + Condition)

#Grouping and run the DESeq2 for populaitons
dds$group <- factor(paste0(dds$Type, dds$Condition))
design(dds) <- ~ group
dds <- DESeq(dds)
resultsNames(dds)

plotDispEsts(dds, main= "All samples") #Dispersion plot

ddsPra <- DESeq(dds ,fitType = "parametric") #Test another option for fitType as parametric
plotDispEsts(ddsPra, main= "All samples_parametric") #Dispersion plot for parametric

ddsmean <- DESeq(dds ,fitType = "mean") #Test another option for fitType as mean
plotDispEsts(ddsmean, main= "All samples_mean") #Dispersion plot for mean


dds.SF <- estimateSizeFactors(dds) #Check the size factors calculated by geometric mean
print(sizeFactors(dds.SF))

plot(sizeFactors(dds.SF), colSums(counts(dds.SF)),
     xlab = "Size factors",
     ylab = "Library sizes") #Correlation between size factor and library size
abline(lm(colSums(counts(dds.SF)) ~ sizeFactors(dds.SF) + 0))


#Correlation between size factor and library size with tissue info
df.SF <- as.data.frame(sizeFactors(dds.SF))
df.colsum <- as.data.frame(colSums(counts(dds.SF)))
dev.off()
plot(df.SF[1:36,], df.colsum[1:36,],
     xlab = "Size factors",  xlim = c(0.68,1.7), cex.lab=1.2,
     ylab = "Library sizes", ylim=c(8900000,24000000),col="blue") 
par(new=TRUE)
plot(df.SF[37:72,], df.colsum[37:72,],
     xlab = "Size factors", xlim = c(0.68,1.7),cex.lab=1.2,
     ylab = "Library sizes", ylim=c(8900000,24000000), col="red")
abline(lm(colSums(counts(dds.SF)) ~ sizeFactors(dds.SF) + 0))
legend(x=1.5, y=14000000, legend=c('Shoot', 'Root'), col=c('blue', 'red'), pch = c(1,1), box.lty=0, cex= 1.2)

cor.test(sizeFactors(dds.SF), colSums(counts(dds.SF)))
cor.test(df.SF[1:36,], df.colsum[1:36,])
cor.test(df.SF[37:72,], df.colsum[37:72,])

#Make plots for size factor vs library
df.SF.LS <- cbind(coldata,df.SF,df.colsum)
model.y <- lm(colSums(counts(dds.SF)) ~ sizeFactors(dds.SF), df.SF.LS)
coef.y <- coef(model.y)
ggplot(df.SF.LS, 
       aes(x=sizeFactors(dds.SF), y=colSums(counts(dds.SF)), color=Tissue, shape=Population)) +
      geom_point(size=3) + 
      xlab("Size factor (median of ratios)") +
      ylab("Library size (total counts)") +
      geom_abline(intercept=coef.y[1], slope=coef.y[2], size=1.1, color= "grey50")

model.X <- lm(sizeFactors(dds.SF) ~ colSums(counts(dds.SF)), df.SF.LS)
coef.X <- coef(model.X)
ggplot(df.SF.LS, 
       aes(x=colSums(counts(dds.SF)), y=sizeFactors(dds.SF), color=Tissue, shape=Population)) +
      geom_point(size=3) + 
      xlab("Library size (total counts)") +
      ylab("Size factor") +
      geom_abline(intercept=coef.X[1], slope=coef.X[2], size=1.1, color= "grey50")



#Size factor in different Groups (Tissue or population)
dev.off()
boxplot(df.SF[1:36,], df.SF[37:72,], names = c("Shoot", "Root"), ylab = "Size factor", ylim = c(0.5,1.7))
boxplot(df.SF[c(1,2,7,8,13,14,19,20,25,26,31,32,37,38,43,44,49,50,55,56,61,62,67,68),], 
        df.SF[c(3,4,9,10,15,16,21,22,27,28,33,34,39,40,45,46,51,52,57,58,63,64,69,70),], 
        df.SF[c(5,6,11,12,17,18,23,24,29,30,35,36,41,42,47,48,53,54,59,60,65,66,71,72),], 
        names = c("Noss", "Pais", "Wall"), ylab = "Size factor", ylim = c(0.5,1.7))

#Library size in different Groups (Tissue or population)
boxplot(df.colsum[1:36,], df.colsum[37:72,], names = c("Shoot", "Root"), ylab = "Library size")
boxplot(df.colsum[c(1,2,7,8,13,14,19,20,25,26,31,32,37,38,43,44,49,50,55,56,61,62,67,68),], 
        df.colsum[c(3,4,9,10,15,16,21,22,27,28,33,34,39,40,45,46,51,52,57,58,63,64,69,70),], 
        df.colsum[c(5,6,11,12,17,18,23,24,29,30,35,36,41,42,47,48,53,54,59,60,65,66,71,72),], 
        names = c("Noss", "Pais", "Wall"), ylab = "Library size")


NormalizedCount_all <- counts(dds.SF, normalized = T) #Extract the normalized counts
head(NormalizedCount_all)
#write.table(NormalizedCount_all, file="Normalized_counts_woGTF.txt", sep="\t", quote=F, col.names=NA)

boxplot(log10(assays(dds)[["cooks"]]), range=0, las=2, main="Cook's distances") # Check any outliers by Cook's D 

#Grouping and run the DESeq2 for individuals
dds_Ind <- dds
dds_Ind$group <- factor(paste0(dds_Ind$Type2, dds_Ind$Condition))
design(dds_Ind) <- ~ group
dds_Ind <- DESeq(dds_Ind)
resultsNames(dds_Ind)

#Data frames for root and shoot, respectively
ddsShoot <- DESeqDataSetFromMatrix(
countData = countdata[,1:36],
colData = coldata[1:36,],
design = ~ Population + Condition)
ddsShoot


plotDispEsts(ddsShoot, main= "Shoot") #Dispersion plot

ddsShoot.SF <- estimateSizeFactors(ddsShoot) #Check the size factors calculated by geometric mean
print(sizeFactors(ddsShoot.SF))
NormalizedCount_Shoot <- counts(ddsShoot.SF, normalized = T) #Extract the normalized counts
head(NormalizedCount_Shoot)
#write.table(NormalizedCount_Shoot, file="Normalized_counts_woGTF_Shoot.txt", sep="\t", quote=F, col.names=NA)


ddsRoot <- DESeqDataSetFromMatrix(
countData = countdata[,37:72],
colData = coldata[37:72,],
design = ~ Population + Condition)
ddsRoot

ddsRoot$group <- factor(paste0(ddsRoot$Population, ddsRoot$Condition))
design(ddsRoot) <- ~ group
ddsRoot <- DESeq(ddsRoot)
resultsNames(ddsRoot)

plotDispEsts(ddsRoot, main= "Root") #Dispersion plot

ddsRoot.SF <- estimateSizeFactors(ddsRoot) #Check the size factors calculated by geometric mean
print(sizeFactors(ddsRoot.SF))
NormalizedCount_Root <- counts(ddsRoot.SF, normalized = T) #Extract the normalized counts
head(NormalizedCount_Root)
#write.table(NormalizedCount_Root, file="Normalized_counts_woGTF_Root.txt", sep="\t", quote=F, col.names=NA)

#Correlation between size factor and library size with tissue info

dfRoot.SF <- as.data.frame(sizeFactors(ddsRoot.SF))
dfShoot.SF <- as.data.frame(sizeFactors(ddsShoot.SF))

dfRoot.colsum <- as.data.frame(colSums(counts(ddsRoot.SF)))
dfShoot.colsum <- as.data.frame(colSums(counts(ddsShoot.SF)))

dfShoot <- data.frame(dfShoot.SF, dfShoot.colsum)
dfRoot <- data.frame(dfRoot.SF, dfRoot.colsum)


dev.off()
plot(dfShoot,
     xlab = "Size factors",  xlim = c(0.68,1.7), cex.lab=1.2,
     ylab = "Library sizes", ylim=c(8900000,24000000),col="blue") 
par(new=TRUE)
plot(dfRoot,
     xlab = "Size factors", xlim = c(0.68,1.7),cex.lab=1.2,
     ylab = "Library sizes", ylim=c(8900000,24000000), col="red")
abline(lm(colSums(counts(dds.SF)) ~ sizeFactors(dds.SF) + 0))
legend(x=1.5, y=14000000, legend=c('Shoot', 'Root'), col=c('blue', 'red'), pch = c(1,1), box.lty=0, cex= 1.2)




#Make a data frame form matrices of each population 

# 1) For each Population between Cd vs Ctr

NossS <-DESeqDataSetFromMatrix(
countData = countdata[,c(1,2,7,8,13,14,19,20,25,26,31,32)],
colData = coldata[c(1,2,7,8,13,14,19,20,25,26,31,32),],
design = ~  Condition)
NossS
ddsNossS <- DESeq(NossS)

PaisS <-DESeqDataSetFromMatrix(
countData = countdata[,c(3,4,9,10,15,16,21,22,27,28,33,34)],
colData = coldata[c(3,4,9,10,15,16,21,22,27,28,33,34),],
design = ~ Condition)
PaisS
ddsPaisS <- DESeq(PaisS)

WallS <-DESeqDataSetFromMatrix(
countData = countdata[,c(5,6,11,12,17,18,23,24,29,30,35,36)],
colData = coldata[c(5,6,11,12,17,18,23,24,29,30,35,36),],
design = ~ Condition)
WallS
ddsWallS <- DESeq(WallS)

NossR <-DESeqDataSetFromMatrix(
countData = countdata[,c(37,38,43,44,49,50,55,56,61,62,67,68)],
colData = coldata[c(37,38,43,44,49,50,55,56,61,62,67,68),],
design = ~ Condition)
NossR
ddsNossR <- DESeq(NossR)

PaisR <-DESeqDataSetFromMatrix(
countData = countdata[,c(39,40,45,46,51,52,57,58,63,64,69,70)],
colData = coldata[c(39,40,45,46,51,52,57,58,63,64,69,70),],
design = ~ Condition)
PaisR
ddsPaisR <- DESeq(PaisR)

WallR <-DESeqDataSetFromMatrix(
countData = countdata[,c(41,42,47,48,53,54,59,60,65,66,71,72)],
colData = coldata[c(41,42,47,48,53,54,59,60,65,66,71,72),],
design = ~ Condition)
WallR
ddsWallR <- DESeq(WallR)


# 2) For each Individual between Cd vs Ctr
Noss5S <-DESeqDataSetFromMatrix(
countData = countdata[,c(1,7,13,19,25,31)],
colData = coldata[c(1,7,13,19,25,31),],
design = ~ Condition)
Noss5S

Noss10S <-DESeqDataSetFromMatrix(
countData = countdata[,c(2,8,14,20,26,32)],
colData = coldata[c(2,8,14,20,26,32),],
design = ~ Condition)
Noss10S

Pais5S <-DESeqDataSetFromMatrix(
countData = countdata[,c(3,9,15,21,27,33)],
colData = coldata[c(3,9,15,21,27,33),],
design = ~ Condition)
Pais5S

Pais9S <-DESeqDataSetFromMatrix(
countData = countdata[,c(4,10,16,22,28,34)],
colData = coldata[c(4,10,16,22,28,34),],
design = ~ Condition)
Pais9S

Wall7S <-DESeqDataSetFromMatrix(
countData = countdata[,c(5,11,17,23,29,35)],
colData = coldata[c(5,11,17,23,29,35),],
design = ~ Condition)
Wall7S

Wall10S <-DESeqDataSetFromMatrix(
countData = countdata[,c(6,12,18,24,30,36)],
colData = coldata[c(6,12,18,24,30,36),],
design = ~ Condition)
Wall10S

Noss5R <-DESeqDataSetFromMatrix(
countData = countdata[,c(37,43,49,55,61,67)],
colData = coldata[c(37,43,49,55,61,67),],
design = ~ Condition)
Noss5R

Noss10R <-DESeqDataSetFromMatrix(
countData = countdata[,c(38,44,50,56,62,68)],
colData = coldata[c(38,44,50,56,62,68),],
design = ~ Condition)
Noss10R

Pais5R <-DESeqDataSetFromMatrix(
countData = countdata[,c(39,45,51,57,63,69)],
colData = coldata[c(39,45,51,57,63,69),],
design = ~ Condition)
Pais5R

Pais9R <-DESeqDataSetFromMatrix(
countData = countdata[,c(40,46,52,58,64,70)],
colData = coldata[c(40,46,52,58,64,70),],
design = ~ Condition)
Pais9R

Wall7R <-DESeqDataSetFromMatrix(
countData = countdata[,c(41,47,53,59,65,71)],
colData = coldata[c(41,47,53,59,65,71),],
design = ~ Condition)
Wall7R

Wall10R <-DESeqDataSetFromMatrix(
countData = countdata[,c(42,48,54,60,66,72)],
colData = coldata[c(42,48,54,60,66,72),],
design = ~ Condition)
Wall10R


```

## Check for distribution of counts before and after normalization in different ways
```{r}
# Histogram

hist(as.matrix(countdata), col="grey15", border="white",
     breaks=2000, xlim=c(0,2000), main="Counts per gene",
     xlab="Counts (truncated axis)", ylab="Number of genes", 
     las=1, cex.axis=0.7) # before log-transformation
hist(as.matrix(log2(counts(dds, normalized=FALSE)+1)), col="grey15", border="white",
     main="Log2-transformed counts per gene", xlab="log2(counts+1)", ylab="Number of genes", 
     las=1, cex.axis=0.7, ylim=c(0,700000)) # before normalization
hist(as.matrix(log2(counts(dds, normalized=TRUE)+1)), col="grey15", border="white",
     main="Log2-transformed normalized counts per gene", xlab="log2(normalized counts+1)", ylab="Number of genes", 
     las=1, cex.axis=0.7, ylim=c(0,700000)) # after normalization


#Plot the log2-transformed counts before and after normalization
group.col <- c("lightblue","orange")[coldata$Tissue]

dev.off()
par(mfrow=c(1,2))
boxplot(log2(counts(dds.SF, normalized=FALSE)+1),xlab="log2(counts+1)", main="Raw counts" , las = 1, horizontal = TRUE, pch = ",", col = group.col)
boxplot(log2(counts(dds.SF, normalized=TRUE)+1), xlab="log2(normalized counts)", main="Normalized counts", las = 1, horizontal = TRUE, pch = ",", col = group.col)


dev.off()
par(mfrow=c(2,1))
plotDensity(log2(counts(dds.SF)+1), xlab="log2(counts+1)", cex.lab=1, panel.first=grid(), col = group.col, main="Raw counts") 
plotDensity(log2(counts(dds.SF, normalized=TRUE)+1), xlab="log2(normalized counts+1)", cex.lab=1, panel.first=grid(), col = group.col, main="Normalized counts") 



```

## Sample distance by Heatmap and PCA
```{r}
#Dataset for each tissue
ddsShoot_Ind <- DESeqDataSetFromMatrix(
countData = countdata[,1:36],
colData = coldata[1:36,],
design = ~ Individual + Replica + Condition)
ddsShoot

ddsRoot_Ind <- DESeqDataSetFromMatrix(
countData = countdata[,37:72],
colData = coldata[37:72,],
design = ~ Individual + Replica + Condition)
ddsRoot

#DESeq2 in tissue
ddsS_Ind <- DESeq(ddsShoot_Ind)
ddsR_Ind <- DESeq(ddsRoot_Ind)

# Variance stabilizing transformation (can be also done with rlog, but it will take so long time)
vsd <- vst(dds, blind = TRUE)
vsdS <- vst(ddsS_Ind, blind = TRUE)
vsdR <- vst(ddsR_Ind, blind = TRUE)

#Log <- rlog(dds, blind = TRUE)
#LogS <- rlog(ddsS, blind = TRUE)
#LogR <- rlog(ddsR, blind = TRUE)

#Calculate sample distance
sampleDists <- dist( t( assay(vsd) ) )
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste(vsd$Individual, vsd$Tissue, vsd$Condition, sep="-" )
colnames(sampleDistMatrix) <- NULL

colours = colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmap.2( sampleDistMatrix, trace="none", col=colours, margins = c(5,10))


#sampleDistsLog <- dist( t( assay(Log) ) )
#sampleDistMatrixLog <- as.matrix( sampleDistsLog )
#rownames(sampleDistMatrixLog) <- paste(Log$Individual, Log$Tissue, Log$Condition, sep="-" )
#colnames(sampleDistMatrixLog) <- NULL
#colours = colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
#heatmap.2( sampleDistMatrixLog, trace="none", col=colours, margins = c(5,10))

#Test PCA
plotPCA(vsd, intgroup=c("Individual", "Tissue", "Condition"))

#Visualization of PCA

pcaData <- plotPCA(vsd, intgroup=c("Population", "Condition", "Replica", "Individual", "Tissue"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

allPC <- ggplot(pcaData, aes(PC1, PC2, color=Population, shape=Tissue)) +
  geom_point(size=3) +
  xlab(paste0("PC1 (",percentVar[1],"%) ")) +
  ylab(paste0("PC2 (",percentVar[2],"%) ")) + 
  coord_fixed() +
  theme_bw() +
  ggtitle("All") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_color_manual(breaks = pcaData$Population,
  values=c( "#EE3B3B", "#008B00", "#0072B2")) +
  theme(plot.title = element_text(size = 14, face = "bold"),
    legend.title=element_text(size=12), 
    legend.text=element_text(size=12)) +
    theme(panel.grid.major = element_blank()) +
    theme(panel.grid.minor = element_blank())


## For individual tissues

# Shoot
sampleDistsS <- dist( t( assay(vsdS) ) )
sampleDistMatrixS <- as.matrix( sampleDistsS )
rownames(sampleDistMatrixS) <- paste(vsdS$Individual, vsdS$Condition, sep="-" )
colnames(sampleDistMatrixS) <- NULL

colours = colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmap.2( sampleDistMatrixS, trace="none", col=colours, margins = c(5,10))


#plotPCA(vsdS, intgroup=c("Population", "Condition"))

pcaDataS <- plotPCA(vsdS, intgroup=c("Population", "Condition", "Replica", "Individual"), returnData=TRUE)
percentVar <- round(100 * attr(pcaDataS, "percentVar"))

ShootPC <- ggplot(pcaDataS, aes(PC1, PC2, color=Individual, shape=Condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1 (",percentVar[1],"%) ")) +
  ylab(paste0("PC2 (",percentVar[2],"%) ")) + 
  coord_fixed() +
  theme_bw() +
  ggtitle("Shoot") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_color_manual(breaks = pcaDataS$Individual,
  values=c( "#FF4600", "#FF69B4","#008B00", "#32CD32", "#0072B2", "#00008B")) +
  geom_point(aes(shape=Condition)) +
  theme(plot.title = element_text(size = 14, face = "bold"),
    legend.title=element_text(size=12), 
    legend.text=element_text(size=12)) +
    theme(panel.grid.major = element_blank()) +
    theme(panel.grid.minor = element_blank())


pcaDataSLogS <- plotPCA(LogS, intgroup=c("Population", "Condition", "Replica", "Individual"), returnData=TRUE)
percentVarLogS <- round(100 * attr(pcaDataSLogS, "percentVar"))

ShootPCLogS <- ggplot(pcaDataSLogS, aes(PC1, PC2, color=Individual, shape=Condition)) +
  geom_point(size=2.5) +
  xlab(paste0("PC1 (",percentVarLogS[1],"%) ")) +
  ylab(paste0("PC2 (",percentVarLogS[2],"%) ")) + 
  coord_fixed() +
  theme_bw() +
  ggtitle("Shoot samples") + 
  scale_color_manual(breaks = pcaDataSLogS$Individual,
  values=c( "#FF4600", "#FF69B4","#008B00", "#32CD32", "#0072B2", "#00008B")) +
  geom_point(aes(shape=Condition)) +
  scale_shape_manual(values=c(19,15))

# Root
sampleDistsR <- dist( t( assay(vsdR) ) )
sampleDistMatrixR <- as.matrix( sampleDistsR )
rownames(sampleDistMatrixR) <- paste(vsdR$Individual, vsdR$Condition, sep="-" )
colnames(sampleDistMatrixR) <- NULL

colours = colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmap.2( sampleDistMatrixR, trace="none", col=colours, margins = c(5,10))


#plotPCA(vsdR, intgroup=c("Population", "Condition"))

pcaDataR <- plotPCA(vsdR, intgroup=c("Population", "Condition", "Replica", "Individual"), returnData=TRUE)
percentVar <- round(100 * attr(pcaDataR, "percentVar"))

RootPC <- ggplot(pcaDataR, aes(PC1, PC2, color=Individual, shape=Condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1 (",percentVar[1],"%) ")) +
  ylab(paste0("PC2 (",percentVar[2],"%) ")) + 
  coord_fixed() +
  theme_bw() +
  ggtitle("Root") + 
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_color_manual(breaks = pcaDataR$Individual,
  values=c( "#FF4600", "#FF69B4","#008B00", "#32CD32", "#0072B2", "#00008B")) +
  geom_point(aes(shape=Condition))  +
  theme(plot.title = element_text(size = 14, face = "bold"),
    legend.title=element_text(size=12), 
    legend.text=element_text(size=12)) +
    theme(panel.grid.major = element_blank()) +
    theme(panel.grid.minor = element_blank())


pcaDataSLogR <- plotPCA(LogR, intgroup=c("Population", "Condition", "Replica", "Individual"), returnData=TRUE)
percentVarLogR <- round(100 * attr(pcaDataSLogR, "percentVar"))

RootPCLogR <- ggplot(pcaDataSLogR, aes(PC1, PC2, color=Individual, shape=Condition)) +
  geom_point(size=2.5) +
  xlab(paste0("PC1 (",percentVarLogR[1],"%) ")) +
  ylab(paste0("PC2 (",percentVarLogR[2],"%) ")) + 
  coord_fixed() +
  theme_bw() +
  ggtitle("Shoot samples") + 
  scale_color_manual(breaks = pcaDataSLogR$Individual,
  values=c( "#FF4600", "#FF69B4","#008B00", "#32CD32", "#0072B2", "#00008B")) +
  geom_point(aes(shape=Condition)) +
  scale_shape_manual(values=c(19,15))


ggarrange(allPC,labels = "A",
          ggarrange(ShootPC, RootPC, ncol=2, common.legend = TRUE, legend = "right",  labels = c('B', 'C'), align = "h"),
          nrow=2)


#For PC3, PC4...

#biocLite("genefilter")
#biocLite("ggrepel")

plotPCA.san <- function (vsd, intgroup=c("Population", "Condition", "Replica", "Individual"), ntop = 500, returnData = FALSE) 
{
  rv <- rowVars(assay(vsd))
  select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
                                                     length(rv)))]
  pca <- prcomp(t(assay(vsd)[select, ]))
  percentVar <- pca$sdev^2/sum(pca$sdev^2)
  if (!all(intgroup %in% names(colData(vsd)))) {
    stop("the argument 'intgroup' should specify columns of colData(dds)")
  }
  intgroup.df <- as.data.frame(colData(vsd)[, intgroup, drop = FALSE])
  group <- if (length(intgroup) > 1) {
    factor(apply(intgroup.df, 1, paste, collapse = " : "))
  }
  else {
    colData(vsd)[[intgroup]]
  }
  d <- data.frame(PC3 = pca$x[, 3], PC4 = pca$x[, 4], group = group, 
                  intgroup.df, name = colData(vsd)[,1])
  if (returnData) {
    attr(d, "percentVar") <- percentVar[1:2]
    return(d)
  }
    ggplot(data = d, aes_string(x = "PC3", y = "PC4", color = Population, shape=Condition, label = ID)) + geom_point(size = 3) + xlab(paste0("PC3: ", round(percentVar[1] * 100), "% variance")) + ylab(paste0("PC4: ", round(percentVar[2] * 100), "% variance")) + coord_fixed() + geom_text_repel(size=3) 

}

#print(plotPCA(vsd, intgroup=c("Condition", "Population")))

#pcaData2 <- plotPCA.san(vsd, intgroup=c("Population", "Condition", "Individual"), returnData=TRUE)
#percentVar <- round(100 * attr(pcaData2, "percentVar"))
#ggplot(pcaData2, aes(PC3, PC4, color=Individual, shape=Condition)) +
#  geom_point(size=3) +
#  xlab(paste0("PC3: ",percentVar[3],"% variance")) +
#  ylab(paste0("PC4: ",percentVar[4],"% variance")) + 
#  coord_fixed()


#################################################################################
#PCA for populations

#Shoot
vsdNossS <- vst (ddsNossS, blind = FALSE)
vsdPaisS <- vst (ddsPaisS, blind = FALSE)
vsdWallS <- vst (ddsWallS, blind = FALSE)

## For only Noss Shoot

sampleDists <- dist( t( assay(vsdNossS) ) )
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste(vsdNossS$Individual, vsdNossS$Condition, sep="-" )
colnames(sampleDistMatrix) <- NULL

colours = colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmap.2( sampleDistMatrix, trace="none", col=colours, margins = c(5,10))

#plotPCA(vsdNossS, intgroup=c("Replica", "Condition", "Individual"))

pcaDataNS <- plotPCA(vsdNossS, intgroup=c("Condition", "Individual"), returnData=TRUE)
percentVar <- round(100 * attr(pcaDataNS, "percentVar"))

NS <- ggplot(pcaDataNS, aes(PC1, PC2, color=Individual, shape=Condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed() +
  theme_bw() +
  ggtitle("Noss-Shoot")



## For only Pais Shoot

sampleDists <- dist( t( assay(vsdPaisS) ) )
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste(vsdPaisS$Individual, vsdPaisS$Condition, sep="-" )
colnames(sampleDistMatrix) <- NULL

colours = colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmap.2( sampleDistMatrix, trace="none", col=colours, margins = c(5,10))


#plotPCA(vsdPaisS, intgroup=c("Replica", "Condition", "Individual"))


pcaDataPS <- plotPCA(vsdPaisS, intgroup=c("Condition", "Individual"), returnData=TRUE)
percentVar <- round(100 * attr(pcaDataPS, "percentVar"))

PS <- ggplot(pcaDataPS, aes(PC1, PC2, color=Individual, shape=Condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed() +
  theme_bw() +
  ggtitle("Pais-Shoot")



## For only Wall Shoot

sampleDists <- dist( t( assay(vsdWallS) ) )
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste(vsdWallS$Individual, vsdWallS$Condition, sep="-" )
colnames(sampleDistMatrix) <- NULL

colours = colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmap.2( sampleDistMatrix, trace="none", col=colours, margins = c(5,10))


#plotPCA(vsdWallS, intgroup=c("Replica", "Condition", "Individual"))


pcaDataWS <- plotPCA(vsdWallS, intgroup=c("Condition", "Individual", "Replica"), returnData=TRUE)
percentVar <- round(100 * attr(pcaDataWS, "percentVar"))

WS <- ggplot(pcaDataWS, aes(PC1, PC2, color=Individual, shape=Condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed() +
  theme_bw() +
  ggtitle("Wall-Shoot")

#############################################################################
#Root

vsdNossR <- vst (ddsNossR, blind = FALSE)
vsdPaisR <- vst (ddsPaisR, blind = FALSE)
vsdWallR <- vst (ddsWallR, blind = FALSE)
## For only Noss Root

sampleDists <- dist( t( assay(vsdNossR) ) )
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste(vsdNossR$Individual, vsdNossR$Condition, sep="-" )
colnames(sampleDistMatrix) <- NULL

colours = colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmap.2( sampleDistMatrix, trace="none", col=colours, margins = c(5,10))

#plotPCA(vsdNossR, intgroup=c("Replica", "Condition", "Individual"))


pcaDataNR <- plotPCA(vsdNossR, intgroup=c("Condition", "Individual"), returnData=TRUE)
percentVar <- round(100 * attr(pcaDataNR, "percentVar"))

NR <- ggplot(pcaDataNR, aes(PC1, PC2, color=Individual, shape=Condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed() +
  theme_bw() +
  ggtitle("Noss-Root")


## For only Pais Root

sampleDists <- dist( t( assay(vsdPaisR) ) )
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste(vsdPaisR$Individual, vsdPaisR$Condition, sep="-" )
colnames(sampleDistMatrix) <- NULL

colours = colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmap.2( sampleDistMatrix, trace="none", col=colours, margins = c(5,10))

#plotPCA(vsdPaisR, intgroup=c("Replica", "Condition", "Individual"))


pcaDataPR <- plotPCA(vsdPaisR, intgroup=c("Condition", "Individual"), returnData=TRUE)
percentVar <- round(100 * attr(pcaDataPR, "percentVar"))

PR <- ggplot(pcaDataPR, aes(PC1, PC2, color=Individual, shape=Condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed() +
  theme_bw() +
  ggtitle("Pais-Root")


## For only Wall Root

sampleDists <- dist( t( assay(vsdWallR) ) )
sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste(vsdWallR$Individual, vsdWallR$Condition, sep="-" )
colnames(sampleDistMatrix) <- NULL

colours = colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmap.2( sampleDistMatrix, trace="none", col=colours, margins = c(5,10))

#plotPCA(vsdWallR, intgroup=c("Replica", "Condition", "Individual"))


pcaDataWR <- plotPCA(vsdWallR, intgroup=c("Condition", "Individual", "Replica"), returnData=TRUE)
percentVar <- round(100 * attr(pcaDataWR, "percentVar"))

WR <- ggplot(pcaDataWR, aes(PC1, PC2, color=Individual, shape=Condition)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed() +
  theme_bw() +
  ggtitle("Wall-Root")

ggarrange(NS,NR, PS, PR, WS, WR, ncol=2, nrow=3)

```


#Run DESeq2
```{r}
# 1) For each Population (between Cd vs Ctr)
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/CdvsCtr")

ddsNossS <- DESeq(NossS)
plotDispEsts(ddsNossS, main = "Noss_Shoot")
resNossS <- results(ddsNossS, contrast = c("Condition","Cd","Ctr") )
resNossS
plotCounts(ddsNossS, gene=which.min(resNossS$padj), intgroup="Condition", xlab = "Cd-Condition", main = "Noss_Shoot")
plotMA(resNossS, ylim=c(-2,2), main ="Noss-Shoot (Cd vs Ctr)")
hist( resNossS$pvalue, breaks=20, col="grey", main = "Noss-Shoot (Cd vs Ctr)", xlab = "P-value" )
plot(x=resNossS$log2FoldChange, y=-log10(resNossS$padj), xlab="log2(Fold-Change)", ylab="-log10(adjusted P-value)",
     col=ifelse(resNossS$padj<=0.05, "red", "black"), main="Noss_Shoot (Cd vs Ctr)")

ddsPaisS <- DESeq(PaisS)
plotDispEsts(ddsPaisS, main = "Pais_Shoot")
resPaisS <- results(ddsPaisS, contrast = c("Condition","Cd","Ctr"))
resPaisS
plotCounts(ddsPaisS, gene=which.min(resPaisS$padj), intgroup="Condition", xlab = "Cd-Condition", main = "Pais_Shoot")
plotMA(resPaisS, ylim=c(-2,2), main ="Pais-Shoot (Cd vs Ctr)")
hist( resPaisS$pvalue, breaks=20, col="grey", main = "Pais-Shoot (Cd vs Ctr)", xlab = "P-value" )
plot(x=resPaisS$log2FoldChange, y=-log10(resPaisS$padj), xlab="log2(Fold-Change)", ylab="-log10(adjusted P-value)",
     col=ifelse(resPaisS$padj<=0.05, "red", "black"), main="Pais_Shoot (Cd vs Ctr)")


ddsWallS <- DESeq(WallS)
plotDispEsts(ddsWallS, main = "Wall_Shoot")
resWallS <- results(ddsWallS, contrast = c("Condition","Cd","Ctr"))
plotCounts(ddsWallS, gene=which.min(resWallS$padj), intgroup="Condition", xlab = "Cd-Condition", main = "Wall_Shoot")
plotMA(resWallS, ylim=c(-2,2), main ="Wall-Shoot (Cd vs Ctr)")
hist( resWallS$pvalue, breaks=20, col="grey", main = "Wall-Shoot (Cd vs Ctr)", xlab = "P-value" )
plot(x=resWallS$log2FoldChange, y=-log10(resWallS$padj), xlab="log2(Fold-Change)", ylab="-log10(adjusted P-value)",
     col=ifelse(resWallS$padj<=0.05, "red", "black"), main="Wall_Shoot (Cd vs Ctr)")


ddsNossR <- DESeq(NossR)
plotDispEsts(ddsNossR, main = "Noss_Root")
resNossR <- results(ddsNossR, contrast = c("Condition","Cd","Ctr"))
plotCounts(ddsNossR, gene=which.min(resNossR$padj), intgroup="Condition", xlab = "Cd-Condition", main = "Noss_Root")
plotMA(resNossR, ylim=c(-2,2), main ="Noss_Root (Cd vs Ctr)")
hist( resNossR$pvalue, breaks=20, col="grey", main = "Noss_Root (Cd vs Ctr)", xlab = "P-value" )
plot(x=resNossR$log2FoldChange, y=-log10(resNossR$padj), xlab="log2(Fold-Change)", ylab="-log10(adjusted P-value)",
     col=ifelse(resNossR$padj<=0.05, "red", "black"), main="Noss_Root (Cd vs Ctr)", ylim = c(0,2))


ddsPaisR <- DESeq(PaisR)
plotDispEsts(ddsPaisR, main = "Pais_Root")
resPaisR <- results(ddsPaisR, contrast = c("Condition","Cd","Ctr"))
plotCounts(ddsPaisR, gene=which.min(resPaisR$padj), intgroup="Condition", xlab = "Cd-Condition", main = "Pais_Root")
plotMA(resPaisR, ylim=c(-2,2), main ="Pais_Root (Cd vs Ctr)")
hist( resPaisR$pvalue, breaks=20, col="grey", main = "Pais_Root (Cd vs Ctr)", xlab = "P-value" )
plot(x=resPaisR$log2FoldChange, y=-log10(resPaisR$padj), xlab="log2(Fold-Change)", ylab="-log10(adjusted P-value)",
     col=ifelse(resPaisR$padj<=0.05, "red", "black"), main="Pais_Root (Cd vs Ctr)", xlim= c(-5, 5))


ddsWallR <- DESeq(WallR)
plotDispEsts(ddsWallR, main = "Wall_Root")
resWallR <- results(ddsWallR, contrast = c("Condition","Cd","Ctr"))
plotCounts(ddsWallR, gene=which.min(resWallR$padj), intgroup="Condition", xlab = "Cd-Condition", main = "Wall_Root")
plotMA(resWallR, ylim=c(-2,2), main ="Wall_Root (Cd vs Ctr)")
hist( resWallR$pvalue, breaks=20, col="grey", main = "Wall_Root (Cd vs Ctr)", xlab = "P-value" )
plot(x=resWallR$log2FoldChange, y=-log10(resWallR$padj), xlab="log2(Fold-Change)", ylab="-log10(adjusted P-value)",
     col=ifelse(resWallR$padj<=0.05, "red", "black"), main="Wall_Root (Cd vs Ctr)")

#1-2) Sort out based on significance
resNossS0 = cbind(genes, resNossS)
resPaisS0 = cbind(genes, resPaisS)
resWallS0 = cbind(genes, resWallS)
resNossR0 = cbind(genes, resNossR)
resPaisR0 = cbind(genes, resPaisR)
resWallR0 = cbind(genes, resWallR)

resNossS_Sig  = subset(resNossS0, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1)
resPaisS_Sig  = subset(resPaisS0, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1)
resWallS_Sig  = subset(resWallS0, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1)
resNossR_Sig  = subset(resNossR0, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1)
resPaisR_Sig  = subset(resPaisR0, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1)
resWallR_Sig  = subset(resWallR0, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1)

write.table( as.data.frame(resNossS_Sig), "Significant_genes/Noss_Shoot_CdvsCtr_Signifi_Results.txt", sep = "\t", row.names = FALSE)

write.table( as.data.frame(resPaisS_Sig), "Significant_genes/Pais_Shoot_CdvsCtr_Signifi_Results.txt", sep = "\t", row.names = FALSE)

write.table( as.data.frame(resWallS_Sig), "Significant_genes/Wall_Shoot_CdvsCtr_Signifi_Results.txt", sep = "\t", row.names = FALSE)

write.table( as.data.frame(resNossR_Sig), "Significant_genes/Noss_Root_CdvsCtr_Signifi_Results.txt", sep = "\t", row.names = FALSE)

write.table( as.data.frame(resPaisR_Sig), "Significant_genes/Pais_Root_CdvsCtr_Signifi_Results.txt", sep = "\t", row.names = FALSE)

write.table( as.data.frame(resWallR_Sig), "Significant_genes/Wall_Root_CdvsCtr_Signifi_Results.txt", sep = "\t", row.names = FALSE)

resNossS_Sig0.5  = subset(resNossS0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resPaisS_Sig0.5  = subset(resPaisS0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resWallS_Sig0.5  = subset(resWallS0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resNossR_Sig0.5  = subset(resNossR0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resPaisR_Sig0.5  = subset(resPaisR0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resWallR_Sig0.5  = subset(resWallR0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)

write.table( as.data.frame(resNossS_Sig0.5), "Significant_genes/Noss_Shoot_CdvsCtr_Sig0.5_Results.txt", sep = "\t", row.names = FALSE)

write.table( as.data.frame(resPaisS_Sig0.5), "Significant_genes/Pais_Shoot_CdvsCtr_Sig0.5_Results.txt", sep = "\t", row.names = FALSE)

write.table( as.data.frame(resWallS_Sig0.5), "Significant_genes/Wall_Shoot_CdvsCtr_Sig0.5_Results.txt", sep = "\t", row.names = FALSE)

write.table( as.data.frame(resNossR_Sig0.5), "Significant_genes/Noss_Root_CdvsCtr_Sig0.5_Results.txt", sep = "\t", row.names = FALSE)

write.table( as.data.frame(resPaisR_Sig0.5), "Significant_genes/Pais_Root_CdvsCtr_Sig0.5_Results.txt", sep = "\t", row.names = FALSE)

write.table( as.data.frame(resWallR_Sig0.5), "Significant_genes/Wall_Root_CdvsCtr_Sig0.5_Results.txt", sep = "\t", row.names = FALSE)

#Test
resPaisS2 <- results(ddsPaisS)
resPaisS2
resPaisS20 = cbind(genes, resPaisS2)
resPaisS_Sig0.5_2  = subset(resPaisS20, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
dim(resPaisS_Sig0.5_2)

#2) For comparison between population pairs
#Population vs Population (condition-merged) for consitutively differnt gene expression in populations
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/PopvsPop") 
dds2 <- DESeqDataSetFromMatrix(
countData = countdata,
colData = coldata,
design = ~ Type)

dds2 <- DESeq(dds2)
resultsNames(dds2)

resNvsP_Shoot <- results(dds2, contrast=c("Type", "N_S", "P_S"))
resNvsP_Root <- results(dds2, contrast=c("Type", "N_R", "P_R"))

resPvsW_Shoot <- results(dds2, contrast=c("Type", "P_S", "W_S"))
resPvsW_Root <- results(dds2, contrast=c("Type", "P_R", "W_R"))

resNvsW_Shoot <- results(dds2, contrast=c("Type", "N_S", "W_S"))
resNvsW_Root <- results(dds2, contrast=c("Type", "N_R", "W_R"))

resNvsP_Shoot_0 = cbind(genes, resNvsP_Shoot)
resNvsP_Shoot_Sig  = subset(resNvsP_Shoot_0, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1) 
resNvsP_Root_0 = cbind(genes, resNvsP_Root)
resNvsP_Root_Sig  = subset(resNvsP_Root_0, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1) 

resPvsW_Shoot_0 = cbind(genes, resPvsW_Shoot)
resPvsW_Shoot_Sig  = subset(resPvsW_Shoot_0, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1) 
resPvsW_Root_0 = cbind(genes, resPvsW_Root)
resPvsW_Root_Sig  = subset(resPvsW_Root_0, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1) 

resNvsW_Shoot_0 = cbind(genes, resNvsW_Shoot)
resNvsW_Shoot_Sig  = subset(resNvsW_Shoot_0, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1) 
resNvsW_Root_0 = cbind(genes, resNvsW_Root)
resNvsW_Root_Sig  = subset(resNvsW_Root_0, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1) 

write.table(as.data.frame(resNvsP_Shoot_Sig), file = "Significant_genes/resNvsP_Shoot_Sig_Cond_merged.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(resNvsP_Root_Sig), file = "Significant_genes/resNvsP_Root_Sig_Cond_merged.txt", sep = "\t", row.names = FALSE)

write.table(as.data.frame(resPvsW_Shoot_Sig), file = "Significant_genes/resPvsW_Shoot_Sig_Cond_merged.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(resPvsW_Root_Sig), file = "Significant_genes/resPvsW_Root_Sig_Cond_merged.txt", sep = "\t", row.names = FALSE)

write.table(as.data.frame(resNvsW_Shoot_Sig), file = "Significant_genes/resNvsW_Shoot_Sig_Cond_merged.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(resNvsW_Root_Sig), file = "Significant_genes/resNvsW_Root_Sig_Cond_merged.txt", sep = "\t", row.names = FALSE)

#Population vs Population per condition
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/PopvsPop/Significant_genes/Condition_separated")
resNvsP_Ctr_Shoot <- results(dds, contrast=c("group", "N_SCtr", "P_SCtr"))
resNvsP_Cd_Shoot <- results(dds, contrast=c("group", "N_SCd", "P_SCd"))
resNvsP_Ctr_Root <- results(dds, contrast=c("group", "N_RCtr", "P_RCtr"))
resNvsP_Cd_Root <- results(dds, contrast=c("group", "N_RCd", "P_RCd"))

resPvsW_Ctr_Shoot <- results(dds, contrast=c("group", "P_SCtr", "W_SCtr"))
resPvsW_Cd_Shoot <- results(dds, contrast=c("group", "P_SCd", "W_SCd"))
resPvsW_Ctr_Root <- results(dds, contrast=c("group", "P_RCtr", "W_RCtr"))
resPvsW_Cd_Root <- results(dds, contrast=c("group", "P_RCd", "W_RCd"))

resNvsW_Ctr_Shoot <- results(dds, contrast=c("group", "N_SCtr", "W_SCtr"))
resNvsW_Cd_Shoot <- results(dds, contrast=c("group", "N_SCd", "W_SCd"))
resNvsW_Ctr_Root <- results(dds, contrast=c("group", "N_RCtr", "W_RCtr"))
resNvsW_Cd_Root <- results(dds, contrast=c("group", "N_RCd", "W_RCd"))

resNvsP_Ctr_Shoot_0 = cbind(genes, resNvsP_Ctr_Shoot)
resNvsP_Ctr_Shoot_Sig  = subset(resNvsP_Ctr_Shoot_0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5) 
resNvsP_Cd_Shoot_0 = cbind(genes, resNvsP_Cd_Shoot)
resNvsP_Cd_Shoot_Sig  = subset(resNvsP_Cd_Shoot_0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5) 
resNvsP_Ctr_Root_0 = cbind(genes, resNvsP_Ctr_Root)
resNvsP_Ctr_Root_Sig  = subset(resNvsP_Ctr_Root_0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5) 
resNvsP_Cd_Root_0 = cbind(genes, resNvsP_Cd_Root)
resNvsP_Cd_Root_Sig  = subset(resNvsP_Cd_Root_0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5) 

resPvsW_Ctr_Shoot_0 = cbind(genes, resPvsW_Ctr_Shoot)
resPvsW_Ctr_Shoot_Sig  = subset(resPvsW_Ctr_Shoot_0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5) 
resPvsW_Cd_Shoot_0 = cbind(genes, resPvsW_Cd_Shoot)
resPvsW_Cd_Shoot_Sig  = subset(resPvsW_Cd_Shoot_0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5) 
resPvsW_Ctr_Root_0 = cbind(genes, resPvsW_Ctr_Root)
resPvsW_Ctr_Root_Sig  = subset(resPvsW_Ctr_Root_0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5) 
resPvsW_Cd_Root_0 = cbind(genes, resPvsW_Cd_Root)
resPvsW_Cd_Root_Sig  = subset(resPvsW_Cd_Root_0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5) 

resNvsW_Ctr_Shoot_0 = cbind(genes, resNvsW_Ctr_Shoot)
resNvsW_Ctr_Shoot_Sig  = subset(resNvsW_Ctr_Shoot_0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5) 
resNvsW_Cd_Shoot_0 = cbind(genes, resNvsW_Cd_Shoot)
resNvsW_Cd_Shoot_Sig  = subset(resNvsW_Cd_Shoot_0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5) 
resNvsW_Ctr_Root_0 = cbind(genes, resNvsW_Ctr_Root)
resNvsW_Ctr_Root_Sig  = subset(resNvsW_Ctr_Root_0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5) 
resNvsW_Cd_Root_0 = cbind(genes, resNvsW_Cd_Root)
resNvsW_Cd_Root_Sig  = subset(resNvsW_Cd_Root_0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5) 

write.table(as.data.frame(resNvsP_Ctr_Shoot_Sig), file = "resNvsP_Ctr_Shoot_Sig.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(resNvsP_Cd_Shoot_Sig), file = "resNvsP_Cd_Shoot_Sig.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(resNvsP_Ctr_Root_Sig), file = "resNvsP_Ctr_Root_Sig.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(resNvsP_Cd_Root_Sig), file = "resNvsP_Cd_Root_Sig.txt", sep = "\t", row.names = FALSE)

write.table(as.data.frame(resPvsW_Ctr_Shoot_Sig), file = "resPvsW_Ctr_Shoot_Sig.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(resPvsW_Cd_Shoot_Sig), file = "resPvsW_Cd_Shoot_Sig.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(resPvsW_Ctr_Root_Sig), file = "resPvsW_Ctr_Root_Sig.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(resPvsW_Cd_Root_Sig), file = "resPvsW_Cd_Root_Sig.txt", sep = "\t", row.names = FALSE)

write.table(as.data.frame(resNvsW_Ctr_Shoot_Sig), file = "resNvsW_Ctr_Shoot_Sig.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(resNvsW_Cd_Shoot_Sig), file = "resNvsW_Cd_Shoot_Sig.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(resNvsW_Ctr_Root_Sig), file = "resNvsW_Ctr_Root_Sig.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(resNvsW_Cd_Root_Sig), file = "resNvsW_Cd_Root_Sig.txt", sep = "\t", row.names = FALSE)


```

##Sort out DEGs between Cd and Ctr and convert to AGI for GO 
```{r,warning=FALSE, echo=FALSE}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/CdvsCtr/Significant_genes")

AhaAth <- read.delim("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/Significant_genes/AhaAha_genes.txt", header = TRUE)

colnames(AhaAth)[1] <- "genes"

#Cd and Ctr
#Shoot
AGI_SigGenes_CdvsCtr_Pais_Shoot <- merge(AhaAth, resPaisS_Sig, by = "genes")
AGI_SigGenes_CdvsCtr_Wall_Shoot <- merge(AhaAth, resWallS_Sig, by = "genes")

write.table(as.data.frame(AGI_SigGenes_CdvsCtr_Pais_Shoot), "AGImerged/AGI_CdvsCtr_Pais_Shoot.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(AGI_SigGenes_CdvsCtr_Wall_Shoot), "AGImerged/AGI_CdvsCtr_Wall_Shoot.txt", sep = "\t", row.names = FALSE)

#Root
AGI_SigGenes_CdvsCtr_Pais_Root <- merge(AhaAth, resPaisR_Sig, by = "genes")
AGI_SigGenes_CdvsCtr_Wall_Root <- merge(AhaAth, resWallR_Sig, by = "genes")

write.table(as.data.frame(AGI_SigGenes_CdvsCtr_Pais_Root), "AGImerged/AGI_CdvsCtr_Pais_Root.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(AGI_SigGenes_CdvsCtr_Wall_Root), "AGImerged/AGI_CdvsCtr_Wall_Root.txt", sep = "\t", row.names = FALSE)

```


## Extract DEGs with annotation (Cd vs Ctr per population)
```{r}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/CdvsCtr/Annotated_DEGs")

# Add annotation for Noss
Predata <- read.delim("D:/PhD/Project_3_RNAseq/DESeq2/Result/New result2/Tables/Pre_dataset S1.txt")
Predata2 <- Predata[,1:9]
head(Predata2)


#All Sig DEGs per population
Ano_Pais_Root_Sig <- merge(Predata[,c(1:9,18:19)], resPaisR_Sig, by = "genes")
Ano_Pais_Shoot_Sig <- merge(Predata[,c(1:9,12:13)], resPaisS_Sig, by = "genes")
write.table(as.data.frame(Ano_Pais_Root_Sig), file = "Ano_Pais_Root_Sig.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_Pais_Shoot_Sig), file = "Ano_Pais_Shoot_Sig.txt", sep = "\t", row.names = FALSE)

Ano_Wall_Root_Sig <- merge(Predata[,c(1:9,20:21)], resWallR_Sig, by = "genes")
Ano_Wall_Shoot_Sig <- merge(Predata[,c(1:9,14:15)], resWallS_Sig, by = "genes")
write.table(as.data.frame(Ano_Wall_Root_Sig), file = "Ano_Wall_Root_Sig.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_Wall_Shoot_Sig), file = "Ano_Wall_Shoot_Sig.txt", sep = "\t", row.names = FALSE)

#Shared DEGs in Pais and Wall
#Load shared DEGs
CdvsCtr_PW_Up_Root <- read.delim("SharedDEGs/CdvsCtr_PW_Up_Root.txt")
CdvsCtr_PW_Down_Root <- read.delim("SharedDEGs/CdvsCtr_PW_Down_Root.txt")
CdvsCtr_PW_Up_Shoot <- read.delim("SharedDEGs/CdvsCtr_PW_Up_Shoot.txt")
CdvsCtr_PW_Down_Shoot <- read.delim("SharedDEGs/CdvsCtr_PW_Down_Shoot.txt")

Ano_SharedDEGs_PW_Root_Up <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "genes"),
        list(Predata2,
             CdvsCtr_PW_Up_Root,
             resPaisR_Sig,
             resWallR_Sig
             ))
Ano_SharedDEGs_PW_Root_Down <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "genes"),
        list(Predata2,
             CdvsCtr_PW_Down_Root,
             resPaisR_Sig,
             resWallR_Sig
             ))
Ano_SharedDEGs_PW_Shoot_Up <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "genes"),
        list(Predata2,
             CdvsCtr_PW_Up_Shoot,
             resPaisS_Sig,
             resWallS_Sig
             ))
Ano_SharedDEGs_PW_Shoot_Down <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "genes"),
        list(Predata2,
             CdvsCtr_PW_Down_Shoot,
             resPaisS_Sig,
             resWallS_Sig
             ))


write.table(as.data.frame(Ano_SharedDEGs_PW_Root_Up), file = "Ano_SharedDEGs_PW_Root_Up.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_SharedDEGs_PW_Root_Down), file = "Ano_SharedDEGs_PW_Root_Down.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_SharedDEGs_PW_Shoot_Up), file = "Ano_SharedDEGs_PW_Shoot_Up.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_SharedDEGs_PW_Shoot_Down), file = "Ano_SharedDEGs_PW_Shoot_Down.txt", sep = "\t", row.names = FALSE)

#Maximum mean of normalized count for the Table 1.
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table")


Ano_SharedDEGs_PW_Root_Up <- as.data.frame(Ano_SharedDEGs_PW_Root_Up) %>% arrange(desc(log2FoldChange.x))
Ano_SharedDEGs_PW_Root_Down <- as.data.frame(Ano_SharedDEGs_PW_Root_Down) %>% arrange(log2FoldChange.x)
Ano_SharedDEGs_PW_Shoot_Up <- as.data.frame(Ano_SharedDEGs_PW_Shoot_Up) %>% arrange(desc(log2FoldChange.x))
Ano_SharedDEGs_PW_Shoot_Down <- as.data.frame(Ano_SharedDEGs_PW_Shoot_Down) %>% arrange(log2FoldChange.x)

List_Shared_CdvsCtr <- rbind(Ano_SharedDEGs_PW_Root_Up[,1:2],
                             Ano_SharedDEGs_PW_Shoot_Up[,1:2],
                             Ano_SharedDEGs_PW_Root_Down[,1:2],
                             Ano_SharedDEGs_PW_Shoot_Down[,1:2] )

Predata3<- read.delim("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/Predataset3.txt")
Predata3_Glength <- Predata3[,c(1:3,28:52)]

List_Shared_CdvsCtr_MaxMean <- join(List_Shared_CdvsCtr, Predata3_Glength) 

write.table(as.data.frame(List_Shared_CdvsCtr_MaxMean), file = "List_Shared_CdvsCtr_MaxMean.txt", sep = "\t", row.names = FALSE)

# Extract DEGs of constitutively high expressed genes in Noss
#Root
Ano_Shared_DEGs_MvsNM_Root_Ctr_NT1 <- rbind(Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT, Ano_Shared_DEGs_NPW_Root_Ctr_NT)

Predata3_Glength_Noss <- Predata3[,c(1:3,35:37)]

Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT2 <- merge(Predata3_Glength_Noss,Ano_Shared_DEGs_MvsNM_Root_Ctr_NT1, by = "genes")
Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT2.2 <- subset(Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT2, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1)
Ano_resNvsW_Ctr_Root_NT_Sig2<- subset(Ano_resNvsW_Ctr_Root_NT_Sig, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1)
Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT3 <- merge(Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT2.2,Ano_resNvsW_Ctr_Root_NT_Sig2, by = "genes")

Const_Noss_Ano_diff <- merge(Predata2[,c(1:2, 9)],Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT3, by = "genes" )
write.table(as.data.frame(Const_Noss_Ano_diff), file = "Consititutively_diff_NossforFigS2.txt", sep = "\t", row.names = FALSE)

#Shoot
Ano_Shared_DEGs_MvsNM_Shoot_Ctr_NT1 <- rbind(Ano_Shared_DEGs_NvsP_NvsW_Shoot_Ctr_NT, Ano_Shared_DEGs_NPW_Shoot_Ctr_NT)

Predata3_Glength_Noss <- Predata3[,c(1:3,29:31)]

Ano_Shared_DEGs_NvsP_NvsW_Shoot_Ctr_NT2 <- merge(Predata3_Glength_Noss,Ano_Shared_DEGs_MvsNM_Shoot_Ctr_NT1, by = "genes")
Ano_Shared_DEGs_NvsP_NvsW_Shoot_Ctr_NT2.2 <- subset(Ano_Shared_DEGs_NvsP_NvsW_Shoot_Ctr_NT2, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1)
Ano_resNvsW_Ctr_Shoot_NT_Sig2<- subset(Ano_resNvsW_Ctr_Shoot_NT_Sig, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1)
Ano_Shared_DEGs_NvsP_NvsW_Shoot_Ctr_NT3 <- merge(Ano_Shared_DEGs_NvsP_NvsW_Shoot_Ctr_NT2.2,Ano_resNvsW_Ctr_Shoot_NT_Sig2, by = "genes")

Const_Noss_Ano_diff <- merge(Predata2[,c(1:2, 9)],Ano_Shared_DEGs_NvsP_NvsW_Shoot_Ctr_NT3, by = "genes" )
write.table(as.data.frame(Const_Noss_Ano_diff), file = "Consititutively_diff_NossforFigS2_Shoot.txt", sep = "\t", row.names = FALSE)

## For cut off top genes
#Const_Noss_Ano_high0 <- subset(Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT2, Noss_Length_Ctr_root_mean > 100)
#Const_Noss_Ano_low0 <- subset(Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT2, Pais_Length_Ctr_root_mean > 100)

#attach(as.data.frame(Const_Noss_Ano_high0))
#Const_Noss_Ano_high <- Const_Noss_Ano_high0[order(-log2FoldChange),]
#Const_Noss_Ano_high <- Const_Noss_Ano_high[1:10,]

#attach(as.data.frame(Const_Noss_Ano_low0))
#Const_Noss_Ano_low <- Const_Noss_Ano_low0[order(log2FoldChange),]
#Const_Noss_Ano_low <- Const_Noss_Ano_low[1:10,]


```

## Extract DEGs with annotation (Pop vs Pop per condition)
```{r}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/PopvsPop/Normalized_Tissue")


#All Sig DEGs per population

Ano_resNvsP_Ctr_Shoot_NT <- merge(Predata[,c(1:9,10,12)], resNvsP_Ctr_Shoot_NT_Sig, by = "genes")
Ano_resNvsW_Ctr_Shoot_NT <- merge(Predata[,c(1:9,10,14)], resNvsW_Ctr_Shoot_NT_Sig, by = "genes")
Ano_resPvsW_Ctr_Shoot_NT <- merge(Predata[,c(1:9,12,14)], resPvsW_Ctr_Shoot_NT_Sig, by = "genes")

Ano_resNvsP_Cd_Shoot_NT <- merge(Predata[,c(1:9,11,13)], resNvsP_Cd_Shoot_NT_Sig, by = "genes")
Ano_resNvsW_Cd_Shoot_NT <- merge(Predata[,c(1:9,11,15)], resNvsW_Cd_Shoot_NT_Sig, by = "genes")
Ano_resPvsW_Cd_Shoot_NT <- merge(Predata[,c(1:9,13,15)], resPvsW_Cd_Shoot_NT_Sig, by = "genes")

Ano_resNvsP_Ctr_Root_NT <- merge(Predata[,c(1:9,16,18)], resNvsP_Ctr_Root_NT_Sig, by = "genes")
Ano_resNvsW_Ctr_Root_NT <- merge(Predata[,c(1:9,16,20)], resNvsW_Ctr_Root_NT_Sig, by = "genes")
Ano_resPvsW_Ctr_Root_NT <- merge(Predata[,c(1:9,18,20)], resPvsW_Ctr_Root_NT_Sig, by = "genes")

Ano_resNvsP_Cd_Root_NT <- merge(Predata[,c(1:9,17,19)], resNvsP_Cd_Root_NT_Sig, by = "genes")
Ano_resNvsW_Cd_Root_NT <- merge(Predata[,c(1:9,17,21)], resNvsW_Cd_Root_NT_Sig, by = "genes")
Ano_resPvsW_Cd_Root_NT <- merge(Predata[,c(1:9,19,21)], resPvsW_Cd_Root_NT_Sig, by = "genes")


write.table(as.data.frame(Ano_resNvsP_Ctr_Shoot_NT), file = "Ano_resNvsP_Ctr_Shoot_NT.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_resNvsW_Ctr_Shoot_NT), file = "Ano_resNvsW_Ctr_Shoot_NT.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_resPvsW_Ctr_Shoot_NT), file = "Ano_resPvsW_Ctr_Shoot_NT.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_resNvsP_Cd_Shoot_NT), file = "Ano_resNvsP_Cd_Shoot_NT.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_resNvsW_Cd_Shoot_NT), file = "Ano_resNvsW_Cd_Shoot_NT.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_resPvsW_Cd_Shoot_NT), file = "Ano_resPvsW_Cd_Shoot_NT.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_resNvsP_Ctr_Root_NT), file = "Ano_resNvsP_Ctr_Root_NT.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_resNvsW_Ctr_Root_NT), file = "Ano_resNvsW_Ctr_Root_NT.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_resPvsW_Ctr_Root_NT), file = "Ano_resPvsW_Ctr_Root_NT.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_resNvsP_Cd_Root_NT), file = "Ano_resNvsP_Cd_Root_NT.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_resNvsW_Cd_Root_NT), file = "Ano_resNvsW_Cd_Root_NT.txt", sep = "\t", row.names = FALSE)
write.table(as.data.frame(Ano_resPvsW_Cd_Root_NT), file = "Ano_resPvsW_Cd_Root_NT.txt", sep = "\t", row.names = FALSE)


```



#GO analysis
```{r}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/CdvsCtr/GO")
CdvsCtr_GO_PS0 <- read.delim("GO_CdvsCtr_NPS_allPW.txt", check.names=FALSE)

#CdvsCtr_GO_PS0$logPvalue <- ifelse(CdvsCtr_GO_PS0$Regulation=="Down",CdvsCtr_GO_PS0$logPvalue*-1, CdvsCtr_GO_PS0$logPvalue)

CdvsCtr_GO_PS1 <- split(CdvsCtr_GO_PS0, CdvsCtr_GO_PS0$Tissue)
CdvsCtr_GO_PS_Root <- CdvsCtr_GO_PS1$Root
CdvsCtr_GO_PS_Root$logPvalue <- ifelse(CdvsCtr_GO_PS_Root$Regulation=="Down",CdvsCtr_GO_PS_Root$logPvalue*-1, CdvsCtr_GO_PS_Root$logPvalue)
CdvsCtr_GO_PS_Root2 <- split(CdvsCtr_GO_PS_Root, CdvsCtr_GO_PS_Root$Population)
CdvsCtr_GO_PS_Root_Pais <- CdvsCtr_GO_PS_Root2$`Pais (Root)`
CdvsCtr_GO_PS_Root_Wall <- CdvsCtr_GO_PS_Root2$`Wall (Root)`

CdvsCtr_GO_PS_Shoot <- CdvsCtr_GO_PS1$Shoot
CdvsCtr_GO_PS_Shoot$logPvalue <- ifelse(CdvsCtr_GO_PS_Shoot$Regulation=="Down",CdvsCtr_GO_PS_Shoot$logPvalue*-1, CdvsCtr_GO_PS_Shoot$logPvalue)
CdvsCtr_GO_PS_Shoot2 <- split(CdvsCtr_GO_PS_Shoot, CdvsCtr_GO_PS_Shoot$Population)
CdvsCtr_GO_PS_Shoot_Pais <- CdvsCtr_GO_PS_Shoot2$`Pais (Shoot)`
CdvsCtr_GO_PS_Shoot_Wall <- CdvsCtr_GO_PS_Shoot2$`Wall (Shoot)`


CdvsCtr_GO_PS_Root_Pais$just <- ifelse(CdvsCtr_GO_PS_Root_Pais$Regulation=="Down",0,1)
CdvsCtr_GO_PS_Root_Wall$just <- ifelse(CdvsCtr_GO_PS_Root_Wall$Regulation=="Down",0,1)
CdvsCtr_GO_PS_Shoot_Pais$just <- ifelse(CdvsCtr_GO_PS_Shoot_Pais$Regulation=="Down",0,1)
CdvsCtr_GO_PS_Shoot_Wall$just <- ifelse(CdvsCtr_GO_PS_Shoot_Wall$Regulation=="Down",0,1)


GO_PS_Root_Pais <- ggplot(CdvsCtr_GO_PS_Root_Pais, aes(y=logPvalue, x=reorder(GO_name, logPvalue), fill= Type)) + 
      geom_bar( stat="identity") +  
      facet_grid(Type2 ~  Population  , scales="free_y", space="free", labeller = label_bquote("") ) + 
      scale_y_continuous( limits=c(-15,15), breaks = c(-15,-10,-5,0,5,10,15), labels=c(15,10,5,0,5,10,15)) +
      scale_x_discrete("", labels=NULL) +
      coord_flip()  +
      labs(x= "", fill = "GO category", y=expression(
        paste(bold("Down-regulated"),"            -log(adjusted ", italic("P"), "-value)            ", bold("Up-regulated")))) + 
      theme_minimal() +
      geom_text(aes(x=GO_name, y=0, label=GO_name), hjust=CdvsCtr_GO_PS_Root_Pais$just, size = 6) +
      theme(axis.text.y=element_text(size=15),axis.text.x=element_text(size=10), axis.title=element_text(size=12), strip.text.x =
            element_text(size=20), legend.text = element_text(size=12),legend.title = element_text(size=16)) +
      theme(panel.grid.major = element_blank()) +
      theme(panel.grid.minor = element_blank()) + 
      geom_hline(yintercept = 0, col= "grey40")  + 
      scale_fill_manual(values=c("brown1", "orange1","royalblue4", "maroon3"))
 
   
GO_PS_Root_Pais


GO_PS_Root_Wall <- ggplot(CdvsCtr_GO_PS_Root_Wall, aes(y=logPvalue, x=reorder(GO_name, logPvalue), fill=Type)) + 
      geom_bar( stat="identity") +  
      facet_grid(Type2 ~  Population  , scales="free_y", space="free", labeller = label_bquote("") ) + 
      scale_y_continuous( limits=c(-15,15), breaks = c(-15,-10,-5,0,5,10,15), labels=c(15,10,5,0,5,10,15)) +
      scale_x_discrete("", labels=NULL) +
      coord_flip()  +
      labs(x= "", fill = "GO category" , y=expression(
        paste(bold("Down-regulated"),"            -log(adjusted ", italic("P"), "-value)            ", bold("Up-regulated")))) +
      theme_minimal() +
      geom_text(aes(x=GO_name, y=0, label=GO_name), hjust=CdvsCtr_GO_PS_Root_Wall$just, size = 6) +
      theme(axis.text.y=element_text(size=15),axis.text.x=element_text(size=10), axis.title=element_text(size=12), strip.text.x =
            element_text(size=20), legend.text = element_text(size=12),legend.title = element_text(size=16)) +
      theme(panel.grid.major = element_blank()) +
      theme(panel.grid.minor = element_blank())+ 
      geom_hline(yintercept = 0, col= "grey40") + 
      scale_fill_manual(values=c("brown1", "royalblue4", "maroon3"))

GO_PS_Root_Wall

GO_PS_Shoot_Pais <- ggplot(CdvsCtr_GO_PS_Shoot_Pais, aes(y=logPvalue, x=reorder(GO_name, logPvalue), fill=Type)) + 
      geom_bar( stat="identity") +  
      facet_grid(Type2 ~  Population  , scales="free_y", space="free", labeller = label_bquote("") ) + 
      scale_y_continuous( limits=c(-50,50), breaks = c(-50,-25,0,25,50), labels=c(50,25,0,25,50)) +
      scale_x_discrete("", labels=NULL) +
      coord_flip()  +
      labs(x= "", fill = "GO category" , y=expression(
        paste(bold("Down-regulated"),"            -log(adjusted ", italic("P"), "-value)            ", bold("Up-regulated")))) +
      theme_minimal() +
      geom_text(aes(x=GO_name, y=0, label=GO_name), hjust=CdvsCtr_GO_PS_Shoot_Pais$just, size = 6) +
      theme(axis.text.y=element_text(size=15),axis.text.x=element_text(size=10), axis.title=element_text(size=12), strip.text.x =
            element_text(size=20), legend.text = element_text(size=12),legend.title = element_text(size=16)) +
      theme(panel.grid.major = element_blank()) +
      theme(panel.grid.minor = element_blank()) + 
      geom_hline(yintercept = 0, col= "grey40") + 
      scale_fill_manual(values=c("brown1", "orange1","royalblue4", "maroon3")) 

GO_PS_Shoot_Pais

GO_PS_Shoot_Wall <- ggplot(CdvsCtr_GO_PS_Shoot_Wall, aes(y=logPvalue, x=reorder(GO_name, logPvalue), fill=Type)) + 
      geom_bar( stat="identity") +  
      facet_grid(Type2 ~  Population  , scales="free_y", space="free", labeller = label_bquote("") ) + 
      scale_y_continuous( limits=c(-15,15), breaks = c(-15,-10,-5,0,5,10,15), labels=c(15,10,5,0,5,10,15)) +
      scale_x_discrete("", labels=NULL) +
      coord_flip()  +
      labs(x= "", fill = "GO category" , y=expression(
        paste(bold("Down-regulated"),"            -log(adjusted ", italic("P"), "-value)            ", bold("Up-regulated")))) +
      theme_minimal() +
      geom_text(aes(x=GO_name, y=0, label=GO_name), hjust=CdvsCtr_GO_PS_Shoot_Wall$just, size = 6) +
      theme(axis.text.y=element_text(size=15),axis.text.x=element_text(size=10), axis.title=element_text(size=12), strip.text.x =
            element_text(size=20), legend.text = element_text(size=12),legend.title = element_text(size=16)) +
      theme(panel.grid.major = element_blank()) +
      theme(panel.grid.minor = element_blank()) + 
      geom_hline(yintercept = 0, col= "grey40") + 
      scale_fill_manual(values=c("brown1", "orange1","royalblue4", "maroon3")) 

GO_PS_Shoot_Wall

 
GO_legend <- g_legend(GO_PS_Root_Pais + theme(legend.position='bottom'))

grid.arrange(GO_PS_Root_Pais+theme(legend.position='hidden') + theme(axis.line.x = element_line(color="black", size = 0.5)),
             GO_PS_Shoot_Pais+theme(legend.position='hidden')+ theme(axis.line.x = element_line(color="black", size = 0.5)), 
             GO_PS_Root_Wall+theme(legend.position='hidden')+ theme(axis.line.x = element_line(color="black", size = 0.5)), 
             GO_PS_Shoot_Wall+theme(legend.position='hidden')+ theme(axis.line.x = element_line(color="black", size = 0.5)), 
             GO_legend,
             layout_matrix = cbind(c(1,NA,2,NA,3,NA,4,NA,5)),
             heights = c(1.4,0.1,1.95,0.1, 1.2, 0.1, 1.25, 0.1, 0.2)
             )


###
#GO for overlapping DEGs in Pais and Wall

setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/CdvsCtr/GO")
CdvsCtr_GO_Overlap0 <- read.delim("GO_CdvsCtr_NPS_overlapPW.txt", check.names=FALSE)

CdvsCtr_GO_Overlap1 <- split(CdvsCtr_GO_Overlap0, CdvsCtr_GO_Overlap0$Tissue)
CdvsCtr_GO_Overlap_Root <- CdvsCtr_GO_Overlap1$Root
CdvsCtr_GO_Overlap_Root$logPvalue <- ifelse(CdvsCtr_GO_Overlap_Root$Regulation=="Down",CdvsCtr_GO_Overlap_Root$logPvalue*-1, CdvsCtr_GO_Overlap_Root$logPvalue)

CdvsCtr_GO_Overlap_Shoot <- CdvsCtr_GO_Overlap1$Shoot
CdvsCtr_GO_Overlap_Shoot$logPvalue <- ifelse(CdvsCtr_GO_Overlap_Shoot$Regulation=="Down",CdvsCtr_GO_Overlap_Shoot$logPvalue*-1, CdvsCtr_GO_Overlap_Shoot$logPvalue)

CdvsCtr_GO_Overlap_Root$just <- ifelse(CdvsCtr_GO_Overlap_Root$Regulation=="Down",0,1)
CdvsCtr_GO_Overlap_Shoot$just <- ifelse(CdvsCtr_GO_Overlap_Shoot$Regulation=="Down",0,1)

CdvsCtr_GO_Overlap_Root2 <- split(CdvsCtr_GO_Overlap_Root, CdvsCtr_GO_Overlap_Root$Population)
CdvsCtr_GO_Overlap_Root_Pais <- CdvsCtr_GO_Overlap_Root2$`Pais`
CdvsCtr_GO_Overlap_Root_Wall <- CdvsCtr_GO_Overlap_Root2$`Wall`
CdvsCtr_GO_Overlap_Shoot2 <- split(CdvsCtr_GO_Overlap_Shoot, CdvsCtr_GO_Overlap_Shoot$Population)
CdvsCtr_GO_Overlap_Shoot_Pais <- CdvsCtr_GO_Overlap_Shoot2$`Pais`
CdvsCtr_GO_Overlap_Shoot_Wall <- CdvsCtr_GO_Overlap_Shoot2$`Wall`

CdvsCtr_GO_Overlap_Root$Population <- factor(CdvsCtr_GO_Overlap_Root$Population, levels = c("Wall", "Pais"))
CdvsCtr_GO_Overlap_Shoot$Population <- factor(CdvsCtr_GO_Overlap_Shoot$Population, levels = c("Wall", "Pais"))

GO_PS_Root <- ggplot(CdvsCtr_GO_Overlap_Root, aes(y=logPvalue, x=reorder(GO_name, logPvalue), fill= Type, group=Population)) + 
      geom_bar(position=position_dodge(), stat="identity", colour="grey50", size=0.5) +
      facet_grid(Type ~ ., scales="free_y", space="free", labeller = label_bquote("")) +
      scale_y_continuous( limits=c(-15,15), breaks = c(-15,-10,-5,0,5,10,15), labels=c(15,10,5,0,5,10,15)) +
      scale_x_discrete("", labels=NULL) +
      coord_flip()  +
      labs(x= "", fill = "GO category", y=expression(
        paste(bold("Down-regulated"),"            -log(adjusted ", italic("P"), "-value)            ", bold("Up-regulated")))) + 
      theme_minimal() +
      geom_text(aes(x=GO_name, y=0, label=GO_name), hjust=CdvsCtr_GO_Overlap_Root$just, size = 6) +
      geom_text(aes(label=QanT), 
            position = position_stack(vjust = 0.5), size=5, fontface="bold") +
      theme(axis.text.y=element_text(size=15),axis.text.x=element_text(size=10), axis.title=element_text(size=12), strip.text.x =
            element_text(size=20), legend.text = element_text(size=12),legend.title = element_text(size=16)) +
      theme(panel.grid.major = element_blank()) +
      theme(panel.grid.minor = element_blank()) + 
      geom_hline(yintercept = 0, col= "grey40")  + 
      scale_fill_manual(values=c("brown1", "royalblue4", "maroon3")) 
 
   
GO_PS_Root

GO_PS_Shoot <- ggplot(CdvsCtr_GO_Overlap_Shoot, aes(y=logPvalue, x=reorder(GO_name, logPvalue), fill= Type, group=Population)) + 
      geom_bar(position=position_dodge(), stat="identity", colour="grey50", size=0.5) +
      facet_grid(Type ~ ., scales="free_y", space="free", labeller = label_bquote("")) +
      scale_y_continuous( limits=c(-15,15), breaks = c(-15,-10,-5,0,5,10,15), labels=c(15,10,5,0,5,10,15)) +
      scale_x_discrete("", labels=NULL) +
      coord_flip()  +
      labs(x= "", fill = "GO category", y=expression(
        paste(bold("Down-regulated"),"            -log(adjusted ", italic("P"), "-value)            ", bold("Up-regulated")))) + 
      theme_minimal() +
      geom_text(aes(x=GO_name, y=0, label=GO_name), hjust=CdvsCtr_GO_Overlap_Shoot$just, size = 6) +
      geom_text(aes(label=QanT), 
            position = position_stack(vjust = 0.5), size=5, fontface="bold") +
      theme(axis.text.y=element_text(size=15),axis.text.x=element_text(size=10), axis.title=element_text(size=12), strip.text.x =
            element_text(size=20), legend.text = element_text(size=12),legend.title = element_text(size=16)) +
      theme(panel.grid.major = element_blank()) +
      theme(panel.grid.minor = element_blank()) + 
      geom_hline(yintercept = 0, col= "grey40")  + 
      scale_fill_manual(values=c("brown1", "royalblue4", "maroon3")) 
 
   
GO_PS_Shoot

GO_legend <- g_legend(GO_PS_Root + theme(legend.position='bottom'))

grid.arrange(GO_PS_Root+theme(legend.position='hidden') + theme(axis.line.x = element_line(color="black", size = 0.5)),
             GO_PS_Shoot+theme(legend.position='hidden')+ theme(axis.line.x = element_line(color="black", size = 0.5)), 
             GO_legend,
             layout_matrix = cbind(c(1,NA,2,NA,3)),
             heights = c(1.5,0.1,1.75,0.1, 0.2)
             )




###########################################################################################################################
# Consitutively differentially exressed DEGs in Noss
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table")

New_GO_CDEG_Noss0 <- read.delim("New_GO_CDEG_Noss.txt")
New_GO_CDEG_Noss1 <- split(New_GO_CDEG_Noss0, New_GO_CDEG_Noss0$Tissue)
New_GO_CDEG_Noss_Root <- New_GO_CDEG_Noss1$Root
New_GO_CDEG_Noss_Shoot <- New_GO_CDEG_Noss1$Shoot

New_GO_CDEG_Noss_Root$logPvalue <- ifelse(New_GO_CDEG_Noss_Root$Regulation=="Down",New_GO_CDEG_Noss_Root$logPvalue*-1, New_GO_CDEG_Noss_Root$logPvalue)

New_GO_CDEG_Noss_Shoot$logPvalue <- ifelse(New_GO_CDEG_Noss_Shoot$Regulation=="Down",New_GO_CDEG_Noss_Shoot$logPvalue*-1, New_GO_CDEG_Noss_Shoot$logPvalue)


New_GO_CDEG_Noss_Root$just <- ifelse(New_GO_CDEG_Noss_Root$Regulation=="Down",0,1)
New_GO_CDEG_Noss_Shoot$just <- ifelse(New_GO_CDEG_Noss_Shoot$Regulation=="Down",0,1)


New_GO_CDEG_Noss_Root_gg <- ggplot(New_GO_CDEG_Noss_Root, aes(y=logPvalue, x=reorder(GO_name, logPvalue), fill= Type)) + 
      geom_bar( stat="identity") +  
      facet_grid(Type2 ~ ., scales="free_y", space="free", labeller = label_bquote("") ) + 
      scale_y_continuous( limits= c(-3,3), breaks = c(-3,-2,-1,0,1,2,3), labels=c(3,2,1,0,1,2,3)) +
      scale_x_discrete("", labels=NULL) +
      coord_flip()  +
      labs(x= "", fill = "GO category", y=expression(
        paste(bold("Lowly expressed in Noss"),"              -log(adjusted ", italic("P"), "-value)              ", bold("Highly expressed in Noss")))) + 
      theme_minimal() +
      geom_text(aes(x=GO_name, y=0, label=GO_name), hjust=New_GO_CDEG_Noss_Root$just, size = 6) +
      geom_text(aes(label=QanT), 
            position = position_stack(vjust = 0.5), size=5, fontface="bold") +
      theme(axis.text.y=element_text(size=15),axis.text.x=element_text(size=10), axis.title=element_text(size=12), strip.text.x =
            element_text(size=20), legend.text = element_text(size=12),legend.title = element_text(size=16)) +
      theme(panel.grid.major = element_blank()) +
      theme(panel.grid.minor = element_blank()) + 
      geom_hline(yintercept = 0, col= "grey40")  + 
      scale_fill_manual(values=c("brown1", "royalblue4"))

New_GO_CDEG_Noss_Root_gg

New_GO_CDEG_Noss_Shoot_gg <- ggplot(New_GO_CDEG_Noss_Shoot, aes(y=logPvalue, x=reorder(GO_name, logPvalue), fill= Type)) + 
      geom_bar( stat="identity") +  
      facet_grid(Type2 ~ ., scales="free_y", space="free", labeller = label_bquote("") ) + 
      scale_y_continuous( limits= c(-3,3), breaks = c(-3,-2,-1,0,1,2,3), labels=c(3,2,1,0,1,2,3)) +
      scale_x_discrete("", labels=NULL) +
      coord_flip()  +
      labs(x= "", fill = "GO category", y=expression(
        paste(bold("Lowly expressed in Noss"),"              -log(adjusted ", italic("P"), "-value)              ", bold("Highly expressed in Noss")))) + 
      theme_minimal() +
      geom_text(aes(x=GO_name, y=0, label=GO_name), hjust=New_GO_CDEG_Noss_Shoot$just, size = 6) +
      geom_text(aes(label=QanT), 
            position = position_stack(vjust = 0.5), size=5, fontface="bold") +
      theme(axis.text.y=element_text(size=15),axis.text.x=element_text(size=10), axis.title=element_text(size=12), strip.text.x =
            element_text(size=20), legend.text = element_text(size=12),legend.title = element_text(size=16)) +
      theme(panel.grid.major = element_blank()) +
      theme(panel.grid.minor = element_blank()) + 
      geom_hline(yintercept = 0, col= "grey40")  + 
      scale_fill_manual(values=c("brown1", "orange1","royalblue4"))
  
New_GO_CDEG_Noss_Shoot_gg  


GO_legend2 <- g_legend(New_GO_CDEG_Noss_Shoot_gg + theme(legend.position='bottom'))

grid.arrange(New_GO_CDEG_Noss_Root_gg+theme(legend.position='hidden') + theme(axis.line.x = element_line(color="black", size = 0.5)),
             New_GO_CDEG_Noss_Shoot_gg+theme(legend.position='hidden')+ theme(axis.line.x = element_line(color="black", size = 0.5)), 
             GO_legend2,
             layout_matrix = cbind(c(1,NA,2,NA,3)),
             heights = c(0.73,0.3,1.5,0.1,0.2)
             )



```

#Relative expression of interesting genes (Bar plot) with NCPK (Normalized counts per kilobase of gene length)
```{r}
#Make a NCPK table

#install.packages("datadr")
library(datadr)

setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table")

Gene_length <- read.delim("gene_length.txt")

NC <- read.delim("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Counts/Normalized_counts/Normalized_counts_woGTF.txt")
#NC <- as.data.frame(NormalizedCount_all)

#NC[,73] <- rownames(NC)
colnames(NC)[1] <- "genes"

NCPK0 <- merge(Gene_length, NC, by = "genes")

NCPK1 <- NCPK0[,4:75]

#??? NCPK0$PK_length
dim(NCPK1)

NCPK2 <- cbind(NCPK0[,1:3], NCPK1) 
rownames(NCPK1) <- NCPK2[,1]
NCPK_t <- t(NCPK1)

NCPK <- cbind(coldata, NCPK_t)
  
NCPK$Condition <- gsub("Ctr", "Control", NCPK$Condition) 

NCPK$Condition <- as.factor(NCPK$Condition)
NCPK_Root <- NCPK[37:72,]
NCPK_Shoot <- NCPK[1:36,]


```

# Prepare plots for each gene
```{r}
HMA2_Root.aov <- aov( g04213 ~ Population*Condition, data = NCPK_Root)
summary(HMA2_Root.aov)
TukeyHSD(HMA2_Root.aov)
HMA2_Root_res <- HSD.test(HMA2_Root.aov, c("Population", "Condition"), group=TRUE)
HMA2_Root_res$groups
oder.group_HMA2_Root <- HMA2_Root_res$groups[order(rownames(HMA2_Root_res$groups)),] 

HMA2_Shoot.aov <- aov( g04213 ~ Population*Condition, data = NCPK_Shoot)
summary(HMA2_Shoot.aov)
TukeyHSD(HMA2_Shoot.aov)  
HMA2_Shoot_res <- HSD.test(HMA2_Shoot.aov, c("Population", "Condition"), group=TRUE)
HMA2_Shoot_res$groups
oder.group_HMA2_Shoot <- HMA2_Shoot_res$groups[order(rownames(HMA2_Shoot_res$groups)),] 

HMA2_NC <- summarySE(NCPK, measurevar="g04213", groupvars=c("Tissue","Population","Condition"))
HMA2_NC$Condition <- factor(HMA2_NC$Condition, levels = c("Control", "Cd"))
HMA2 <- ggplot(HMA2_NC, aes(x=Population, y=g04213, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20") +
    labs(y="", title="HMA2", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g04213, ymax=g04213+sd),
                  size=.3, color = "grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g04213+sd+60, 
                  label=c(as.character(oder.group_HMA2_Root$groups),as.character(oder.group_HMA2_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

HMA2

NAS3_Root.aov <- aov( g18173 ~ Population*Condition, data = NCPK_Root)
summary(NAS3_Root.aov)
TukeyHSD(NAS3_Root.aov)
NAS3_Root_res <- HSD.test(NAS3_Root.aov, c("Population", "Condition"), group=TRUE)
NAS3_Root_res$groups
oder.group_NAS3_Root <- NAS3_Root_res$groups[order(rownames(NAS3_Root_res$groups)),] 

NAS3_Shoot.aov <- aov( g18173 ~ Population*Condition, data = NCPK_Shoot)
summary(NAS3_Shoot.aov)
TukeyHSD(NAS3_Shoot.aov)  
NAS3_Shoot_res <- HSD.test(NAS3_Shoot.aov, c("Population", "Condition"), group=TRUE)
NAS3_Shoot_res$groups
oder.group_NAS3_Shoot <- NAS3_Shoot_res$groups[order(rownames(NAS3_Shoot_res$groups)),] 

NAS3_NC <- summarySE(NCPK, measurevar="g18173", groupvars=c("Tissue","Population","Condition"))
NAS3_NC$Condition <- factor(NAS3_NC$Condition, levels = c("Control", "Cd"))
NAS3 <- ggplot(NAS3_NC, aes(x=Population, y=g18173, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="NAS3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g18173, ymax=g18173+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g18173+sd+35, 
                  label=c(as.character(oder.group_NAS3_Root$groups),as.character(oder.group_NAS3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

NAS3



ZIP6_Root.aov <- aov( g24253 ~ Population*Condition, data = NCPK_Root)
summary(ZIP6_Root.aov)
TukeyHSD(ZIP6_Root.aov)
ZIP6_Root_res <- HSD.test(ZIP6_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP6_Root_res$groups
oder.group_ZIP6_Root <- ZIP6_Root_res$groups[order(rownames(ZIP6_Root_res$groups)),] 

ZIP6_Shoot.aov <- aov( g24253 ~ Population*Condition, data = NCPK_Shoot)
summary(ZIP6_Shoot.aov)
TukeyHSD(ZIP6_Shoot.aov)  
ZIP6_Shoot_res <- HSD.test(ZIP6_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIP6_Shoot_res$groups
oder.group_ZIP6_Shoot <- ZIP6_Shoot_res$groups[order(rownames(ZIP6_Shoot_res$groups)),] 

ZIP6_NC <- summarySE(NCPK, measurevar="g24253", groupvars=c("Tissue","Population","Condition"))
ZIP6_NC$Condition <- factor(ZIP6_NC$Condition, levels = c("Control", "Cd"))
ZIP6 <- ggplot(ZIP6_NC, aes(x=Population, y=g24253, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZIP6", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g24253, ymax=g24253+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g24253+sd+100, 
                  label=c(as.character(oder.group_ZIP6_Root$groups),as.character(oder.group_ZIP6_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZIP6


VHA_E3_Root.aov <- aov( g32111 ~ Population*Condition, data = NCPK_Root)
summary(VHA_E3_Root.aov)
TukeyHSD(VHA_E3_Root.aov)
VHA_E3_Root_res <- HSD.test(VHA_E3_Root.aov, c("Population", "Condition"), group=TRUE)
VHA_E3_Root_res$groups
oder.group_VHA_E3_Root <- VHA_E3_Root_res$groups[order(rownames(VHA_E3_Root_res$groups)),] 

VHA_E3_Shoot.aov <- aov( g32111 ~ Population*Condition, data = NCPK_Shoot)
summary(VHA_E3_Shoot.aov)
TukeyHSD(VHA_E3_Shoot.aov)  
VHA_E3_Shoot_res <- HSD.test(VHA_E3_Shoot.aov, c("Population", "Condition"), group=TRUE)
VHA_E3_Shoot_res$groups
oder.group_VHA_E3_Shoot <- VHA_E3_Shoot_res$groups[order(rownames(VHA_E3_Shoot_res$groups)),] 

VHA_E3_NC <- summarySE(NCPK, measurevar="g32111", groupvars=c("Tissue","Population","Condition"))
VHA_E3_NC$Condition <- factor(VHA_E3_NC$Condition, levels = c("Control", "Cd"))
VHA_E3 <- ggplot(VHA_E3_NC, aes(x=Population, y=g32111, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="VHA_E3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g32111, ymax=g32111+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g32111+sd+9, 
                  label=c(as.character(oder.group_VHA_E3_Root$groups),as.character(oder.group_VHA_E3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

VHA_E3


ZIP7_like_Root.aov <- aov( g17415 ~ Population*Condition, data = NCPK_Root)
summary(ZIP7_like_Root.aov)
TukeyHSD(ZIP7_like_Root.aov)
ZIP7_like_Root_res <- HSD.test(ZIP7_like_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP7_like_Root_res$groups
oder.group_ZIP7_like_Root <- ZIP7_like_Root_res$groups[order(rownames(ZIP7_like_Root_res$groups)),] 

ZIP7_like_Shoot.aov <- aov( g17415 ~ Population*Condition, data = NCPK_Shoot)
summary(ZIP7_like_Shoot.aov)
TukeyHSD(ZIP7_like_Shoot.aov)  
ZIP7_like_Shoot_res <- HSD.test(ZIP7_like_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIP7_like_Shoot_res$groups
oder.group_ZIP7_like_Shoot <- ZIP7_like_Shoot_res$groups[order(rownames(ZIP7_like_Shoot_res$groups)),] 

ZIP7_like_NC <- summarySE(NCPK, measurevar="g17415", groupvars=c("Tissue","Population","Condition"))
ZIP7_like_NC$Condition <- factor(ZIP7_like_NC$Condition, levels = c("Control", "Cd"))
ZIP7_like <- ggplot(ZIP7_like_NC, aes(x=Population, y=g17415, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZIP7_like", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g17415, ymax=g17415+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g17415+sd+3, 
                  label=c(as.character(oder.group_ZIP7_like_Root$groups),as.character(oder.group_ZIP7_like_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZIP7_like


ZIP7_Root.aov <- aov( g23539 ~ Population*Condition, data = NCPK_Root)
summary(ZIP7_Root.aov)
TukeyHSD(ZIP7_Root.aov)
ZIP7_Root_res <- HSD.test(ZIP7_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP7_Root_res$groups
oder.group_ZIP7_Root <- ZIP7_Root_res$groups[order(rownames(ZIP7_Root_res$groups)),] 

ZIP7_Shoot.aov <- aov( g23539 ~ Population*Condition, data = NCPK_Shoot)
summary(ZIP7_Shoot.aov)
TukeyHSD(ZIP7_Shoot.aov)  
ZIP7_Shoot_res <- HSD.test(ZIP7_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIP7_Shoot_res$groups
oder.group_ZIP7_Shoot <- ZIP7_Shoot_res$groups[order(rownames(ZIP7_Shoot_res$groups)),] 

ZIP7_NC <- summarySE(NCPK, measurevar="g23539", groupvars=c("Tissue","Population","Condition"))
ZIP7_NC$Condition <- factor(ZIP7_NC$Condition, levels = c("Control", "Cd"))
ZIP7 <- ggplot(ZIP7_NC, aes(x=Population, y=g23539, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZIP7", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g23539, ymax=g23539+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g23539+sd+2.5, 
                  label=c(as.character(oder.group_ZIP7_Root$groups),as.character(oder.group_ZIP7_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZIP7

NRAMP4_Root.aov <- aov( g07320 ~ Population*Condition, data = NCPK_Root)
summary(NRAMP4_Root.aov)
TukeyHSD(NRAMP4_Root.aov)
NRAMP4_Root_res <- HSD.test(NRAMP4_Root.aov, c("Population", "Condition"), group=TRUE)
NRAMP4_Root_res$groups
oder.group_NRAMP4_Root <- NRAMP4_Root_res$groups[order(rownames(NRAMP4_Root_res$groups)),] 

NRAMP4_Shoot.aov <- aov( g07320 ~ Population*Condition, data = NCPK_Shoot)
summary(NRAMP4_Shoot.aov)
TukeyHSD(NRAMP4_Shoot.aov)  
NRAMP4_Shoot_res <- HSD.test(NRAMP4_Shoot.aov, c("Population", "Condition"), group=TRUE)
NRAMP4_Shoot_res$groups
oder.group_NRAMP4_Shoot <- NRAMP4_Shoot_res$groups[order(rownames(NRAMP4_Shoot_res$groups)),] 

NRAMP4_NC <- summarySE(NCPK, measurevar="g07320", groupvars=c("Tissue","Population","Condition"))
NRAMP4_NC$Condition <- factor(NRAMP4_NC$Condition, levels = c("Control", "Cd"))
NRAMP4 <- ggplot(NRAMP4_NC, aes(x=Population, y=g07320, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="NRAMP4", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g07320, ymax=g07320+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g07320+sd+380, 
                  label=c(as.character(oder.group_NRAMP4_Root$groups),as.character(oder.group_NRAMP4_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

NRAMP4


FRO1_Root.aov <- aov( g01760 ~ Population*Condition, data = NCPK_Root)
summary(FRO1_Root.aov)
TukeyHSD(FRO1_Root.aov)
FRO1_Root_res <- HSD.test(FRO1_Root.aov, c("Population", "Condition"), group=TRUE)
FRO1_Root_res$groups
oder.group_FRO1_Root <- FRO1_Root_res$groups[order(rownames(FRO1_Root_res$groups)),] 

FRO1_Shoot.aov <- aov( g01760 ~ Population*Condition, data = NCPK_Shoot)
summary(FRO1_Shoot.aov)
TukeyHSD(FRO1_Shoot.aov)  
FRO1_Shoot_res <- HSD.test(FRO1_Shoot.aov, c("Population", "Condition"), group=TRUE)
FRO1_Shoot_res$groups
oder.group_FRO1_Shoot <- FRO1_Shoot_res$groups[order(rownames(FRO1_Shoot_res$groups)),] 

FRO1_NC <- summarySE(NCPK, measurevar="g01760", groupvars=c("Tissue","Population","Condition"))
FRO1_NC$Condition <- factor(FRO1_NC$Condition, levels = c("Control", "Cd"))
FRO1 <- ggplot(FRO1_NC, aes(x=Population, y=g01760, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity",  color="grey20") +
    labs(y="", title="FRO1", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g01760, ymax=g01760+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g01760+sd+150, 
                  label=c(as.character(oder.group_FRO1_Root$groups),as.character(oder.group_FRO1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

FRO1


FRO5_Root.aov <- aov( g25255 ~ Population*Condition, data = NCPK_Root)
summary(FRO5_Root.aov)
TukeyHSD(FRO5_Root.aov)
FRO5_Root_res <- HSD.test(FRO5_Root.aov, c("Population", "Condition"), group=TRUE)
FRO5_Root_res$groups
oder.group_FRO5_Root <- FRO5_Root_res$groups[order(rownames(FRO5_Root_res$groups)),] 

#FRO5_Shoot.aov <- aov( g25255 ~ Population*Condition, data = NCPK_Shoot)
#summary(FRO5_Shoot.aov)
#TukeyHSD(FRO5_Shoot.aov)  
#FRO5_Shoot_res <- HSD.test(FRO5_Shoot.aov, c("Population", "Condition"), group=TRUE)
#FRO5_Shoot_res$groups
#oder.group_FRO5_Shoot <- FRO5_Shoot_res$groups[order(rownames(FRO5_Shoot_res$groups)),] 

FRO5_NC <- summarySE(NCPK, measurevar="g25255", groupvars=c("Population","Condition"))
FRO5_NC$Condition <- factor(FRO5_NC$Condition, levels = c("Control", "Cd"))
FRO5 <- ggplot(FRO5_NC, aes(x=Population, y=g25255, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity",  color="grey20") +
    labs(y="", title="FRO5", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g25255, ymax=g25255+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g25255+sd+1.5, 
                  label=c(as.character(oder.group_FRO5_Root$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

FRO5



FRO3_Root.aov <- aov( g13702 ~ Population*Condition, data = NCPK_Root)
summary(FRO3_Root.aov)
TukeyHSD(FRO3_Root.aov)
FRO3_Root_res <- HSD.test(FRO3_Root.aov, c("Population", "Condition"), group=TRUE)
FRO3_Root_res$groups
oder.group_FRO3_Root <- FRO3_Root_res$groups[order(rownames(FRO3_Root_res$groups)),] 

FRO3_Shoot.aov <- aov( g13702 ~ Population*Condition, data = NCPK_Shoot)
summary(FRO3_Shoot.aov)
TukeyHSD(FRO3_Shoot.aov)  
FRO3_Shoot_res <- HSD.test(FRO3_Shoot.aov, c("Population", "Condition"), group=TRUE)
FRO3_Shoot_res$groups
oder.group_FRO3_Shoot <- FRO3_Shoot_res$groups[order(rownames(FRO3_Shoot_res$groups)),] 

FRO3_NC <- summarySE(NCPK, measurevar="g13702", groupvars=c("Tissue","Population","Condition"))
FRO3_NC$Condition <- factor(FRO3_NC$Condition, levels = c("Control", "Cd"))
FRO3 <- ggplot(FRO3_NC, aes(x=Population, y=g13702, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="FRO3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g13702, ymax=g13702+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g13702+sd+230, 
                  label=c(as.character(oder.group_FRO3_Root$groups),as.character(oder.group_FRO3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

FRO3


SPL3_Root.aov <- aov( g03285 ~ Population*Condition, data = NCPK_Root)
summary(SPL3_Root.aov)
TukeyHSD(SPL3_Root.aov)
SPL3_Root_res <- HSD.test(SPL3_Root.aov, c("Population", "Condition"), group=TRUE)
SPL3_Root_res$groups
oder.group_SPL3_Root <- SPL3_Root_res$groups[order(rownames(SPL3_Root_res$groups)),] 

SPL3_Shoot.aov <- aov( g03285 ~ Population*Condition, data = NCPK_Shoot)
summary(SPL3_Shoot.aov)
TukeyHSD(SPL3_Shoot.aov)  
SPL3_Shoot_res <- HSD.test(SPL3_Shoot.aov, c("Population", "Condition"), group=TRUE)
SPL3_Shoot_res$groups
oder.group_SPL3_Shoot <- SPL3_Shoot_res$groups[order(rownames(SPL3_Shoot_res$groups)),] 

SPL3_NC <- summarySE(NCPK, measurevar="g03285", groupvars=c("Tissue","Population","Condition"))
SPL3_NC$Condition <- factor(SPL3_NC$Condition, levels = c("Control", "Cd"))
SPL3 <- ggplot(SPL3_NC, aes(x=Population, y=g03285, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="SPL3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g03285, ymax=g03285+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g03285+sd+38, 
                  label=c(as.character(oder.group_SPL3_Root$groups),as.character(oder.group_SPL3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

SPL3


FER4_Root.aov <- aov( g12617 ~ Population*Condition, data = NCPK_Root)
summary(FER4_Root.aov)
TukeyHSD(FER4_Root.aov)
FER4_Root_res <- HSD.test(FER4_Root.aov, c("Population", "Condition"), group=TRUE)
FER4_Root_res$groups
oder.group_FER4_Root <- FER4_Root_res$groups[order(rownames(FER4_Root_res$groups)),] 

FER4_Shoot.aov <- aov( g12617 ~ Population*Condition, data = NCPK_Shoot)
summary(FER4_Shoot.aov)
TukeyHSD(FER4_Shoot.aov)  
FER4_Shoot_res <- HSD.test(FER4_Shoot.aov, c("Population", "Condition"), group=TRUE)
FER4_Shoot_res$groups
oder.group_FER4_Shoot <- FER4_Shoot_res$groups[order(rownames(FER4_Shoot_res$groups)),] 

FER4_NC <- summarySE(NCPK, measurevar="g12617", groupvars=c("Tissue","Population","Condition"))
FER4_NC$Condition <- factor(FER4_NC$Condition, levels = c("Control", "Cd"))
FER4 <- ggplot(FER4_NC, aes(x=Population, y=g12617, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="FER4", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g12617, ymax=g12617+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g12617+sd+78, 
                  label=c(as.character(oder.group_FER4_Root$groups),as.character(oder.group_FER4_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

FER4


FER1_Root.aov <- aov( g27931 ~ Population*Condition, data = NCPK_Root)
summary(FER1_Root.aov)
TukeyHSD(FER1_Root.aov)
FER1_Root_res <- HSD.test(FER1_Root.aov, c("Population", "Condition"), group=TRUE)
FER1_Root_res$groups
oder.group_FER1_Root <- FER1_Root_res$groups[order(rownames(FER1_Root_res$groups)),] 

FER1_Shoot.aov <- aov( g27931 ~ Population*Condition, data = NCPK_Shoot)
summary(FER1_Shoot.aov)
TukeyHSD(FER1_Shoot.aov)  
FER1_Shoot_res <- HSD.test(FER1_Shoot.aov, c("Population", "Condition"), group=TRUE)
FER1_Shoot_res$groups
oder.group_FER1_Shoot <- FER1_Shoot_res$groups[order(rownames(FER1_Shoot_res$groups)),] 

FER1_NC <- summarySE(NCPK, measurevar="g27931", groupvars=c("Tissue","Population","Condition"))
FER1_NC$Condition <- factor(FER1_NC$Condition, levels = c("Control", "Cd"))
FER1 <- ggplot(FER1_NC, aes(x=Population, y=g27931, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="FER1", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g27931, ymax=g27931+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g27931+sd+350, 
                  label=c(as.character(oder.group_FER1_Root$groups),as.character(oder.group_FER1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

FER1

NAS4_Root.aov <- aov( g24765 ~ Population*Condition, data = NCPK_Root)
summary(NAS4_Root.aov)
TukeyHSD(NAS4_Root.aov)
NAS4_Root_res <- HSD.test(NAS4_Root.aov, c("Population", "Condition"), group=TRUE)
NAS4_Root_res$groups
oder.group_NAS4_Root <- NAS4_Root_res$groups[order(rownames(NAS4_Root_res$groups)),] 

NAS4_Shoot.aov <- aov( g24765 ~ Population*Condition, data = NCPK_Shoot)
summary(NAS4_Shoot.aov)
TukeyHSD(NAS4_Shoot.aov)  
NAS4_Shoot_res <- HSD.test(NAS4_Shoot.aov, c("Population", "Condition"), group=TRUE)
NAS4_Shoot_res$groups
oder.group_NAS4_Shoot <- NAS4_Shoot_res$groups[order(rownames(NAS4_Shoot_res$groups)),] 

NAS4_NC <- summarySE(NCPK, measurevar="g24765", groupvars=c("Tissue","Population","Condition"))
NAS4_NC$Condition <- factor(NAS4_NC$Condition, levels = c("Control", "Cd"))
NAS4 <- ggplot(NAS4_NC, aes(x=Population, y=g24765, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="NAS4", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g24765, ymax=g24765+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g24765+sd+35, 
                  label=c(as.character(oder.group_NAS4_Root$groups),as.character(oder.group_NAS4_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

NAS4


CCCl2_Root.aov <- aov( g13452 ~ Population*Condition, data = NCPK_Root)
summary(CCCl2_Root.aov)
TukeyHSD(CCCl2_Root.aov)
CCCl2_Root_res <- HSD.test(CCCl2_Root.aov, c("Population", "Condition"), group=TRUE)
CCCl2_Root_res$groups
oder.group_CCCl2_Root <- CCCl2_Root_res$groups[order(rownames(CCCl2_Root_res$groups)),] 

CCCl2_Shoot.aov <- aov( g13452 ~ Population*Condition, data = NCPK_Shoot)
summary(CCCl2_Shoot.aov)
TukeyHSD(CCCl2_Shoot.aov)  
CCCl2_Shoot_res <- HSD.test(CCCl2_Shoot.aov, c("Population", "Condition"), group=TRUE)
CCCl2_Shoot_res$groups
oder.group_CCCl2_Shoot <- CCCl2_Shoot_res$groups[order(rownames(CCCl2_Shoot_res$groups)),] 

CCCl2_NC <- summarySE(NCPK, measurevar="g13452", groupvars=c("Tissue","Population","Condition"))
CCCl2_NC$Condition <- factor(CCCl2_NC$Condition, levels = c("Control", "Cd"))
CCCl2 <- ggplot(CCCl2_NC, aes(x=Population, y=g13452, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="CCCl2", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g13452, ymax=g13452+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g13452+sd+16, 
                  label=c(as.character(oder.group_CCCl2_Root$groups),as.character(oder.group_CCCl2_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

CCCl2


CCCl3_Root.aov <- aov( g27355 ~ Population*Condition, data = NCPK_Root)
summary(CCCl3_Root.aov)
TukeyHSD(CCCl3_Root.aov)
CCCl3_Root_res <- HSD.test(CCCl3_Root.aov, c("Population", "Condition"), group=TRUE)
CCCl3_Root_res$groups
oder.group_CCCl3_Root <- CCCl3_Root_res$groups[order(rownames(CCCl3_Root_res$groups)),] 

CCCl3_Shoot.aov <- aov( g27355 ~ Population*Condition, data = NCPK_Shoot)
summary(CCCl3_Shoot.aov)
TukeyHSD(CCCl3_Shoot.aov)  
CCCl3_Shoot_res <- HSD.test(CCCl3_Shoot.aov, c("Population", "Condition"), group=TRUE)
CCCl3_Shoot_res$groups
oder.group_CCCl3_Shoot <- CCCl3_Shoot_res$groups[order(rownames(CCCl3_Shoot_res$groups)),] 

CCCl3_NC <- summarySE(NCPK, measurevar="g27355", groupvars=c("Tissue","Population","Condition"))
CCCl3_NC$Condition <- factor(CCCl3_NC$Condition, levels = c("Control", "Cd"))
CCCl3 <- ggplot(CCCl3_NC, aes(x=Population, y=g27355, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="CCCl3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g27355, ymax=g27355+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g27355+sd+1.5, 
                  label=c(as.character(oder.group_CCCl3_Root$groups),as.character(oder.group_CCCl3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

CCCl3


#CCCl4,5, CAX7?



VTL5_Root.aov <- aov( g30123 ~ Population*Condition, data = NCPK_Root)
summary(VTL5_Root.aov)
TukeyHSD(VTL5_Root.aov)
VTL5_Root_res <- HSD.test(VTL5_Root.aov, c("Population", "Condition"), group=TRUE)
VTL5_Root_res$groups
oder.group_VTL5_Root <- VTL5_Root_res$groups[order(rownames(VTL5_Root_res$groups)),] 

VTL5_Shoot.aov <- aov( g30123 ~ Population*Condition, data = NCPK_Shoot)
summary(VTL5_Shoot.aov)
TukeyHSD(VTL5_Shoot.aov)  
VTL5_Shoot_res <- HSD.test(VTL5_Shoot.aov, c("Population", "Condition"), group=TRUE)
VTL5_Shoot_res$groups
oder.group_VTL5_Shoot <- VTL5_Shoot_res$groups[order(rownames(VTL5_Shoot_res$groups)),] 

VTL5_NC <- summarySE(NCPK, measurevar="g30123", groupvars=c("Tissue","Population","Condition"))
VTL5_NC$Condition <- factor(VTL5_NC$Condition, levels = c("Control", "Cd"))
VTL5 <- ggplot(VTL5_NC, aes(x=Population, y=g30123, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="VTL5", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g30123, ymax=g30123+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g30123+sd+10, 
                  label=c(as.character(oder.group_VTL5_Root$groups),as.character(oder.group_VTL5_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

VTL5

VTL1_Root.aov <- aov( g13452 ~ Population*Condition, data = NCPK_Root)
summary(VTL1_Root.aov)
TukeyHSD(VTL1_Root.aov)
VTL1_Root_res <- HSD.test(VTL1_Root.aov, c("Population", "Condition"), group=TRUE)
VTL1_Root_res$groups
oder.group_VTL1_Root <- VTL1_Root_res$groups[order(rownames(VTL1_Root_res$groups)),] 

VTL1_Shoot.aov <- aov( g13452 ~ Population*Condition, data = NCPK_Shoot)
summary(VTL1_Shoot.aov)
TukeyHSD(VTL1_Shoot.aov)  
VTL1_Shoot_res <- HSD.test(VTL1_Shoot.aov, c("Population", "Condition"), group=TRUE)
VTL1_Shoot_res$groups
oder.group_VTL1_Shoot <- VTL1_Shoot_res$groups[order(rownames(VTL1_Shoot_res$groups)),] 

VTL1_NC <- summarySE(NCPK, measurevar="g13452", groupvars=c("Tissue","Population","Condition"))
VTL1_NC$Condition <- factor(VTL1_NC$Condition, levels = c("Control", "Cd"))
VTL1 <- ggplot(VTL1_NC, aes(x=Population, y=g13452, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="VTL1", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g13452, ymax=g13452+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g13452+sd+18, 
                  label=c(as.character(oder.group_VTL1_Root$groups),as.character(oder.group_VTL1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

VTL1



VTL2_Root.aov <- aov( g01236 ~ Population*Condition, data = NCPK_Root)
summary(VTL2_Root.aov)
TukeyHSD(VTL2_Root.aov)
VTL2_Root_res <- HSD.test(VTL2_Root.aov, c("Population", "Condition"), group=TRUE)
VTL2_Root_res$groups
oder.group_VTL2_Root <- VTL2_Root_res$groups[order(rownames(VTL2_Root_res$groups)),] 

VTL2_Shoot.aov <- aov( g01236 ~ Population*Condition, data = NCPK_Shoot)
summary(VTL2_Shoot.aov)
TukeyHSD(VTL2_Shoot.aov)  
VTL2_Shoot_res <- HSD.test(VTL2_Shoot.aov, c("Population", "Condition"), group=TRUE)
VTL2_Shoot_res$groups
oder.group_VTL2_Shoot <- VTL2_Shoot_res$groups[order(rownames(VTL2_Shoot_res$groups)),] 

VTL2_NC <- summarySE(NCPK, measurevar="g01236", groupvars=c("Tissue","Population","Condition"))
VTL2_NC$Condition <- factor(VTL2_NC$Condition, levels = c("Control", "Cd"))
VTL2 <- ggplot(VTL2_NC, aes(x=Population, y=g01236, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="VTL2", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g01236, ymax=g01236+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g01236+sd+5, 
                  label=c(as.character(oder.group_VTL2_Root$groups),as.character(oder.group_VTL2_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

VTL2

YSL1_Root.aov <- aov( g10414 ~ Population*Condition, data = NCPK_Root)
summary(YSL1_Root.aov)
TukeyHSD(YSL1_Root.aov)
YSL1_Root_res <- HSD.test(YSL1_Root.aov, c("Population", "Condition"), group=TRUE)
YSL1_Root_res$groups
oder.group_YSL1_Root <- YSL1_Root_res$groups[order(rownames(YSL1_Root_res$groups)),] 

YSL1_Shoot.aov <- aov( g10414 ~ Population*Condition, data = NCPK_Shoot)
summary(YSL1_Shoot.aov)
TukeyHSD(YSL1_Shoot.aov)  
YSL1_Shoot_res <- HSD.test(YSL1_Shoot.aov, c("Population", "Condition"), group=TRUE)
YSL1_Shoot_res$groups
oder.group_YSL1_Shoot <- YSL1_Shoot_res$groups[order(rownames(YSL1_Shoot_res$groups)),] 

YSL1_NC <- summarySE(NCPK, measurevar="g10414", groupvars=c("Tissue","Population","Condition"))
YSL1_NC$Condition <- factor(YSL1_NC$Condition, levels = c("Control", "Cd"))
YSL1 <- ggplot(YSL1_NC, aes(x=Population, y=g10414, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="YSL1", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g10414, ymax=g10414+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g10414+sd+40, 
                  label=c(as.character(oder.group_YSL1_Root$groups),as.character(oder.group_YSL1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

YSL1

AGO9_Root.aov <- aov( g12002 ~ Population*Condition, data = NCPK_Root)
summary(AGO9_Root.aov)
TukeyHSD(AGO9_Root.aov)
AGO9_Root_res <- HSD.test(AGO9_Root.aov, c("Population", "Condition"), group=TRUE)
AGO9_Root_res$groups
oder.group_AGO9_Root <- AGO9_Root_res$groups[order(rownames(AGO9_Root_res$groups)),] 

#
plot(AGO9_Root.aov, 1)
leveneTest(g12002 ~ Population*Condition, data = NCPK_Root) #Equal variance?
leveneTest(g12002 ~ Population, data = NCPK_Root)

plot(AGO9_Root.aov, 2)
histogram(AGO9_Root.aov$residuals)
shapiro.test(AGO9_Root.aov$residuals) #Normality?

AGO9_Root.fit = aov(g12002 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(AGO9_Root.fit)

AGO9_Root.kru_post <- dunnTest(g12002 ~ Population, data = NCPK_Root, method="bh")#Non-parametric KW test
AGO9_Root.kru_post

AGO9_Root.vanw <- vanWaerden.test(g12002 ~ Population, data=NCPK_Root)
AGO9_Root.vanw_post <- posthoc.vanWaerden.test(NCPK_Root$g12002, NCPK_Root$Population, "bonferroni")

durbin.test(g12002 ~ Population | Condition, data = NCPK_Root)
posthoc.durbin.test(g12002 ~ Population, NCPK_Root$Condition, data = NCPK_Root, p.adj="none")

NCPK_Root$log_g12002 <- log(NCPK_Root$g12002 + 0.01)
leveneTest(log_g12002 ~ Population*Condition, data = NCPK_Root)

AGO9_Root_log.aov <- aov( log_g12002 ~ Population*Condition, data = NCPK_Root)
shapiro.test(AGO9_Root_log.aov$residuals)
summary(AGO9_Root_log.aov)
TukeyHSD(AGO9_Root_log.aov)
AGO9_Root_log_res <- HSD.test(AGO9_Root_log.aov, c("Population", "Condition"), group=TRUE)
AGO9_Root_log_res$groups
#

AGO9_Shoot.aov <- aov( g12002 ~ Population*Condition, data = NCPK_Shoot)
summary(AGO9_Shoot.aov)
TukeyHSD(AGO9_Shoot.aov)  
AGO9_Shoot_res <- HSD.test(AGO9_Shoot.aov, c("Population", "Condition"), group=TRUE)
AGO9_Shoot_res$groups
oder.group_AGO9_Shoot <- AGO9_Shoot_res$groups[order(rownames(AGO9_Shoot_res$groups)),] 

#
plot(AGO9_Shoot.aov, 1)
leveneTest(g12002 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?
leveneTest(g12002 ~ Population, data = NCPK_Shoot)
plot(AGO9_Shoot.aov, 2)
histogram(AGO9_Shoot.aov$residuals)
shapiro.test(AGO9_Shoot.aov$residuals) #Normality?

AGO9_Shoot.fit = aov(g12002 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(AGO9_Shoot.fit)

AGO9_Shoot.kru_post <- dunnTest(g12002 ~ Population, data = NCPK_Shoot, method="bh") #Non-parametric KW test
AGO9_Shoot.kru_post

NCPK_Shoot$log_g12002 <- log10(NCPK_Shoot$g12002 + 0.01)
leveneTest(log_g12002 ~ Population*Condition, data = NCPK_Shoot)

AGO9_Shoot_log.aov <- aov( log_g12002 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(AGO9_Shoot_log.aov$residuals)
summary(AGO9_Shoot_log.aov)
TukeyHSD(AGO9_Shoot_log.aov)
AGO9_Shoot_log_res <- HSD.test(AGO9_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
AGO9_Shoot_log_res$groups
#

AGO9_NC <- summarySE(NCPK, measurevar="g12002", groupvars=c("Tissue","Population","Condition"))


AGO9_NC$Condition <- factor(AGO9_NC$Condition, levels = c("Control", "Cd"))
AGO9 <- ggplot(AGO9_NC, aes(x=Population, y=g12002, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="AGO9", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g12002, ymax=g12002+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g12002+sd+5, 
                  label=c(as.character(oder.group_AGO9_Root$groups),as.character(oder.group_AGO9_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

AGO9

ZYP1a_Root.aov <- aov( g13785 ~ Population*Condition, data = NCPK_Root)
summary(ZYP1a_Root.aov)
TukeyHSD(ZYP1a_Root.aov)
ZYP1a_Root_res <- HSD.test(ZYP1a_Root.aov, c("Population", "Condition"), group=TRUE)
ZYP1a_Root_res$groups
oder.group_ZYP1a_Root <- ZYP1a_Root_res$groups[order(rownames(ZYP1a_Root_res$groups)),] 

#
plot(ZYP1a_Root.aov, 1)
leveneTest(g13785 ~ Population*Condition, data = NCPK_Root) #Equal variance?

plot(ZYP1a_Root.aov, 2)
histogram(ZYP1a_Root.aov$residuals)
shapiro.test(ZYP1a_Root.aov$residuals) #Normality?

ZYP1a_Root.fit = aov(g13785 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(ZYP1a_Root.fit)

ZYP1a_Root.kru_post <- dunnTest(g13785 ~ Population, data = NCPK_Root, method="bh")#Non-parametric KW test
ZYP1a_Root.kru_post

NCPK_Root$log_g13785 <- log(NCPK_Root$g13785 + 0.01)
leveneTest(log_g13785 ~ Population*Condition, data = NCPK_Root)

ZYP1a_Root_log.aov <- aov( log_g13785 ~ Population*Condition, data = NCPK_Root)
shapiro.test(ZYP1a_Root_log.aov$residuals)
summary(ZYP1a_Root_log.aov)
TukeyHSD(ZYP1a_Root_log.aov)
ZYP1a_Root_log_res <- HSD.test(ZYP1a_Root_log.aov, c("Population", "Condition"), group=TRUE)
ZYP1a_Root_log_res$groups
#


ZYP1a_Shoot.aov <- aov( g13785 ~ Population*Condition, data = NCPK_Shoot)
summary(ZYP1a_Shoot.aov)
TukeyHSD(ZYP1a_Shoot.aov)  
ZYP1a_Shoot_res <- HSD.test(ZYP1a_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZYP1a_Shoot_res$groups
oder.group_ZYP1a_Shoot <- ZYP1a_Shoot_res$groups[order(rownames(ZYP1a_Shoot_res$groups)),] 

#
plot(ZYP1a_Shoot.aov, 1)
leveneTest(g13785 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

plot(ZYP1a_Shoot.aov, 2)
histogram(ZYP1a_Shoot.aov$residuals)
shapiro.test(ZYP1a_Shoot.aov$residuals) #Normality?

ZYP1a_Shoot.fit = aov(g13785 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(ZYP1a_Shoot.fit)

ZYP1a_Shoot.kru_post <- dunnTest(g13785 ~ Population, data = NCPK_Shoot, method="bh")#Non-parametric KW test
ZYP1a_Shoot.kru_post

NCPK_Shoot$log_g13785 <- log(NCPK_Shoot$g13785 + 0.01)
leveneTest(log_g13785 ~ Population*Condition, data = NCPK_Shoot)

ZYP1a_Shoot_log.aov <- aov( log_g13785 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(ZYP1a_Shoot_log.aov$residuals)
summary(ZYP1a_Shoot_log.aov)
TukeyHSD(ZYP1a_Shoot_log.aov)
ZYP1a_Shoot_log_res <- HSD.test(ZYP1a_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
ZYP1a_Shoot_log_res$groups
#


ZYP1a_NC <- summarySE(NCPK, measurevar="g13785", groupvars=c("Tissue","Population","Condition"))
ZYP1a_NC$Condition <- factor(ZYP1a_NC$Condition, levels = c("Control", "Cd"))
ZYP1a <- ggplot(ZYP1a_NC, aes(x=Population, y=g13785, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZYP1a", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g13785, ymax=g13785+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g13785+sd+1, 
                  label=c(as.character(oder.group_ZYP1a_Root$groups),as.character(oder.group_ZYP1a_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZYP1a

ZYP1b_Root.aov <- aov( g13782 ~ Population*Condition, data = NCPK_Root)
summary(ZYP1b_Root.aov)
TukeyHSD(ZYP1b_Root.aov)
ZYP1b_Root_res <- HSD.test(ZYP1b_Root.aov, c("Population", "Condition"), group=TRUE)
ZYP1b_Root_res$groups
oder.group_ZYP1b_Root <- ZYP1b_Root_res$groups[order(rownames(ZYP1b_Root_res$groups)),] 

#
plot(ZYP1b_Root.aov, 1)
leveneTest(g13782 ~ Population*Condition, data = NCPK_Root) #Equal variance?

plot(ZYP1b_Root.aov, 2)
histogram(ZYP1b_Root.aov$residuals)
shapiro.test(ZYP1b_Root.aov$residuals) #Normality?

ZYP1b_Root.fit = aov(g13782 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(ZYP1b_Root.fit)

ZYP1b_Root.kru_post <- dunnTest(g13782 ~ Population, data = NCPK_Root, method="bh")#Non-parametric KW test
ZYP1b_Root.kru_post

NCPK_Root$log_g13782 <- log(NCPK_Root$g13782 + 0.01)
leveneTest(log_g13782 ~ Population*Condition, data = NCPK_Root)

ZYP1b_Root_log.aov <- aov( log_g13782 ~ Population*Condition, data = NCPK_Root)
shapiro.test(ZYP1b_Root_log.aov$residuals)
summary(ZYP1b_Root_log.aov)
TukeyHSD(ZYP1b_Root_log.aov)
ZYP1b_Root_log_res <- HSD.test(ZYP1b_Root_log.aov, c("Population", "Condition"), group=TRUE)
ZYP1b_Root_log_res$groups
#


ZYP1b_Shoot.aov <- aov( g13782 ~ Population*Condition, data = NCPK_Shoot)
summary(ZYP1b_Shoot.aov)
TukeyHSD(ZYP1b_Shoot.aov)  
ZYP1b_Shoot_res <- HSD.test(ZYP1b_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZYP1b_Shoot_res$groups
oder.group_ZYP1b_Shoot <- ZYP1b_Shoot_res$groups[order(rownames(ZYP1b_Shoot_res$groups)),] 

#
plot(ZYP1b_Shoot.aov, 1)
leveneTest(g13782 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

plot(ZYP1b_Shoot.aov, 2)
histogram(ZYP1b_Shoot.aov$residuals)
shapiro.test(ZYP1b_Shoot.aov$residuals) #Normality?

ZYP1b_Shoot.fit = aov(g13782 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(ZYP1b_Shoot.fit)

ZYP1b_Shoot.kru_post <- dunnTest(g13782 ~ Population, data = NCPK_Shoot, method="bh")#Non-parametric KW test
ZYP1b_Shoot.kru_post

NCPK_Shoot$log_g13782 <- log(NCPK_Shoot$g13782 + 0.01)
leveneTest(log_g13782 ~ Population*Condition, data = NCPK_Shoot)

ZYP1b_Shoot_log.aov <- aov( log_g13782 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(ZYP1b_Shoot_log.aov$residuals)
summary(ZYP1b_Shoot_log.aov)
TukeyHSD(ZYP1b_Shoot_log.aov)
ZYP1b_Shoot_log_res <- HSD.test(ZYP1b_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
ZYP1b_Shoot_log_res$groups
#


ZYP1b_NC <- summarySE(NCPK, measurevar="g13782", groupvars=c("Tissue","Population","Condition"))
ZYP1b_NC$Condition <- factor(ZYP1b_NC$Condition, levels = c("Control", "Cd"))
ZYP1b <- ggplot(ZYP1b_NC, aes(x=Population, y=g13782, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZYP1b", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g13782, ymax=g13782+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g13782+sd+1, 
                  label=c(as.character(oder.group_ZYP1b_Root$groups),as.character(oder.group_ZYP1b_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZYP1b





SMC2_Root.aov <- aov( g12815 ~ Population*Condition, data = NCPK_Root)
summary(SMC2_Root.aov)
TukeyHSD(SMC2_Root.aov)
SMC2_Root_res <- HSD.test(SMC2_Root.aov, c("Population", "Condition"), group=TRUE)
SMC2_Root_res$groups
oder.group_SMC2_Root <- SMC2_Root_res$groups[order(rownames(SMC2_Root_res$groups)),] 

#
plot(SMC2_Root.aov, 1)
leveneTest(g12815 ~ Population*Condition, data = NCPK_Root) #Equal variance?

plot(SMC2_Root.aov, 2)
histogram(SMC2_Root.aov$residuals)
shapiro.test(SMC2_Root.aov$residuals) #Normality?

SMC2_Root.fit = aov(g12815 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(SMC2_Root.fit)

SMC2_Root.kru_post <- dunnTest(g12815 ~ Population, data = NCPK_Root, method="bh")#Non-parametric KW test
SMC2_Root.kru_post

NCPK_Root$log_g12815 <- log(NCPK_Root$g12815 + 0.01)
leveneTest(log_g12815 ~ Population*Condition, data = NCPK_Root)

SMC2_Root_log.aov <- aov( log_g12815 ~ Population*Condition, data = NCPK_Root)
shapiro.test(SMC2_Root_log.aov$residuals)
summary(SMC2_Root_log.aov)
TukeyHSD(SMC2_Root_log.aov)
SMC2_Root_log_res <- HSD.test(SMC2_Root_log.aov, c("Population", "Condition"), group=TRUE)
SMC2_Root_log_res$groups
#



SMC2_Shoot.aov <- aov( g12815 ~ Population*Condition, data = NCPK_Shoot)
summary(SMC2_Shoot.aov)
TukeyHSD(SMC2_Shoot.aov)  
SMC2_Shoot_res <- HSD.test(SMC2_Shoot.aov, c("Population", "Condition"), group=TRUE)
SMC2_Shoot_res$groups
oder.group_SMC2_Shoot <- SMC2_Shoot_res$groups[order(rownames(SMC2_Shoot_res$groups)),] 

#
plot(SMC2_Shoot.aov, 1)
leveneTest(g12815 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

plot(SMC2_Shoot.aov, 2)
histogram(SMC2_Shoot.aov$residuals)
shapiro.test(SMC2_Shoot.aov$residuals) #Normality?

SMC2_Shoot.fit = aov(g12815 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(SMC2_Shoot.fit)

SMC2_Shoot.kru_post <- dunnTest(g12815 ~ Population, data = NCPK_Shoot, method="bh")#Non-parametric KW test
SMC2_Shoot.kru_post

NCPK_Shoot$log_g12815 <- log(NCPK_Shoot$g12815 + 0.01)
leveneTest(log_g12815 ~ Population*Condition, data = NCPK_Shoot)

SMC2_Shoot_log.aov <- aov( log_g12815 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(SMC2_Shoot_log.aov$residuals)
summary(SMC2_Shoot_log.aov)
TukeyHSD(SMC2_Shoot_log.aov)
SMC2_Shoot_log_res <- HSD.test(SMC2_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
SMC2_Shoot_log_res$groups
#


SMC2_NC <- summarySE(NCPK, measurevar="g12815", groupvars=c("Tissue","Population","Condition"))
SMC2_NC$Condition <- factor(SMC2_NC$Condition, levels = c("Control", "Cd"))
SMC2 <- ggplot(SMC2_NC, aes(x=Population, y=g12815, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="SMC2", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g12815, ymax=g12815+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g12815+sd+3, 
                  label=c(as.character(oder.group_SMC2_Root$groups),as.character(oder.group_SMC2_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

SMC2


BIP3_Root.aov <- aov( g18192 ~ Population*Condition, data = NCPK_Root)
summary(BIP3_Root.aov)
TukeyHSD(BIP3_Root.aov)
BIP3_Root_res <- HSD.test(BIP3_Root.aov, c("Population", "Condition"), group=TRUE)
BIP3_Root_res$groups
oder.group_BIP3_Root <- BIP3_Root_res$groups[order(rownames(BIP3_Root_res$groups)),] 

BIP3_Shoot.aov <- aov( g18192 ~ Population*Condition, data = NCPK_Shoot)
summary(BIP3_Shoot.aov)
TukeyHSD(BIP3_Shoot.aov)  
BIP3_Shoot_res <- HSD.test(BIP3_Shoot.aov, c("Population", "Condition"), group=TRUE)
BIP3_Shoot_res$groups
oder.group_BIP3_Shoot <- BIP3_Shoot_res$groups[order(rownames(BIP3_Shoot_res$groups)),] 

BIP3_NC <- summarySE(NCPK, measurevar="g18192", groupvars=c("Tissue","Population","Condition"))
BIP3_NC$Condition <- factor(BIP3_NC$Condition, levels = c("Control", "Cd"))
BIP3 <- ggplot(BIP3_NC, aes(x=Population, y=g18192, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="BIP3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g18192, ymax=g18192+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g18192+sd+40, 
                  label=c(as.character(oder.group_BIP3_Root$groups),as.character(oder.group_BIP3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

BIP3

g19002_Root.aov <- aov( g19002 ~ Population*Condition, data = NCPK_Root)
summary(g19002_Root.aov)
TukeyHSD(g19002_Root.aov)
g19002_Root_res <- HSD.test(g19002_Root.aov, c("Population", "Condition"), group=TRUE)
g19002_Root_res$groups
oder.group_g19002_Root <- g19002_Root_res$groups[order(rownames(g19002_Root_res$groups)),] 

g19002_Shoot.aov <- aov( g19002 ~ Population*Condition, data = NCPK_Shoot)
summary(g19002_Shoot.aov)
TukeyHSD(g19002_Shoot.aov)  
g19002_Shoot_res <- HSD.test(g19002_Shoot.aov, c("Population", "Condition"), group=TRUE)
g19002_Shoot_res$groups
oder.group_g19002_Shoot <- g19002_Shoot_res$groups[order(rownames(g19002_Shoot_res$groups)),] 

g19002_NC <- summarySE(NCPK, measurevar="g19002", groupvars=c("Tissue","Population","Condition"))
g19002_NC$Condition <- factor(g19002_NC$Condition, levels = c("Control", "Cd"))
g19002 <- ggplot(g19002_NC, aes(x=Population, y=g19002, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="g19002", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g19002, ymax=g19002+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g19002+sd+100, 
                  label=c(as.character(oder.group_g19002_Root$groups),as.character(oder.group_g19002_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

g19002

g19003_Root.aov <- aov( g19003 ~ Population*Condition, data = NCPK_Root)
summary(g19003_Root.aov)
TukeyHSD(g19003_Root.aov)
g19003_Root_res <- HSD.test(g19003_Root.aov, c("Population", "Condition"), group=TRUE)
g19003_Root_res$groups
oder.group_g19003_Root <- g19003_Root_res$groups[order(rownames(g19003_Root_res$groups)),] 

g19003_Shoot.aov <- aov( g19003 ~ Population*Condition, data = NCPK_Shoot)
summary(g19003_Shoot.aov)
TukeyHSD(g19003_Shoot.aov)  
g19003_Shoot_res <- HSD.test(g19003_Shoot.aov, c("Population", "Condition"), group=TRUE)
g19003_Shoot_res$groups
oder.group_g19003_Shoot <- g19003_Shoot_res$groups[order(rownames(g19003_Shoot_res$groups)),] 

g19003_NC <- summarySE(NCPK, measurevar="g19003", groupvars=c("Tissue","Population","Condition"))
g19003_NC$Condition <- factor(g19003_NC$Condition, levels = c("Control", "Cd"))
g19003 <- ggplot(g19003_NC, aes(x=Population, y=g19003, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="g19003", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g19003, ymax=g19003+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g19003+sd+100, 
                  label=c(as.character(oder.group_g19003_Root$groups),as.character(oder.group_g19003_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

g19003

g19004_Root.aov <- aov( g19004 ~ Population*Condition, data = NCPK_Root)
summary(g19004_Root.aov)
TukeyHSD(g19004_Root.aov)
g19004_Root_res <- HSD.test(g19004_Root.aov, c("Population", "Condition"), group=TRUE)
g19004_Root_res$groups
oder.group_g19004_Root <- g19004_Root_res$groups[order(rownames(g19004_Root_res$groups)),] 

g19004_Shoot.aov <- aov( g19004 ~ Population*Condition, data = NCPK_Shoot)
summary(g19004_Shoot.aov)
TukeyHSD(g19004_Shoot.aov)  
g19004_Shoot_res <- HSD.test(g19004_Shoot.aov, c("Population", "Condition"), group=TRUE)
g19004_Shoot_res$groups
oder.group_g19004_Shoot <- g19004_Shoot_res$groups[order(rownames(g19004_Shoot_res$groups)),] 

g19004_NC <- summarySE(NCPK, measurevar="g19004", groupvars=c("Tissue","Population","Condition"))
g19004_NC$Condition <- factor(g19004_NC$Condition, levels = c("Control", "Cd"))
g19004 <- ggplot(g19004_NC, aes(x=Population, y=g19004, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="g19004", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g19004, ymax=g19004+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g19004+sd+100, 
                  label=c(as.character(oder.group_g19004_Root$groups),as.character(oder.group_g19004_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

g19004

g19005_Root.aov <- aov( g19005 ~ Population*Condition, data = NCPK_Root)
summary(g19005_Root.aov)
TukeyHSD(g19005_Root.aov)
g19005_Root_res <- HSD.test(g19005_Root.aov, c("Population", "Condition"), group=TRUE)
g19005_Root_res$groups
oder.group_g19005_Root <- g19005_Root_res$groups[order(rownames(g19005_Root_res$groups)),] 

g19005_Shoot.aov <- aov( g19005 ~ Population*Condition, data = NCPK_Shoot)
summary(g19005_Shoot.aov)
TukeyHSD(g19005_Shoot.aov)  
g19005_Shoot_res <- HSD.test(g19005_Shoot.aov, c("Population", "Condition"), group=TRUE)
g19005_Shoot_res$groups
oder.group_g19005_Shoot <- g19005_Shoot_res$groups[order(rownames(g19005_Shoot_res$groups)),] 

g19005_NC <- summarySE(NCPK, measurevar="g19005", groupvars=c("Tissue","Population","Condition"))
g19005_NC$Condition <- factor(g19005_NC$Condition, levels = c("Control", "Cd"))
g19005 <- ggplot(g19005_NC, aes(x=Population, y=g19005, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="g19005", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g19005, ymax=g19005+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g19005+sd+80, 
                  label=c(as.character(oder.group_g19005_Root$groups),as.character(oder.group_g19005_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

g19005

OPT3_Root.aov <- aov( g23102 ~ Population*Condition, data = NCPK_Root)
summary(OPT3_Root.aov)
TukeyHSD(OPT3_Root.aov)
OPT3_Root_res <- HSD.test(OPT3_Root.aov, c("Population", "Condition"), group=TRUE)
OPT3_Root_res$groups
oder.group_OPT3_Root <- OPT3_Root_res$groups[order(rownames(OPT3_Root_res$groups)),] 

OPT3_Shoot.aov <- aov( g23102 ~ Population*Condition, data = NCPK_Shoot)
summary(OPT3_Shoot.aov)
TukeyHSD(OPT3_Shoot.aov)  
OPT3_Shoot_res <- HSD.test(OPT3_Shoot.aov, c("Population", "Condition"), group=TRUE)
OPT3_Shoot_res$groups
oder.group_OPT3_Shoot <- OPT3_Shoot_res$groups[order(rownames(OPT3_Shoot_res$groups)),] 

OPT3_NC <- summarySE(NCPK, measurevar="g23102", groupvars=c("Tissue","Population","Condition"))
OPT3_NC$Condition <- factor(OPT3_NC$Condition, levels = c("Control", "Cd"))
OPT3 <- ggplot(OPT3_NC, aes(x=Population, y=g23102, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="OPT3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g23102, ymax=g23102+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g23102+sd+600, 
                  label=c(as.character(oder.group_OPT3_Root$groups),as.character(oder.group_OPT3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

OPT3

ZIP11_Root.aov <- aov( g24576 ~ Population*Condition, data = NCPK_Root)
summary(ZIP11_Root.aov)
TukeyHSD(ZIP11_Root.aov)
ZIP11_Root_res <- HSD.test(ZIP11_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP11_Root_res$groups
oder.group_ZIP11_Root <- ZIP11_Root_res$groups[order(rownames(ZIP11_Root_res$groups)),] 

ZIP11_Shoot.aov <- aov( g24576 ~ Population*Condition, data = NCPK_Shoot)
summary(ZIP11_Shoot.aov)
TukeyHSD(ZIP11_Shoot.aov)  
ZIP11_Shoot_res <- HSD.test(ZIP11_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIP11_Shoot_res$groups
oder.group_ZIP11_Shoot <- ZIP11_Shoot_res$groups[order(rownames(ZIP11_Shoot_res$groups)),] 

ZIP11_NC <- summarySE(NCPK, measurevar="g24576", groupvars=c("Tissue","Population","Condition"))
ZIP11_NC$Condition <- factor(ZIP11_NC$Condition, levels = c("Control", "Cd"))
ZIP11 <- ggplot(ZIP11_NC, aes(x=Population, y=g24576, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZIP11", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g24576, ymax=g24576+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g24576+sd+12, 
                  label=c(as.character(oder.group_ZIP11_Root$groups),as.character(oder.group_ZIP11_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZIP11


ZIP8_Root.aov <- aov( g29372 ~ Population*Condition, data = NCPK_Root)
summary(ZIP8_Root.aov)
TukeyHSD(ZIP8_Root.aov)
ZIP8_Root_res <- HSD.test(ZIP8_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP8_Root_res$groups
oder.group_ZIP8_Root <- ZIP8_Root_res$groups[order(rownames(ZIP8_Root_res$groups)),] 

ZIP8_Shoot.aov <- aov( g29372 ~ Population*Condition, data = NCPK_Shoot)
summary(ZIP8_Shoot.aov)
TukeyHSD(ZIP8_Shoot.aov)  
ZIP8_Shoot_res <- HSD.test(ZIP8_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIP8_Shoot_res$groups
oder.group_ZIP8_Shoot <- ZIP8_Shoot_res$groups[order(rownames(ZIP8_Shoot_res$groups)),] 

ZIP8_NC <- summarySE(NCPK, measurevar="g29372", groupvars=c("Tissue","Population","Condition"))
ZIP8_NC$Condition <- factor(ZIP8_NC$Condition, levels = c("Control", "Cd"))
ZIP8 <- ggplot(ZIP8_NC, aes(x=Population, y=g29372, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZIP8", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g29372, ymax=g29372+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g29372+sd+8, 
                  label=c(as.character(oder.group_ZIP8_Root$groups),as.character(oder.group_ZIP8_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZIP8

ZIF1_Root.aov <- aov( g19325 ~ Population*Condition, data = NCPK_Root)
summary(ZIF1_Root.aov)
TukeyHSD(ZIF1_Root.aov)
ZIF1_Root_res <- HSD.test(ZIF1_Root.aov, c("Population", "Condition"), group=TRUE)
ZIF1_Root_res$groups
oder.group_ZIF1_Root <- ZIF1_Root_res$groups[order(rownames(ZIF1_Root_res$groups)),] 

ZIF1_Shoot.aov <- aov( g19325 ~ Population*Condition, data = NCPK_Shoot)
summary(ZIF1_Shoot.aov)
TukeyHSD(ZIF1_Shoot.aov)  
ZIF1_Shoot_res <- HSD.test(ZIF1_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIF1_Shoot_res$groups
oder.group_ZIF1_Shoot <- ZIF1_Shoot_res$groups[order(rownames(ZIF1_Shoot_res$groups)),] 

ZIF1_NC <- summarySE(NCPK, measurevar="g19325", groupvars=c("Tissue","Population","Condition"))
ZIF1_NC$Condition <- factor(ZIF1_NC$Condition, levels = c("Control", "Cd"))
ZIF1 <- ggplot(ZIF1_NC, aes(x=Population, y=g19325, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZIF1", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g19325, ymax=g19325+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g19325+sd+260, 
                  label=c(as.character(oder.group_ZIF1_Root$groups),as.character(oder.group_ZIF1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZIF1


CAX3_Root.aov <- aov( g14842 ~ Population*Condition, data = NCPK_Root)
summary(CAX3_Root.aov)
TukeyHSD(CAX3_Root.aov)
CAX3_Root_res <- HSD.test(CAX3_Root.aov, c("Population", "Condition"), group=TRUE)
CAX3_Root_res$groups
oder.group_CAX3_Root <- CAX3_Root_res$groups[order(rownames(CAX3_Root_res$groups)),] 

CAX3_Shoot.aov <- aov( g14842 ~ Population*Condition, data = NCPK_Shoot)
summary(CAX3_Shoot.aov)
TukeyHSD(CAX3_Shoot.aov)  
CAX3_Shoot_res <- HSD.test(CAX3_Shoot.aov, c("Population", "Condition"), group=TRUE)
CAX3_Shoot_res$groups
oder.group_CAX3_Shoot <- CAX3_Shoot_res$groups[order(rownames(CAX3_Shoot_res$groups)),] 

CAX3_NC <- summarySE(NCPK, measurevar="g14842", groupvars=c("Tissue","Population","Condition"))
CAX3_NC$Condition <- factor(CAX3_NC$Condition, levels = c("Control", "Cd"))
CAX3 <- ggplot(CAX3_NC, aes(x=Population, y=g14842, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="CAX3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g14842, ymax=g14842+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g14842+sd+15, 
                  label=c(as.character(oder.group_CAX3_Root$groups),as.character(oder.group_CAX3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

CAX3


FRO2_Root.aov <- aov( g01761 ~ Population*Condition, data = NCPK_Root)
summary(FRO2_Root.aov)
TukeyHSD(FRO2_Root.aov)
FRO2_Root_res <- HSD.test(FRO2_Root.aov, c("Population", "Condition"), group=TRUE)
FRO2_Root_res$groups
oder.group_FRO2_Root <- FRO2_Root_res$groups[order(rownames(FRO2_Root_res$groups)),] 

FRO2_Shoot.aov <- aov( g01761 ~ Population*Condition, data = NCPK_Shoot)
summary(FRO2_Shoot.aov)
TukeyHSD(FRO2_Shoot.aov)  
FRO2_Shoot_res <- HSD.test(FRO2_Shoot.aov, c("Population", "Condition"), group=TRUE)
FRO2_Shoot_res$groups
oder.group_FRO2_Shoot <- FRO2_Shoot_res$groups[order(rownames(FRO2_Shoot_res$groups)),] 

FRO2_NC <- summarySE(NCPK, measurevar="g01761", groupvars=c("Tissue","Population","Condition"))
FRO2_NC$Condition <- factor(FRO2_NC$Condition, levels = c("Control", "Cd"))
FRO2 <- ggplot(FRO2_NC, aes(x=Population, y=g01761, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="FRO2", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g01761, ymax=g01761+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g01761+sd+73, 
                  label=c(as.character(oder.group_FRO2_Root$groups),as.character(oder.group_FRO2_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

FRO2


PYE_Root.aov <- aov( g12843 ~ Population*Condition, data = NCPK_Root)
summary(PYE_Root.aov)
TukeyHSD(PYE_Root.aov)
PYE_Root_res <- HSD.test(PYE_Root.aov, c("Population", "Condition"), group=TRUE)
PYE_Root_res$groups
oder.group_PYE_Root <- PYE_Root_res$groups[order(rownames(PYE_Root_res$groups)),] 

PYE_Shoot.aov <- aov( g12843 ~ Population*Condition, data = NCPK_Shoot)
summary(PYE_Shoot.aov)
TukeyHSD(PYE_Shoot.aov)  
PYE_Shoot_res <- HSD.test(PYE_Shoot.aov, c("Population", "Condition"), group=TRUE)
PYE_Shoot_res$groups
oder.group_PYE_Shoot <- PYE_Shoot_res$groups[order(rownames(PYE_Shoot_res$groups)),] 

PYE_NC <- summarySE(NCPK, measurevar="g12843", groupvars=c("Tissue","Population","Condition"))
PYE_NC$Condition <- factor(PYE_NC$Condition, levels = c("Control", "Cd"))
PYE <- ggplot(PYE_NC, aes(x=Population, y=g12843, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="PYE", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g12843, ymax=g12843+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g12843+sd+42, 
                  label=c(as.character(oder.group_PYE_Root$groups),as.character(oder.group_PYE_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

PYE


SPL15_Root.aov <- aov( g16599 ~ Population*Condition, data = NCPK_Root)
summary(SPL15_Root.aov)
TukeyHSD(SPL15_Root.aov)
SPL15_Root_res <- HSD.test(SPL15_Root.aov, c("Population", "Condition"), group=TRUE)
SPL15_Root_res$groups
oder.group_SPL15_Root <- SPL15_Root_res$groups[order(rownames(SPL15_Root_res$groups)),] 

SPL15_Shoot.aov <- aov( g16599 ~ Population*Condition, data = NCPK_Shoot)
summary(SPL15_Shoot.aov)
TukeyHSD(SPL15_Shoot.aov)  
SPL15_Shoot_res <- HSD.test(SPL15_Shoot.aov, c("Population", "Condition"), group=TRUE)
SPL15_Shoot_res$groups
oder.group_SPL15_Shoot <- SPL15_Shoot_res$groups[order(rownames(SPL15_Shoot_res$groups)),] 

SPL15_NC <- summarySE(NCPK, measurevar="g16599", groupvars=c("Tissue","Population","Condition"))
SPL15_NC$Condition <- factor(SPL15_NC$Condition, levels = c("Control", "Cd"))
SPL15 <- ggplot(SPL15_NC, aes(x=Population, y=g16599, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="SPL15", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g16599, ymax=g16599+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g16599+sd+3.5, 
                  label=c(as.character(oder.group_SPL15_Root$groups),as.character(oder.group_SPL15_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

SPL15

CAX7_Root.aov <- aov( g13319 ~ Population*Condition, data = NCPK_Root)
summary(CAX7_Root.aov)
TukeyHSD(CAX7_Root.aov)
CAX7_Root_res <- HSD.test(CAX7_Root.aov, c("Population", "Condition"), group=TRUE)
CAX7_Root_res$groups
oder.group_CAX7_Root <- CAX7_Root_res$groups[order(rownames(CAX7_Root_res$groups)),] 

CAX7_Shoot.aov <- aov( g13319 ~ Population*Condition, data = NCPK_Shoot)
summary(CAX7_Shoot.aov)
TukeyHSD(CAX7_Shoot.aov)  
CAX7_Shoot_res <- HSD.test(CAX7_Shoot.aov, c("Population", "Condition"), group=TRUE)
CAX7_Shoot_res$groups
oder.group_CAX7_Shoot <- CAX7_Shoot_res$groups[order(rownames(CAX7_Shoot_res$groups)),] 

CAX7_NC <- summarySE(NCPK, measurevar="g13319", groupvars=c("Tissue","Population","Condition"))
CAX7_NC$Condition <- factor(CAX7_NC$Condition, levels = c("Control", "Cd"))
CAX7 <- ggplot(CAX7_NC, aes(x=Population, y=g13319, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="CAX7", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g13319, ymax=g13319+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g13319+sd+25, 
                  label=c(as.character(oder.group_CAX7_Root$groups),as.character(oder.group_CAX7_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

CAX7

ZIP2_Root.aov <- aov( g11007 ~ Population*Condition, data = NCPK_Root)
summary(ZIP2_Root.aov)
TukeyHSD(ZIP2_Root.aov)
ZIP2_Root_res <- HSD.test(ZIP2_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP2_Root_res$groups
oder.group_ZIP2_Root <- ZIP2_Root_res$groups[order(rownames(ZIP2_Root_res$groups)),] 

ZIP2_Shoot.aov <- aov( g11007 ~ Population*Condition, data = NCPK_Shoot)
summary(ZIP2_Shoot.aov)
TukeyHSD(ZIP2_Shoot.aov)  
ZIP2_Shoot_res <- HSD.test(ZIP2_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIP2_Shoot_res$groups
oder.group_ZIP2_Shoot <- ZIP2_Shoot_res$groups[order(rownames(ZIP2_Shoot_res$groups)),] 

ZIP2_NC <- summarySE(NCPK, measurevar="g11007", groupvars=c("Tissue","Population","Condition"))
ZIP2_NC$Condition <- factor(ZIP2_NC$Condition, levels = c("Control", "Cd"))
ZIP2 <- ggplot(ZIP2_NC, aes(x=Population, y=g11007, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZIP2", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g11007, ymax=g11007+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g11007+sd+13, 
                  label=c(as.character(oder.group_ZIP2_Root$groups),as.character(oder.group_ZIP2_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZIP2

FRO4_Root.aov <- aov( g25254 ~ Population*Condition, data = NCPK_Root)
summary(FRO4_Root.aov)
TukeyHSD(FRO4_Root.aov)
FRO4_Root_res <- HSD.test(FRO4_Root.aov, c("Population", "Condition"), group=TRUE)
FRO4_Root_res$groups
oder.group_FRO4_Root <- FRO4_Root_res$groups[order(rownames(FRO4_Root_res$groups)),] 

FRO4_Shoot.aov <- aov( g25254 ~ Population*Condition, data = NCPK_Shoot)
summary(FRO4_Shoot.aov)
TukeyHSD(FRO4_Shoot.aov)  
FRO4_Shoot_res <- HSD.test(FRO4_Shoot.aov, c("Population", "Condition"), group=TRUE)
FRO4_Shoot_res$groups
oder.group_FRO4_Shoot <- FRO4_Shoot_res$groups[order(rownames(FRO4_Shoot_res$groups)),] 

FRO4_NC <- summarySE(NCPK, measurevar="g25254", groupvars=c("Tissue","Population","Condition"))
FRO4_NC$Condition <- factor(FRO4_NC$Condition, levels = c("Control", "Cd"))
FRO4 <- ggplot(FRO4_NC, aes(x=Population, y=g25254, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="FRO4", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g25254, ymax=g25254+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g25254+sd+37, 
                  label=c(as.character(oder.group_FRO4_Root$groups),as.character(oder.group_FRO4_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

FRO4

PCR8_Root.aov <- aov( g15174 ~ Population*Condition, data = NCPK_Root)
summary(PCR8_Root.aov)
TukeyHSD(PCR8_Root.aov)
PCR8_Root_res <- HSD.test(PCR8_Root.aov, c("Population", "Condition"), group=TRUE)
PCR8_Root_res$groups
oder.group_PCR8_Root <- PCR8_Root_res$groups[order(rownames(PCR8_Root_res$groups)),] 

PCR8_Shoot.aov <- aov( g15174 ~ Population*Condition, data = NCPK_Shoot)
summary(PCR8_Shoot.aov)
TukeyHSD(PCR8_Shoot.aov)  
PCR8_Shoot_res <- HSD.test(PCR8_Shoot.aov, c("Population", "Condition"), group=TRUE)
PCR8_Shoot_res$groups
oder.group_PCR8_Shoot <- PCR8_Shoot_res$groups[order(rownames(PCR8_Shoot_res$groups)),] 

PCR8_NC <- summarySE(NCPK, measurevar="g15174", groupvars=c("Tissue","Population","Condition"))
PCR8_NC$Condition <- factor(PCR8_NC$Condition, levels = c("Control", "Cd"))
PCR8 <- ggplot(PCR8_NC, aes(x=Population, y=g15174, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="PCR8", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g15174, ymax=g15174+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g15174+sd+37, 
                  label=c(as.character(oder.group_PCR8_Root$groups),as.character(oder.group_PCR8_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

PCR8

PCR9_Root.aov <- aov( g24696 ~ Population*Condition, data = NCPK_Root)
summary(PCR9_Root.aov)
TukeyHSD(PCR9_Root.aov)
PCR9_Root_res <- HSD.test(PCR9_Root.aov, c("Population", "Condition"), group=TRUE)
PCR9_Root_res$groups
oder.group_PCR9_Root <- PCR9_Root_res$groups[order(rownames(PCR9_Root_res$groups)),] 

PCR9_Shoot.aov <- aov( g24696 ~ Population*Condition, data = NCPK_Shoot)
summary(PCR9_Shoot.aov)
TukeyHSD(PCR9_Shoot.aov)  
PCR9_Shoot_res <- HSD.test(PCR9_Shoot.aov, c("Population", "Condition"), group=TRUE)
PCR9_Shoot_res$groups
oder.group_PCR9_Shoot <- PCR9_Shoot_res$groups[order(rownames(PCR9_Shoot_res$groups)),] 

PCR9_NC <- summarySE(NCPK, measurevar="g24696", groupvars=c("Tissue","Population","Condition"))
PCR9_NC$Condition <- factor(PCR9_NC$Condition, levels = c("Control", "Cd"))
PCR9 <- ggplot(PCR9_NC, aes(x=Population, y=g24696, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="PCR9", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g24696, ymax=g24696+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g24696+sd+80, 
                  label=c(as.character(oder.group_PCR9_Root$groups),as.character(oder.group_PCR9_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

PCR9

bZIP19_Root.aov <- aov( g00424 ~ Population*Condition, data = NCPK_Root)
summary(bZIP19_Root.aov)
TukeyHSD(bZIP19_Root.aov)
bZIP19_Root_res <- HSD.test(bZIP19_Root.aov, c("Population", "Condition"), group=TRUE)
bZIP19_Root_res$groups
oder.group_bZIP19_Root <- bZIP19_Root_res$groups[order(rownames(bZIP19_Root_res$groups)),] 

bZIP19_Shoot.aov <- aov( g00424 ~ Population*Condition, data = NCPK_Shoot)
summary(bZIP19_Shoot.aov)
TukeyHSD(bZIP19_Shoot.aov)  
bZIP19_Shoot_res <- HSD.test(bZIP19_Shoot.aov, c("Population", "Condition"), group=TRUE)
bZIP19_Shoot_res$groups
oder.group_bZIP19_Shoot <- bZIP19_Shoot_res$groups[order(rownames(bZIP19_Shoot_res$groups)),] 

bZIP19_NC <- summarySE(NCPK, measurevar="g00424", groupvars=c("Tissue","Population","Condition"))
bZIP19_NC$Condition <- factor(bZIP19_NC$Condition, levels = c("Control", "Cd"))
bZIP19 <- ggplot(bZIP19_NC, aes(x=Population, y=g00424, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="bZIP19", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g00424, ymax=g00424+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g00424+sd+30, 
                  label=c(as.character(oder.group_bZIP19_Root$groups),as.character(oder.group_bZIP19_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

bZIP19


PCR6_Root.aov <- aov( g24315 ~ Population*Condition, data = NCPK_Root)
summary(PCR6_Root.aov)
TukeyHSD(PCR6_Root.aov)
PCR6_Root_res <- HSD.test(PCR6_Root.aov, c("Population", "Condition"), group=TRUE)
PCR6_Root_res$groups
oder.group_PCR6_Root <- PCR6_Root_res$groups[order(rownames(PCR6_Root_res$groups)),] 

PCR6_Shoot.aov <- aov( g24315 ~ Population*Condition, data = NCPK_Shoot)
summary(PCR6_Shoot.aov)
TukeyHSD(PCR6_Shoot.aov)  
PCR6_Shoot_res <- HSD.test(PCR6_Shoot.aov, c("Population", "Condition"), group=TRUE)
PCR6_Shoot_res$groups
oder.group_PCR6_Shoot <- PCR6_Shoot_res$groups[order(rownames(PCR6_Shoot_res$groups)),] 

PCR6_NC <- summarySE(NCPK, measurevar="g24315", groupvars=c("Tissue","Population","Condition"))
PCR6_NC$Condition <- factor(PCR6_NC$Condition, levels = c("Control", "Cd"))
PCR6 <- ggplot(PCR6_NC, aes(x=Population, y=g24315, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="PCR6", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g24315, ymax=g24315+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g24315+sd+300, 
                  label=c(as.character(oder.group_PCR6_Root$groups),as.character(oder.group_PCR6_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

PCR6


CHX3_Root.aov <- aov( g27281 ~ Population*Condition, data = NCPK_Root)
summary(CHX3_Root.aov)
TukeyHSD(CHX3_Root.aov)
CHX3_Root_res <- HSD.test(CHX3_Root.aov, c("Population", "Condition"), group=TRUE)
CHX3_Root_res$groups
oder.group_CHX3_Root <- CHX3_Root_res$groups[order(rownames(CHX3_Root_res$groups)),] 

CHX3_Shoot.aov <- aov( g27281 ~ Population*Condition, data = NCPK_Shoot)
summary(CHX3_Shoot.aov)
TukeyHSD(CHX3_Shoot.aov)  
CHX3_Shoot_res <- HSD.test(CHX3_Shoot.aov, c("Population", "Condition"), group=TRUE)
CHX3_Shoot_res$groups
oder.group_CHX3_Shoot <- CHX3_Shoot_res$groups[order(rownames(CHX3_Shoot_res$groups)),] 

CHX3_NC <- summarySE(NCPK, measurevar="g27281", groupvars=c("Tissue","Population","Condition"))
CHX3_NC$Condition <- factor(CHX3_NC$Condition, levels = c("Control", "Cd"))
CHX3 <- ggplot(CHX3_NC, aes(x=Population, y=g27281, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="CHX3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g27281, ymax=g27281+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g27281+sd+2, 
                  label=c(as.character(oder.group_CHX3_Root$groups),as.character(oder.group_CHX3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

CHX3

CAX10_Root.aov <- aov( g32370 ~ Population*Condition, data = NCPK_Root)
summary(CAX10_Root.aov)
TukeyHSD(CAX10_Root.aov)
CAX10_Root_res <- HSD.test(CAX10_Root.aov, c("Population", "Condition"), group=TRUE)
CAX10_Root_res$groups
oder.group_CAX10_Root <- CAX10_Root_res$groups[order(rownames(CAX10_Root_res$groups)),] 

CAX10_Shoot.aov <- aov( g32370 ~ Population*Condition, data = NCPK_Shoot)
summary(CAX10_Shoot.aov)
TukeyHSD(CAX10_Shoot.aov)  
CAX10_Shoot_res <- HSD.test(CAX10_Shoot.aov, c("Population", "Condition"), group=TRUE)
CAX10_Shoot_res$groups
oder.group_CAX10_Shoot <- CAX10_Shoot_res$groups[order(rownames(CAX10_Shoot_res$groups)),] 

CAX10_NC <- summarySE(NCPK, measurevar="g32370", groupvars=c("Tissue","Population","Condition"))
CAX10_NC$Condition <- factor(CAX10_NC$Condition, levels = c("Control", "Cd"))
CAX10 <- ggplot(CAX10_NC, aes(x=Population, y=g32370, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="CAX10", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g32370, ymax=g32370+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g32370+sd+7, 
                  label=c(as.character(oder.group_CAX10_Root$groups),as.character(oder.group_CAX10_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

CAX10

ZIP10_Root.aov <- aov( g27549 ~ Population*Condition, data = NCPK_Root)
summary(ZIP10_Root.aov)
TukeyHSD(ZIP10_Root.aov)
ZIP10_Root_res <- HSD.test(ZIP10_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP10_Root_res$groups
oder.group_ZIP10_Root <- ZIP10_Root_res$groups[order(rownames(ZIP10_Root_res$groups)),] 

ZIP10_Shoot.aov <- aov( g27549 ~ Population*Condition, data = NCPK_Shoot)
summary(ZIP10_Shoot.aov)
TukeyHSD(ZIP10_Shoot.aov)  
ZIP10_Shoot_res <- HSD.test(ZIP10_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIP10_Shoot_res$groups
oder.group_ZIP10_Shoot <- ZIP10_Shoot_res$groups[order(rownames(ZIP10_Shoot_res$groups)),] 

ZIP10_NC <- summarySE(NCPK, measurevar="g27549", groupvars=c("Tissue","Population","Condition"))
ZIP10_NC$Condition <- factor(ZIP10_NC$Condition, levels = c("Control", "Cd"))
ZIP10 <- ggplot(ZIP10_NC, aes(x=Population, y=g27549, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZIP10", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g27549, ymax=g27549+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g27549+sd+20, 
                  label=c(as.character(oder.group_ZIP10_Root$groups),as.character(oder.group_ZIP10_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZIP10

NAS2_Root.aov <- aov( g23475 ~ Population*Condition, data = NCPK_Root)
summary(NAS2_Root.aov)
TukeyHSD(NAS2_Root.aov)
NAS2_Root_res <- HSD.test(NAS2_Root.aov, c("Population", "Condition"), group=TRUE)
NAS2_Root_res$groups
oder.group_NAS2_Root <- NAS2_Root_res$groups[order(rownames(NAS2_Root_res$groups)),] 

NAS2_Shoot.aov <- aov( g23475 ~ Population*Condition, data = NCPK_Shoot)
summary(NAS2_Shoot.aov)
TukeyHSD(NAS2_Shoot.aov)  
NAS2_Shoot_res <- HSD.test(NAS2_Shoot.aov, c("Population", "Condition"), group=TRUE)
NAS2_Shoot_res$groups
oder.group_NAS2_Shoot <- NAS2_Shoot_res$groups[order(rownames(NAS2_Shoot_res$groups)),] 

NAS2_NC <- summarySE(NCPK, measurevar="g23475", groupvars=c("Tissue","Population","Condition"))
NAS2_NC$Condition <- factor(NAS2_NC$Condition, levels = c("Control", "Cd"))
NAS2 <- ggplot(NAS2_NC, aes(x=Population, y=g23475, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="NAS2", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g23475, ymax=g23475+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g23475+sd+300, 
                  label=c(as.character(oder.group_NAS2_Root$groups),as.character(oder.group_NAS2_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

NAS2

FER3_Root.aov <- aov( g11645 ~ Population*Condition, data = NCPK_Root)
summary(FER3_Root.aov)
TukeyHSD(FER3_Root.aov)
FER3_Root_res <- HSD.test(FER3_Root.aov, c("Population", "Condition"), group=TRUE)
FER3_Root_res$groups
oder.group_FER3_Root <- FER3_Root_res$groups[order(rownames(FER3_Root_res$groups)),] 

FER3_Shoot.aov <- aov( g11645 ~ Population*Condition, data = NCPK_Shoot)
summary(FER3_Shoot.aov)
TukeyHSD(FER3_Shoot.aov)  
FER3_Shoot_res <- HSD.test(FER3_Shoot.aov, c("Population", "Condition"), group=TRUE)
FER3_Shoot_res$groups
oder.group_FER3_Shoot <- FER3_Shoot_res$groups[order(rownames(FER3_Shoot_res$groups)),] 

FER3_NC <- summarySE(NCPK, measurevar="g11645", groupvars=c("Tissue","Population","Condition"))
FER3_NC$Condition <- factor(FER3_NC$Condition, levels = c("Control", "Cd"))
FER3 <- ggplot(FER3_NC, aes(x=Population, y=g11645, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="FER3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g11645, ymax=g11645+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g11645+sd+70, 
                  label=c(as.character(oder.group_FER3_Root$groups),as.character(oder.group_FER3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

FER3

ZIP9_Root.aov <- aov( g32214 ~ Population*Condition, data = NCPK_Root)
summary(ZIP9_Root.aov)
TukeyHSD(ZIP9_Root.aov)
ZIP9_Root_res <- HSD.test(ZIP9_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP9_Root_res$groups
oder.group_ZIP9_Root <- ZIP9_Root_res$groups[order(rownames(ZIP9_Root_res$groups)),] 

ZIP9_Shoot.aov <- aov( g32214 ~ Population*Condition, data = NCPK_Shoot)
summary(ZIP9_Shoot.aov)
TukeyHSD(ZIP9_Shoot.aov)  
ZIP9_Shoot_res <- HSD.test(ZIP9_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIP9_Shoot_res$groups
oder.group_ZIP9_Shoot <- ZIP9_Shoot_res$groups[order(rownames(ZIP9_Shoot_res$groups)),] 

ZIP9_NC <- summarySE(NCPK, measurevar="g32214", groupvars=c("Tissue","Population","Condition"))
ZIP9_NC$Condition <- factor(ZIP9_NC$Condition, levels = c("Control", "Cd"))
ZIP9 <- ggplot(ZIP9_NC, aes(x=Population, y=g32214, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZIP9", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g32214, ymax=g32214+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g32214+sd+250, 
                  label=c(as.character(oder.group_ZIP9_Root$groups),as.character(oder.group_ZIP9_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZIP9


NRAMP3_Root.aov <- aov( g21000 ~ Population*Condition, data = NCPK_Root)
summary(NRAMP3_Root.aov)
TukeyHSD(NRAMP3_Root.aov)
NRAMP3_Root_res <- HSD.test(NRAMP3_Root.aov, c("Population", "Condition"), group=TRUE)
NRAMP3_Root_res$groups
oder.group_NRAMP3_Root <- NRAMP3_Root_res$groups[order(rownames(NRAMP3_Root_res$groups)),] 

NRAMP3_Shoot.aov <- aov( g21000 ~ Population*Condition, data = NCPK_Shoot)
summary(NRAMP3_Shoot.aov)
TukeyHSD(NRAMP3_Shoot.aov)  
NRAMP3_Shoot_res <- HSD.test(NRAMP3_Shoot.aov, c("Population", "Condition"), group=TRUE)
NRAMP3_Shoot_res$groups
oder.group_NRAMP3_Shoot <- NRAMP3_Shoot_res$groups[order(rownames(NRAMP3_Shoot_res$groups)),] 

NRAMP3_NC <- summarySE(NCPK, measurevar="g21000", groupvars=c("Tissue","Population","Condition"))
NRAMP3_NC$Condition <- factor(NRAMP3_NC$Condition, levels = c("Control", "Cd"))
NRAMP3 <- ggplot(NRAMP3_NC, aes(x=Population, y=g21000, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="NRAMP3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g21000, ymax=g21000+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g21000+sd+25, 
                  label=c(as.character(oder.group_NRAMP3_Root$groups),as.character(oder.group_NRAMP3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

NRAMP3



GSTU7_Root.aov <- aov( g06968 ~ Population*Condition, data = NCPK_Root)
summary(GSTU7_Root.aov)
TukeyHSD(GSTU7_Root.aov)
GSTU7_Root_res <- HSD.test(GSTU7_Root.aov, c("Population", "Condition"), group=TRUE)
GSTU7_Root_res$groups
oder.group_GSTU7_Root <- GSTU7_Root_res$groups[order(rownames(GSTU7_Root_res$groups)),] 

GSTU7_Shoot.aov <- aov( g06968 ~ Population*Condition, data = NCPK_Shoot)
summary(GSTU7_Shoot.aov)
TukeyHSD(GSTU7_Shoot.aov)  
GSTU7_Shoot_res <- HSD.test(GSTU7_Shoot.aov, c("Population", "Condition"), group=TRUE)
GSTU7_Shoot_res$groups
oder.group_GSTU7_Shoot <- GSTU7_Shoot_res$groups[order(rownames(GSTU7_Shoot_res$groups)),] 

GSTU7_NC <- summarySE(NCPK, measurevar="g06968", groupvars=c("Tissue","Population","Condition"))
GSTU7_NC$Condition <- factor(GSTU7_NC$Condition, levels = c("Control", "Cd"))
GSTU7 <- ggplot(GSTU7_NC, aes(x=Population, y=g06968, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="GSTU7", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g06968, ymax=g06968+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g06968+sd+150, 
                  label=c(as.character(oder.group_GSTU7_Root$groups),as.character(oder.group_GSTU7_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

GSTU7

ZIP3_Root.aov <- aov( g07961 ~ Population*Condition, data = NCPK_Root)
summary(ZIP3_Root.aov)
TukeyHSD(ZIP3_Root.aov)
ZIP3_Root_res <- HSD.test(ZIP3_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP3_Root_res$groups
oder.group_ZIP3_Root <- ZIP3_Root_res$groups[order(rownames(ZIP3_Root_res$groups)),] 

ZIP3_Shoot.aov <- aov( g07961 ~ Population*Condition, data = NCPK_Shoot)
summary(ZIP3_Shoot.aov)
TukeyHSD(ZIP3_Shoot.aov)  
ZIP3_Shoot_res <- HSD.test(ZIP3_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIP3_Shoot_res$groups
oder.group_ZIP3_Shoot <- ZIP3_Shoot_res$groups[order(rownames(ZIP3_Shoot_res$groups)),] 

ZIP3_NC <- summarySE(NCPK, measurevar="g07961", groupvars=c("Tissue","Population","Condition"))
ZIP3_NC$Condition <- factor(ZIP3_NC$Condition, levels = c("Control", "Cd"))
ZIP3 <- ggplot(ZIP3_NC, aes(x=Population, y=g07961, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZIP3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g07961, ymax=g07961+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g07961+sd+270, 
                  label=c(as.character(oder.group_ZIP3_Root$groups),as.character(oder.group_ZIP3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZIP3


ZIP8_Root.aov <- aov( g29372 ~ Population*Condition, data = NCPK_Root)
summary(ZIP8_Root.aov)
TukeyHSD(ZIP8_Root.aov)
ZIP8_Root_res <- HSD.test(ZIP8_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP8_Root_res$groups
oder.group_ZIP8_Root <- ZIP8_Root_res$groups[order(rownames(ZIP8_Root_res$groups)),] 

ZIP8_Shoot.aov <- aov( g29372 ~ Population*Condition, data = NCPK_Shoot)
summary(ZIP8_Shoot.aov)
TukeyHSD(ZIP8_Shoot.aov)  
ZIP8_Shoot_res <- HSD.test(ZIP8_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIP8_Shoot_res$groups
oder.group_ZIP8_Shoot <- ZIP8_Shoot_res$groups[order(rownames(ZIP8_Shoot_res$groups)),] 

ZIP8_NC <- summarySE(NCPK, measurevar="g29372", groupvars=c("Tissue","Population","Condition"))
ZIP8_NC$Condition <- factor(ZIP8_NC$Condition, levels = c("Control", "Cd"))
ZIP8 <- ggplot(ZIP8_NC, aes(x=Population, y=g29372, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZIP8", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g29372, ymax=g29372+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g29372+sd+7, 
                  label=c(as.character(oder.group_ZIP8_Root$groups),as.character(oder.group_ZIP8_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZIP8

ZIFl1_Root.aov <- aov( g19324 ~ Population*Condition, data = NCPK_Root)
summary(ZIFl1_Root.aov)
TukeyHSD(ZIFl1_Root.aov)
ZIFl1_Root_res <- HSD.test(ZIFl1_Root.aov, c("Population", "Condition"), group=TRUE)
ZIFl1_Root_res$groups
oder.group_ZIFl1_Root <- ZIFl1_Root_res$groups[order(rownames(ZIFl1_Root_res$groups)),] 

ZIFl1_Shoot.aov <- aov( g19324 ~ Population*Condition, data = NCPK_Shoot)
summary(ZIFl1_Shoot.aov)
TukeyHSD(ZIFl1_Shoot.aov)  
ZIFl1_Shoot_res <- HSD.test(ZIFl1_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIFl1_Shoot_res$groups
oder.group_ZIFl1_Shoot <- ZIFl1_Shoot_res$groups[order(rownames(ZIFl1_Shoot_res$groups)),] 

ZIFl1_NC <- summarySE(NCPK, measurevar="g19324", groupvars=c("Tissue","Population","Condition"))
ZIFl1_NC$Condition <- factor(ZIFl1_NC$Condition, levels = c("Control", "Cd"))
ZIFl1 <- ggplot(ZIFl1_NC, aes(x=Population, y=g19324, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZIFl1", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g19324, ymax=g19324+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g19324+sd+45, 
                  label=c(as.character(oder.group_ZIFl1_Root$groups),as.character(oder.group_ZIFl1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZIFl1



bHLH39_Root.aov <- aov( g21154 ~ Population*Condition, data = NCPK_Root)
summary(bHLH39_Root.aov)
TukeyHSD(bHLH39_Root.aov)
bHLH39_Root_res <- HSD.test(bHLH39_Root.aov, c("Population", "Condition"), group=TRUE)
bHLH39_Root_res$groups
oder.group_bHLH39_Root <- bHLH39_Root_res$groups[order(rownames(bHLH39_Root_res$groups)),] 

bHLH39_Shoot.aov <- aov( g21154 ~ Population*Condition, data = NCPK_Shoot)
summary(bHLH39_Shoot.aov)
TukeyHSD(bHLH39_Shoot.aov)  
bHLH39_Shoot_res <- HSD.test(bHLH39_Shoot.aov, c("Population", "Condition"), group=TRUE)
bHLH39_Shoot_res$groups
oder.group_bHLH39_Shoot <- bHLH39_Shoot_res$groups[order(rownames(bHLH39_Shoot_res$groups)),] 

bHLH39_NC <- summarySE(NCPK, measurevar="g21154", groupvars=c("Tissue","Population","Condition"))
bHLH39_NC$Condition <- factor(bHLH39_NC$Condition, levels = c("Control", "Cd"))
bHLH39 <- ggplot(bHLH39_NC, aes(x=Population, y=g21154, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="bHLH39", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g21154, ymax=g21154+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g21154+sd+80, 
                  label=c(as.character(oder.group_bHLH39_Root$groups),as.character(oder.group_bHLH39_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

bHLH39

bHLH101_Root.aov <- aov( g15774 ~ Population*Condition, data = NCPK_Root)
summary(bHLH101_Root.aov)
TukeyHSD(bHLH101_Root.aov)
bHLH101_Root_res <- HSD.test(bHLH101_Root.aov, c("Population", "Condition"), group=TRUE)
bHLH101_Root_res$groups
oder.group_bHLH101_Root <- bHLH101_Root_res$groups[order(rownames(bHLH101_Root_res$groups)),] 

bHLH101_Shoot.aov <- aov( g15774 ~ Population*Condition, data = NCPK_Shoot)
summary(bHLH101_Shoot.aov)
TukeyHSD(bHLH101_Shoot.aov)  
bHLH101_Shoot_res <- HSD.test(bHLH101_Shoot.aov, c("Population", "Condition"), group=TRUE)
bHLH101_Shoot_res$groups
oder.group_bHLH101_Shoot <- bHLH101_Shoot_res$groups[order(rownames(bHLH101_Shoot_res$groups)),] 

bHLH101_NC <- summarySE(NCPK, measurevar="g15774", groupvars=c("Tissue","Population","Condition"))
bHLH101_NC$Condition <- factor(bHLH101_NC$Condition, levels = c("Control", "Cd"))
bHLH101 <- ggplot(bHLH101_NC, aes(x=Population, y=g15774, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="bHLH101", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g15774, ymax=g15774+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g15774+sd+58, 
                  label=c(as.character(oder.group_bHLH101_Root$groups),as.character(oder.group_bHLH101_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

bHLH101

bHLH100_Root.aov <- aov( g24495 ~ Population*Condition, data = NCPK_Root)
summary(bHLH100_Root.aov)
TukeyHSD(bHLH100_Root.aov)
bHLH100_Root_res <- HSD.test(bHLH100_Root.aov, c("Population", "Condition"), group=TRUE)
bHLH100_Root_res$groups
oder.group_bHLH100_Root <- bHLH100_Root_res$groups[order(rownames(bHLH100_Root_res$groups)),] 

bHLH100_Shoot.aov <- aov( g24495 ~ Population*Condition, data = NCPK_Shoot)
summary(bHLH100_Shoot.aov)
TukeyHSD(bHLH100_Shoot.aov)  
bHLH100_Shoot_res <- HSD.test(bHLH100_Shoot.aov, c("Population", "Condition"), group=TRUE)
bHLH100_Shoot_res$groups
oder.group_bHLH100_Shoot <- bHLH100_Shoot_res$groups[order(rownames(bHLH100_Shoot_res$groups)),] 

bHLH100_NC <- summarySE(NCPK, measurevar="g24495", groupvars=c("Tissue","Population","Condition"))
bHLH100_NC$Condition <- factor(bHLH100_NC$Condition, levels = c("Control", "Cd"))
bHLH100 <- ggplot(bHLH100_NC, aes(x=Population, y=g24495, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="bHLH100", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g24495, ymax=g24495+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g24495+sd+40, 
                  label=c(as.character(oder.group_bHLH100_Root$groups),as.character(oder.group_bHLH100_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

bHLH100

bHLH38_Root.aov <- aov( g21155 ~ Population*Condition, data = NCPK_Root)
summary(bHLH38_Root.aov)
TukeyHSD(bHLH38_Root.aov)
bHLH38_Root_res <- HSD.test(bHLH38_Root.aov, c("Population", "Condition"), group=TRUE)
bHLH38_Root_res$groups
oder.group_bHLH38_Root <- bHLH38_Root_res$groups[order(rownames(bHLH38_Root_res$groups)),] 

bHLH38_Shoot.aov <- aov( g21155 ~ Population*Condition, data = NCPK_Shoot)
summary(bHLH38_Shoot.aov)
TukeyHSD(bHLH38_Shoot.aov)  
bHLH38_Shoot_res <- HSD.test(bHLH38_Shoot.aov, c("Population", "Condition"), group=TRUE)
bHLH38_Shoot_res$groups
oder.group_bHLH38_Shoot <- bHLH38_Shoot_res$groups[order(rownames(bHLH38_Shoot_res$groups)),] 

bHLH38_NC <- summarySE(NCPK, measurevar="g21155", groupvars=c("Tissue","Population","Condition"))
bHLH38_NC$Condition <- factor(bHLH38_NC$Condition, levels = c("Control", "Cd"))
bHLH38 <- ggplot(bHLH38_NC, aes(x=Population, y=g21155, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="bHLH38", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g21155, ymax=g21155+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g21155+sd+50, 
                  label=c(as.character(oder.group_bHLH38_Root$groups),as.character(oder.group_bHLH38_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

bHLH38

ORG1_Root.aov <- aov( g23050 ~ Population*Condition, data = NCPK_Root)
summary(ORG1_Root.aov)
TukeyHSD(ORG1_Root.aov)
ORG1_Root_res <- HSD.test(ORG1_Root.aov, c("Population", "Condition"), group=TRUE)
ORG1_Root_res$groups
oder.group_ORG1_Root <- ORG1_Root_res$groups[order(rownames(ORG1_Root_res$groups)),] 

ORG1_Shoot.aov <- aov( g23050 ~ Population*Condition, data = NCPK_Shoot)
summary(ORG1_Shoot.aov)
TukeyHSD(ORG1_Shoot.aov)  
ORG1_Shoot_res <- HSD.test(ORG1_Shoot.aov, c("Population", "Condition"), group=TRUE)
ORG1_Shoot_res$groups
oder.group_ORG1_Shoot <- ORG1_Shoot_res$groups[order(rownames(ORG1_Shoot_res$groups)),] 

ORG1_NC <- summarySE(NCPK, measurevar="g23050", groupvars=c("Tissue","Population","Condition"))
ORG1_NC$Condition <- factor(ORG1_NC$Condition, levels = c("Control", "Cd"))
ORG1 <- ggplot(ORG1_NC, aes(x=Population, y=g23050, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ORG1", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g23050, ymax=g23050+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g23050+sd+450, 
                  label=c(as.character(oder.group_ORG1_Root$groups),as.character(oder.group_ORG1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ORG1

CGLD27_Root.aov <- aov( g07324 ~ Population*Condition, data = NCPK_Root)
summary(CGLD27_Root.aov)
TukeyHSD(CGLD27_Root.aov)
CGLD27_Root_res <- HSD.test(CGLD27_Root.aov, c("Population", "Condition"), group=TRUE)
CGLD27_Root_res$groups
oder.group_CGLD27_Root <- CGLD27_Root_res$groups[order(rownames(CGLD27_Root_res$groups)),] 

CGLD27_Shoot.aov <- aov( g07324 ~ Population*Condition, data = NCPK_Shoot)
summary(CGLD27_Shoot.aov)
TukeyHSD(CGLD27_Shoot.aov)  
CGLD27_Shoot_res <- HSD.test(CGLD27_Shoot.aov, c("Population", "Condition"), group=TRUE)
CGLD27_Shoot_res$groups
oder.group_CGLD27_Shoot <- CGLD27_Shoot_res$groups[order(rownames(CGLD27_Shoot_res$groups)),] 

CGLD27_NC <- summarySE(NCPK, measurevar="g07324", groupvars=c("Tissue","Population","Condition"))
CGLD27_NC$Condition <- factor(CGLD27_NC$Condition, levels = c("Control", "Cd"))
CGLD27 <- ggplot(CGLD27_NC, aes(x=Population, y=g07324, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="CGLD27", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g07324, ymax=g07324+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g07324+sd+400, 
                  label=c(as.character(oder.group_CGLD27_Root$groups),as.character(oder.group_CGLD27_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

CGLD27


BTS_Root.aov <- aov( g21576 ~ Population*Condition, data = NCPK_Root)
summary(BTS_Root.aov)
TukeyHSD(BTS_Root.aov)
BTS_Root_res <- HSD.test(BTS_Root.aov, c("Population", "Condition"), group=TRUE)
BTS_Root_res$groups
oder.group_BTS_Root <- BTS_Root_res$groups[order(rownames(BTS_Root_res$groups)),] 

BTS_Shoot.aov <- aov( g21576 ~ Population*Condition, data = NCPK_Shoot)
summary(BTS_Shoot.aov)
TukeyHSD(BTS_Shoot.aov)  
BTS_Shoot_res <- HSD.test(BTS_Shoot.aov, c("Population", "Condition"), group=TRUE)
BTS_Shoot_res$groups
oder.group_BTS_Shoot <- BTS_Shoot_res$groups[order(rownames(BTS_Shoot_res$groups)),] 

BTS_NC <- summarySE(NCPK, measurevar="g21576", groupvars=c("Tissue","Population","Condition"))
BTS_NC$Condition <- factor(BTS_NC$Condition, levels = c("Control", "Cd"))
BTS <- ggplot(BTS_NC, aes(x=Population, y=g21576, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="BTS", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g21576, ymax=g21576+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g21576+sd+850, 
                  label=c(as.character(oder.group_BTS_Root$groups),as.character(oder.group_BTS_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

BTS

CUTA_Root.aov <- aov( g03274 ~ Population*Condition, data = NCPK_Root)
summary(CUTA_Root.aov)
TukeyHSD(CUTA_Root.aov)
CUTA_Root_res <- HSD.test(CUTA_Root.aov, c("Population", "Condition"), group=TRUE)
CUTA_Root_res$groups
oder.group_CUTA_Root <- CUTA_Root_res$groups[order(rownames(CUTA_Root_res$groups)),] 

CUTA_Shoot.aov <- aov( g03274 ~ Population*Condition, data = NCPK_Shoot)
summary(CUTA_Shoot.aov)
TukeyHSD(CUTA_Shoot.aov)  
CUTA_Shoot_res <- HSD.test(CUTA_Shoot.aov, c("Population", "Condition"), group=TRUE)
CUTA_Shoot_res$groups
oder.group_CUTA_Shoot <- CUTA_Shoot_res$groups[order(rownames(CUTA_Shoot_res$groups)),] 

CUTA_NC <- summarySE(NCPK, measurevar="g03274", groupvars=c("Tissue","Population","Condition"))
CUTA_NC$Condition <- factor(CUTA_NC$Condition, levels = c("Control", "Cd"))
CUTA <- ggplot(CUTA_NC, aes(x=Population, y=g03274, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="CUTA", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g03274, ymax=g03274+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g03274+sd+18, 
                  label=c(as.character(oder.group_CUTA_Root$groups),as.character(oder.group_CUTA_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

CUTA


HMA4_Root.aov <- aov( g22694 ~ Population*Condition, data = NCPK_Root)
summary(HMA4_Root.aov)
TukeyHSD(HMA4_Root.aov)
HMA4_Root_res <- HSD.test(HMA4_Root.aov, c("Population", "Condition"), group=TRUE)
HMA4_Root_res$groups
oder.group_HMA4_Root <- HMA4_Root_res$groups[order(rownames(HMA4_Root_res$groups)),] 

HMA4_Shoot.aov <- aov( g22694 ~ Population*Condition, data = NCPK_Shoot)
summary(HMA4_Shoot.aov)
TukeyHSD(HMA4_Shoot.aov)  
HMA4_Shoot_res <- HSD.test(HMA4_Shoot.aov, c("Population", "Condition"), group=TRUE)
HMA4_Shoot_res$groups
oder.group_HMA4_Shoot <- HMA4_Shoot_res$groups[order(rownames(HMA4_Shoot_res$groups)),] 

HMA4_NC <- summarySE(NCPK, measurevar="g22694", groupvars=c("Tissue","Population","Condition"))
HMA4_NC$Condition <- factor(HMA4_NC$Condition, levels = c("Control", "Cd"))
HMA4 <- ggplot(HMA4_NC, aes(x=Population, y=g22694, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="HMA4", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g22694, ymax=g22694+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g22694+sd+250, 
                  label=c(as.character(oder.group_HMA4_Root$groups),as.character(oder.group_HMA4_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())
HMA4

HMA3_Root.aov <- aov( g22694 ~ Population*Condition, data = NCPK_Root)
summary(HMA3_Root.aov)
TukeyHSD(HMA3_Root.aov)
HMA3_Root_res <- HSD.test(HMA3_Root.aov, c("Population", "Condition"), group=TRUE)
HMA3_Root_res$groups
oder.group_HMA3_Root <- HMA3_Root_res$groups[order(rownames(HMA3_Root_res$groups)),] 

HMA3_Shoot.aov <- aov( g22694 ~ Population*Condition, data = NCPK_Shoot)
summary(HMA3_Shoot.aov)
TukeyHSD(HMA3_Shoot.aov)  
HMA3_Shoot_res <- HSD.test(HMA3_Shoot.aov, c("Population", "Condition"), group=TRUE)
HMA3_Shoot_res$groups
oder.group_HMA3_Shoot <- HMA3_Shoot_res$groups[order(rownames(HMA3_Shoot_res$groups)),] 

HMA3_NC <- summarySE(NCPK, measurevar="g04212", groupvars=c("Tissue","Population","Condition"))
HMA3_NC$Condition <- factor(HMA3_NC$Condition, levels = c("Control", "Cd"))
HMA3 <- ggplot(HMA3_NC, aes(x=Population, y=g04212, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="HMA3", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g04212, ymax=g04212+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g04212+sd+50, 
                  label=c(as.character(oder.group_HMA3_Root$groups),as.character(oder.group_HMA3_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())
HMA3

HMA5_Root.aov <- aov( g22918 ~ Population*Condition, data = NCPK_Root)
summary(HMA5_Root.aov)
TukeyHSD(HMA5_Root.aov)
HMA5_Root_res <- HSD.test(HMA5_Root.aov, c("Population", "Condition"), group=TRUE)
HMA5_Root_res$groups
oder.group_HMA5_Root <- HMA5_Root_res$groups[order(rownames(HMA5_Root_res$groups)),] 

HMA5_Shoot.aov <- aov( g22918 ~ Population*Condition, data = NCPK_Shoot)
summary(HMA5_Shoot.aov)
TukeyHSD(HMA5_Shoot.aov)  
HMA5_Shoot_res <- HSD.test(HMA5_Shoot.aov, c("Population", "Condition"), group=TRUE)
HMA5_Shoot_res$groups
oder.group_HMA5_Shoot <- HMA5_Shoot_res$groups[order(rownames(HMA5_Shoot_res$groups)),] 

HMA5_NC <- summarySE(NCPK, measurevar="g22918", groupvars=c("Tissue","Population","Condition"))
HMA5_NC$Condition <- factor(HMA5_NC$Condition, levels = c("Control", "Cd"))
HMA5 <- ggplot(HMA5_NC, aes(x=Population, y=g22918, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="HMA5", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g22918, ymax=g22918+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g22918+sd+80, 
                  label=c(as.character(oder.group_HMA5_Root$groups),as.character(oder.group_HMA5_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())
HMA5

MTP1_Root.aov <- aov( g09643 ~ Population*Condition, data = NCPK_Root)
summary(MTP1_Root.aov)
TukeyHSD(MTP1_Root.aov)
MTP1_Root_res <- HSD.test(MTP1_Root.aov, c("Population", "Condition"), group=TRUE)
MTP1_Root_res$groups
oder.group_MTP1_Root <- MTP1_Root_res$groups[order(rownames(MTP1_Root_res$groups)),] 

MTP1_Shoot.aov <- aov( g09643 ~ Population*Condition, data = NCPK_Shoot)
summary(MTP1_Shoot.aov)
TukeyHSD(MTP1_Shoot.aov)  
MTP1_Shoot_res <- HSD.test(MTP1_Shoot.aov, c("Population", "Condition"), group=TRUE)
MTP1_Shoot_res$groups
oder.group_MTP1_Shoot <- MTP1_Shoot_res$groups[order(rownames(MTP1_Shoot_res$groups)),] 

MTP1_NC <- summarySE(NCPK, measurevar="g09643", groupvars=c("Tissue","Population","Condition"))
MTP1_NC$Condition <- factor(MTP1_NC$Condition, levels = c("Control", "Cd"))
MTP1 <- ggplot(MTP1_NC, aes(x=Population, y=g09643, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="MTP1", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g09643, ymax=g09643+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g09643+sd+380, 
                  label=c(as.character(oder.group_MTP1_Root$groups),as.character(oder.group_MTP1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

MTP1



IRT1_Root.aov <- aov( g15474 ~ Population*Condition, data = NCPK_Root)
summary(IRT1_Root.aov)
TukeyHSD(IRT1_Root.aov)
IRT1_Root_res <- HSD.test(IRT1_Root.aov, c("Population", "Condition"), group=TRUE)
IRT1_Root_res$groups
oder.group_IRT1_Root <- IRT1_Root_res$groups[order(rownames(IRT1_Root_res$groups)),] 

IRT1_Shoot.aov <- aov( g15474 ~ Population*Condition, data = NCPK_Shoot)
summary(IRT1_Shoot.aov)
TukeyHSD(IRT1_Shoot.aov)  
IRT1_Shoot_res <- HSD.test(IRT1_Shoot.aov, c("Population", "Condition"), group=TRUE)
IRT1_Shoot_res$groups
oder.group_IRT1_Shoot <- IRT1_Shoot_res$groups[order(rownames(IRT1_Shoot_res$groups)),] 

IRT1_NC <- summarySE(NCPK, measurevar="g15474", groupvars=c("Tissue","Population","Condition"))
IRT1_NC$Condition <- factor(IRT1_NC$Condition, levels = c("Control", "Cd"))
IRT1 <- ggplot(IRT1_NC, aes(x=Population, y=g15474, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="IRT1", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g15474, ymax=g15474+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g15474+sd+190, 
                  label=c(as.character(oder.group_IRT1_Root$groups),as.character(oder.group_IRT1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

IRT1


VIT1_Root.aov <- aov( g14565 ~ Population*Condition, data = NCPK_Root)
summary(VIT1_Root.aov)
TukeyHSD(VIT1_Root.aov)
VIT1_Root_res <- HSD.test(VIT1_Root.aov, c("Population", "Condition"), group=TRUE)
VIT1_Root_res$groups
oder.group_VIT1_Root <- VIT1_Root_res$groups[order(rownames(VIT1_Root_res$groups)),] 

VIT1_Shoot.aov <- aov( g14565 ~ Population*Condition, data = NCPK_Shoot)
summary(VIT1_Shoot.aov)
TukeyHSD(VIT1_Shoot.aov)  
VIT1_Shoot_res <- HSD.test(VIT1_Shoot.aov, c("Population", "Condition"), group=TRUE)
VIT1_Shoot_res$groups
oder.group_VIT1_Shoot <- VIT1_Shoot_res$groups[order(rownames(VIT1_Shoot_res$groups)),] 

VIT1_NC <- summarySE(NCPK, measurevar="g14565", groupvars=c("Tissue","Population","Condition"))
VIT1_NC$Condition <- factor(VIT1_NC$Condition, levels = c("Control", "Cd"))
VIT1 <- ggplot(VIT1_NC, aes(x=Population, y=g14565, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="VIT1", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g14565, ymax=g14565+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g14565+sd+0.3, 
                  label=c(as.character(oder.group_VIT1_Root$groups),as.character(oder.group_VIT1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

VIT1


NRAMP1_Root.aov <- aov( g01559 ~ Population*Condition, data = NCPK_Root)
summary(NRAMP1_Root.aov)
TukeyHSD(NRAMP1_Root.aov)
NRAMP1_Root_res <- HSD.test(NRAMP1_Root.aov, c("Population", "Condition"), group=TRUE)
NRAMP1_Root_res$groups
oder.group_NRAMP1_Root <- NRAMP1_Root_res$groups[order(rownames(NRAMP1_Root_res$groups)),] 

NRAMP1_Shoot.aov <- aov( g01559 ~ Population*Condition, data = NCPK_Shoot)
summary(NRAMP1_Shoot.aov)
TukeyHSD(NRAMP1_Shoot.aov)  
NRAMP1_Shoot_res <- HSD.test(NRAMP1_Shoot.aov, c("Population", "Condition"), group=TRUE)
NRAMP1_Shoot_res$groups
oder.group_NRAMP1_Shoot <- NRAMP1_Shoot_res$groups[order(rownames(NRAMP1_Shoot_res$groups)),] 

NRAMP1_NC <- summarySE(NCPK, measurevar="g01559", groupvars=c("Tissue","Population","Condition"))
NRAMP1_NC$Condition <- factor(NRAMP1_NC$Condition, levels = c("Control", "Cd"))
NRAMP1 <- ggplot(NRAMP1_NC, aes(x=Population, y=g01559, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="NRAMP1", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g01559, ymax=g01559+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g01559+sd+150, 
                  label=c(as.character(oder.group_NRAMP1_Root$groups),as.character(oder.group_NRAMP1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

NRAMP1

CHX8_Root.aov <- aov( g07136 ~ Population*Condition, data = NCPK_Root)
summary(CHX8_Root.aov)
TukeyHSD(CHX8_Root.aov)
CHX8_Root_res <- HSD.test(CHX8_Root.aov, c("Population", "Condition"), group=TRUE)
CHX8_Root_res$groups
oder.group_CHX8_Root <- CHX8_Root_res$groups[order(rownames(CHX8_Root_res$groups)),] 

CHX8_Shoot.aov <- aov( g07136 ~ Population*Condition, data = NCPK_Shoot)
summary(CHX8_Shoot.aov)
TukeyHSD(CHX8_Shoot.aov)  
CHX8_Shoot_res <- HSD.test(CHX8_Shoot.aov, c("Population", "Condition"), group=TRUE)
CHX8_Shoot_res$groups
oder.group_CHX8_Shoot <- CHX8_Shoot_res$groups[order(rownames(CHX8_Shoot_res$groups)),] 

CHX8_NC <- summarySE(NCPK, measurevar="g07136", groupvars=c("Tissue","Population","Condition"))
CHX8_NC$Condition <- factor(CHX8_NC$Condition, levels = c("Control", "Cd"))
CHX8 <- ggplot(CHX8_NC, aes(x=Population, y=g07136, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="CHX8", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g07136, ymax=g07136+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g07136+sd+1.5, 
                  label=c(as.character(oder.group_CHX8_Root$groups),as.character(oder.group_CHX8_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

CHX8

CMT1_Root.aov <- aov( g21322 ~ Population*Condition, data = NCPK_Root)
summary(CMT1_Root.aov)
TukeyHSD(CMT1_Root.aov)
CMT1_Root_res <- HSD.test(CMT1_Root.aov, c("Population", "Condition"), group=TRUE)
CMT1_Root_res$groups
oder.group_CMT1_Root <- CMT1_Root_res$groups[order(rownames(CMT1_Root_res$groups)),] 

CMT1_Shoot.aov <- aov( g21322 ~ Population*Condition, data = NCPK_Shoot)
summary(CMT1_Shoot.aov)
TukeyHSD(CMT1_Shoot.aov)  
CMT1_Shoot_res <- HSD.test(CMT1_Shoot.aov, c("Population", "Condition"), group=TRUE)
CMT1_Shoot_res$groups
oder.group_CMT1_Shoot <- CMT1_Shoot_res$groups[order(rownames(CMT1_Shoot_res$groups)),] 

CMT1_NC <- summarySE(NCPK, measurevar="g21322", groupvars=c("Tissue","Population","Condition"))
CMT1_NC$Condition <- factor(CMT1_NC$Condition, levels = c("Control", "Cd"))
CMT1 <- ggplot(CMT1_NC, aes(x=Population, y=g21322, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="CMT1", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g21322, ymax=g21322+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g21322+sd+80, 
                  label=c(as.character(oder.group_CMT1_Root$groups),as.character(oder.group_CMT1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

CMT1

YSL4_Root.aov <- aov( g22406 ~ Population*Condition, data = NCPK_Root)
summary(YSL4_Root.aov)
TukeyHSD(YSL4_Root.aov)
YSL4_Root_res <- HSD.test(YSL4_Root.aov, c("Population", "Condition"), group=TRUE)
YSL4_Root_res$groups
oder.group_YSL4_Root <- YSL4_Root_res$groups[order(rownames(YSL4_Root_res$groups)),] 

YSL4_Shoot.aov <- aov( g22406 ~ Population*Condition, data = NCPK_Shoot)
summary(YSL4_Shoot.aov)
TukeyHSD(YSL4_Shoot.aov)  
YSL4_Shoot_res <- HSD.test(YSL4_Shoot.aov, c("Population", "Condition"), group=TRUE)
YSL4_Shoot_res$groups
oder.group_YSL4_Shoot <- YSL4_Shoot_res$groups[order(rownames(YSL4_Shoot_res$groups)),] 

YSL4_NC <- summarySE(NCPK, measurevar="g22406", groupvars=c("Tissue","Population","Condition"))
YSL4_NC$Condition <- factor(YSL4_NC$Condition, levels = c("Control", "Cd"))
YSL4 <- ggplot(YSL4_NC, aes(x=Population, y=g22406, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="YSL4", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g22406, ymax=g22406+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g22406+sd+20, 
                  label=c(as.character(oder.group_YSL4_Root$groups),as.character(oder.group_YSL4_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

YSL4

CHX2_Root.aov <- aov( g01507 ~ Population*Condition, data = NCPK_Root)
summary(CHX2_Root.aov)
TukeyHSD(CHX2_Root.aov)
CHX2_Root_res <- HSD.test(CHX2_Root.aov, c("Population", "Condition"), group=TRUE)
CHX2_Root_res$groups
oder.group_CHX2_Root <- CHX2_Root_res$groups[order(rownames(CHX2_Root_res$groups)),] 

CHX2_Shoot.aov <- aov( g01507 ~ Population*Condition, data = NCPK_Shoot)
summary(CHX2_Shoot.aov)
TukeyHSD(CHX2_Shoot.aov)  
CHX2_Shoot_res <- HSD.test(CHX2_Shoot.aov, c("Population", "Condition"), group=TRUE)
CHX2_Shoot_res$groups
oder.group_CHX2_Shoot <- CHX2_Shoot_res$groups[order(rownames(CHX2_Shoot_res$groups)),] 

CHX2_NC <- summarySE(NCPK, measurevar="g01507", groupvars=c("Tissue","Population","Condition"))
CHX2_NC$Condition <- factor(CHX2_NC$Condition, levels = c("Control", "Cd"))
CHX2 <- ggplot(CHX2_NC, aes(x=Population, y=g01507, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="CHX2", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g01507, ymax=g01507+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g01507+sd+5, 
                  label=c(as.character(oder.group_CHX2_Root$groups),as.character(oder.group_CHX2_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

CHX2


```


## Only root for Wall
```{r}
NAS2_Root <- ggplot(NAS2_NC[1:6,], aes(x=Population, y=g23475, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="NAS2", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g23475, ymax=g23475+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g23475+sd+340, 
                  label=c(as.character(oder.group_NAS2_Root$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

NAS2_Root


NRAMP3_Root <- ggplot(NRAMP3_NC[1:6,], aes(x=Population, y=g21000, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="NRAMP3", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g21000, ymax=g21000+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g21000+sd+13, 
                  label=c(as.character(oder.group_NRAMP3_Root$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

NRAMP3_Root

ZIP10_Root <- ggplot(ZIP10_NC[1:6,], aes(x=Population, y=g27549, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="ZIP10", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g27549, ymax=g27549+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g27549+sd+12, 
                  label=c(as.character(oder.group_ZIP10_Root$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

ZIP10_Root

CUTA_Root <- ggplot(CUTA_NC[1:6,], aes(x=Population, y=g03274, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="CUTA", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g03274, ymax=g03274+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g03274+sd+7, 
                  label=c(as.character(oder.group_CUTA_Root$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

CUTA_Root

PCR6_Root <- ggplot(PCR6_NC[1:6,], aes(x=Population, y=g24315, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="PCR6", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g24315, ymax=g24315+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g24315+sd+250, 
                  label=c(as.character(oder.group_PCR6_Root$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

PCR6_Root

CHX8_Root <- ggplot(CHX8_NC[1:6,], aes(x=Population, y=g07136, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="CHX8", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g07136, ymax=g07136+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g07136+sd+0.6, 
                  label=c(as.character(oder.group_CHX8_Root$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

CHX8_Root

CMT1_Root <- ggplot(CMT1_NC[1:6,], aes(x=Population, y=g21322, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="CMT1", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g21322, ymax=g21322+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g21322+sd+8, 
                  label=c(as.character(oder.group_CMT1_Root$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

CMT1_Root

YSL4_Root <- ggplot(YSL4_NC[1:6,], aes(x=Population, y=g22406, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="YSL4", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g22406, ymax=g22406+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g22406+sd+5, 
                  label=c(as.character(oder.group_YSL4_Root$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

YSL4_Root

ZIP7_Shoot <- ggplot(ZIP7_NC[7:12,], aes(x=Population, y=g23539, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="ZIP7", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g23539, ymax=g23539+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g23539+sd+2.5, 
                  label=c(as.character(oder.group_ZIP7_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

ZIP7_Shoot

CHX8_Shoot <- ggplot(CHX8_NC[7:12,], aes(x=Population, y=g07136, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="CHX8", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g07136, ymax=g07136+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g07136+sd+0.2, 
                  label=c(as.character(oder.group_CHX8_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

CHX8_Shoot



CMT1_Shoot <- ggplot(CMT1_NC[7:12,], aes(x=Population, y=g21322, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="CMT1", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g21322, ymax=g21322+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g21322+sd+40, 
                  label=c(as.character(oder.group_CMT1_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

CMT1_Shoot

CUTA_Shoot <- ggplot(CUTA_NC[7:12,], aes(x=Population, y=g03274, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="CUTA", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g03274, ymax=g03274+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g03274+sd+7.5, 
                  label=c(as.character(oder.group_CUTA_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

CUTA_Shoot

HIPP20_Shoot <- ggplot(HIPP20_NC[7:12,], aes(x=Population, y=g03852, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="HIPP20", x="") +
    theme(axis.text.y=element_text(size=10, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g03852, ymax=g03852+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g03852+sd+7, 
                  label=c(as.character(oder.group_HIPP20_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +            
    theme(panel.grid.major = element_blank())

HIPP20_Shoot

#Highly expressed MH genes in root of Wall under Cd treatment
M_gene_plot_Wall_R <- ggarrange(CHX8_Root,CUTA_Root,ZIP10_Root, CMT1_Root, PCR6_Root,
                                ZIP7_Shoot,CHX8_Shoot,CUTA_Shoot,CMT1_Shoot, HIPP20_Shoot,  
                                common.legend = TRUE, legend = "bottom", nrow = 2, ncol=5, align = "hv")

annotate_figure(M_gene_plot_Wall_R,
                left = text_grob("Nomalized counts per kilobase of gene (NCPK)", color = "gray10", rot = 90, size = 15))

```

## Only root for Noss
```{r}



AGO9_Root <- ggplot(AGO9_NC[1:6,], aes(x=Population, y=g12002, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=0.4) +
    labs(y="", title="AGO9", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g12002, ymax=g12002+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g12002+sd+1.5, 
                  label=c(as.character(oder.group_AGO9_Root$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

AGO9_Root

ZYP1a_Root <- ggplot(ZYP1a_NC[1:6,], aes(x=Population, y=g13785, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=0.4) +
    labs(y="", title="ZYP1a", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g13785, ymax=g13785+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g13785+sd+0.3, 
                  label=c(as.character(oder.group_ZYP1a_Root$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

ZYP1a_Root


ZYP1b_Root <- ggplot(ZYP1b_NC[1:6,], aes(x=Population, y=g13782, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=0.4) +
    labs(y="", title="ZYP1b", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g13782, ymax=g13782+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g13782+sd+0.3, 
                  label=c(as.character(oder.group_ZYP1b_Root$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

ZYP1b_Root

SMC2_Root <- ggplot(SMC2_NC[1:6,], aes(x=Population, y=g12815, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=0.4) +
    labs(y="", title="SMC2", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g12815, ymax=g12815+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g12815+sd+3, 
                  label=c(as.character(oder.group_SMC2_Root$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

SMC2_Root






ZIP2_Root <- ggplot(ZIP2_NC[1:6,], aes(x=Population, y=g11007, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=0.4) +
    labs(y="", title="ZIP2", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g11007, ymax=g11007+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g11007+sd+7, 
                  label=c(as.character(oder.group_ZIP2_Root$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

ZIP2_Root

HMA2_Root <- ggplot(HMA2_NC[1:6,], aes(x=Population, y=g04213, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="HMA2", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g04213, ymax=g04213+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g04213+sd+65, 
                  label=c(as.character(oder.group_HMA2_Root$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

HMA2_Root

ZIP6_Root <- ggplot(ZIP6_NC[1:6,], aes(x=Population, y=g24253, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="ZIP6", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g24253, ymax=g24253+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g24253+sd+70, 
                  label=c(as.character(oder.group_ZIP6_Root$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

ZIP6_Root

IRT1_Root <- ggplot(IRT1_NC[1:6,], aes(x=Population, y=g15474, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="IRT1", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g15474, ymax=g15474+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g15474+sd+160, 
                  label=c(as.character(oder.group_IRT1_Root$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

IRT1_Root

CAX10_Root <- ggplot(CAX10_NC[1:6,], aes(x=Population, y=g32370, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="CAX10", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g32370, ymax=g32370+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g32370+sd+4, 
                  label=c(as.character(oder.group_CAX10_Root$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

CAX10_Root

PCR9_Root <- ggplot(PCR9_NC[1:6,], aes(x=Population, y=g24696, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="PCR9", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g24696, ymax=g24696+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g24696+sd+70, 
                  label=c(as.character(oder.group_PCR9_Root$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

PCR9_Root

## Only shoot for Noss
PCR8_Shoot <- ggplot(PCR8_NC[7:12,], aes(x=Population, y=g15174, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="PCR8", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g15174, ymax=g15174+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g15174+sd+13, 
                  label=c(as.character(oder.group_PCR8_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

PCR8_Shoot

HMA2_Shoot <- ggplot(HMA2_NC[7:12,], aes(x=Population, y=g04213, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="HMA2", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g04213, ymax=g04213+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g04213+sd+7, 
                  label=c(as.character(oder.group_HMA2_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

HMA2_Shoot

CHX2_Shoot <- ggplot(CHX2_NC[7:12,], aes(x=Population, y=g01507, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="CHX2", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g01507, ymax=g01507+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g01507+sd+1.5, 
                  label=c(as.character(oder.group_CHX2_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

CHX2_Shoot

CAX10_Shoot <- ggplot(CAX10_NC[7:12,], aes(x=Population, y=g32370, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=.4) +
    labs(y="", title="CAX10", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g32370, ymax=g32370+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g32370+sd+3.5, 
                  label=c(as.character(oder.group_CAX10_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

CAX10_Shoot


AGO9_Shoot <- ggplot(AGO9_NC[7:12,], aes(x=Population, y=g12002, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=0.4) +
    labs(y="", title="AGO9", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g12002, ymax=g12002+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g12002+sd+5, 
                  label=c(as.character(oder.group_AGO9_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

AGO9_Shoot

ZYP1a_Shoot <- ggplot(ZYP1a_NC[7:12,], aes(x=Population, y=g13785, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=0.4) +
    labs(y="", title="ZYP1a", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g13785, ymax=g13785+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g13785+sd+1, 
                  label=c(as.character(oder.group_ZYP1a_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

ZYP1a_Shoot


ZYP1b_Shoot <- ggplot(ZYP1b_NC[7:12,], aes(x=Population, y=g13782, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=0.4) +
    labs(y="", title="ZYP1b", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g13782, ymax=g13782+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g13782+sd+1, 
                  label=c(as.character(oder.group_ZYP1b_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

ZYP1b_Shoot


SMC2_Shoot <- ggplot(SMC2_NC[7:12,], aes(x=Population, y=g12815, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color = "grey20", size=0.4) +
    labs(y="", title="SMC2", x="") +
    theme(axis.text.y=element_text(size=11, color="black"),axis.text.x=element_text(size=13, color="black"), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g12815, ymax=g12815+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    geom_text(aes(x=Population, y=g12815+sd+2.5, 
                  label=c(as.character(oder.group_SMC2_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4.5) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd")) +       
    theme(panel.grid.major = element_blank())

SMC2_Shoot



#Merge for plots of genes that are constitutively differentially expressed in Noss
Meio_plot <- ggarrange(AGO9_Root, ZYP1a_Root, SMC2_Root, AGO9_Shoot, ZYP1a_Shoot, SMC2_Shoot,  
                                common.legend = TRUE, legend = "bottom", nrow = 2, ncol=3,
                                labels = c("C", NA, NA,
                                           "D", NA, NA),
                                font.label = list(size = 17), align = "hv")

annotate_figure(Meio_plot,
                left = text_grob("Nomalized counts per kilobase of gene (NCPK)", color = "gray10", rot = 90, size = 15))



M_gene_plot_Noss <- ggarrange(ZIP2_Root, HMA2_Root, ZIP6_Root, IRT1_Root, CAX10_Root, PCR9_Root,
                                PCR8_Shoot, HMA2_Shoot, CHX2_Shoot, CAX10_Shoot, NA, NA,  
                                common.legend = TRUE, legend = "bottom", nrow = 4, ncol=3,
                                labels = c("A", NA, NA,
                                           "B", NA, NA,
                                           "C", NA, NA,
                                           "D", NA, NA),
                                font.label = list(size = 17), align = "hv")

annotate_figure(M_gene_plot_Noss,
                left = text_grob("Nomalized counts per kilobase of gene (NCPK)", color = "gray10", rot = 90, size = 15))


```

#Volcano plots
```{r}

## Reformat the DESeq2 results

gene_df <- Predata2[,c(1,7,8)]
gene_df$Ath_name <- gsub("BHLH", "bHLH", gene_df$Ath_name)

  
resPaisS_g <- cbind(rownames(resPaisS), resPaisS)
colnames(resPaisS_g)[1] <- "genes"
resPaisS_v2 <- merge(gene_df, resPaisS_g, by ="genes")
rownames(resPaisS_v2) <- make.names(resPaisS_v2[,3],TRUE)

resWallS_g <- cbind(rownames(resWallS), resWallS)
colnames(resWallS_g)[1] <- "genes"
resWallS_v2 <- merge(gene_df, resWallS_g, by ="genes")
rownames(resWallS_v2) <- make.names(resWallS_v2[,3],TRUE)

resPaisR_g <- cbind(rownames(resPaisR), resPaisR)
colnames(resPaisR_g)[1] <- "genes"
resPaisR_v2 <- merge(gene_df, resPaisR_g, by ="genes")
rownames(resPaisR_v2) <- make.names(resPaisR_v2[,3],TRUE)

resWallR_g <- cbind(rownames(resWallR), resWallR)
colnames(resWallR_g)[1] <- "genes"
resWallR_v2 <- merge(gene_df, resWallR_g, by ="genes")
rownames(resWallR_v2) <- make.names(resWallR_v2[,3],TRUE)

###Enhanced version of volcano plots for pubication

#BiocManager::install("EnhancedVolcano")
#devtools::install_github("kevinblighe/EnhancedVolcano")
#install.packages("ggrepel")

library(EnhancedVolcano)

##Modified version based on source for EnhancedVolcano

#Normalized by all samples
EnhancedVolcano_GJL <- function(
  toptable,
  lab,
  x,
  y,
  selectLab = NULL,
  xlim = c(min(toptable[,x], na.rm=TRUE),
    max(toptable[,x], na.rm=TRUE)),
  ylim = c(0, max(-log10(toptable[,y]), na.rm=TRUE) + 5),
  xlab = bquote(~Log[2]~ "fold change"),
  ylab = bquote(~-Log[10]~italic(P)),
  axisLabSize = 12,
  pCutoff = 0.05,
  pLabellingCutoff = pCutoff,
  FCcutoff = 0.5,
  title = "",
  titleLabSize = 16,
  transcriptPointSize = 0.5,
  transcriptLabSize = 3.0,
  labhjust = -2,
  labvjust = -2,
  col = c("grey30",  "royalblue", "red2"),
  colOverride = NULL,
  colAlpha = 1/2,
  legend = c("NS", "DOWN","UP"),
  legendPosition = "top",
  legendLabSize = 10,
  legendIconSize = 3.0,
  legendVisible = TRUE,
  shade = NULL,
  shadeLabel = NULL,
  shadeAlpha = 1/2,
  shadeFill = "grey",
  shadeSize = 0.01,
  shadeBins = 2,
  drawConnectors = TRUE,
  widthConnectors = 0.6,
  colConnectors = "black",
  boxedLabels = TRUE,
  cutoffLineType = "longdash",
  cutoffLineCol = "black",
  cutoffLineWidth = 0.4,
  gridlines.major = TRUE,
  gridlines.minor = TRUE,
  border = "partial",
  borderWidth = 0.8,
  borderColour = "black")
{
  if(!requireNamespace("ggplot2")) {
    stop("Please install ggplot2 first.", call.=FALSE)
  }

  if(!requireNamespace("ggrepel")) {
    stop("Please install ggrepel first.", call.=FALSE)
  }

  if(!is.numeric(toptable[,x])) {
    stop(paste(x, " is not numeric!", sep=""))
  }

  if(!is.numeric(toptable[,y])) {
    stop(paste(y, " is not numeric!", sep=""))
  }

  i <- xvals <- yvals <- Sig <- NULL

  toptable <- as.data.frame(toptable)
  toptable$Sig <- "NS"
  toptable$Sig[(toptable[,y]<pCutoff) &
    (toptable[,x] >FCcutoff)] <- "FC_P_UP"
  toptable$Sig[(toptable[,y]<pCutoff) &
    (toptable[,x] < -FCcutoff)] <- "FC_P_DOWN"
  toptable$Sig <- factor(toptable$Sig,
    levels=c("NS","FC_P_DOWN", "FC_P_UP"))

  # some software programs return 0 for very low p-values
  # These throw an error in EnhancedVolcano
  # Detect these, issue warning, and convert these to
  # machine-lowest value possible
  if (min(toptable[,y], na.rm=TRUE) == 0) {
    warning(paste("One or more P values is 0.",
      "Converting to minimum possible value..."),
      call. = FALSE)
    toptable[which(toptable[,y] == 0), y] <- .Machine$double.xmin
  }

  toptable$lab <- lab
  toptable$xvals <- toptable[,x]
  toptable$yvals <- toptable[,y]

  # If user has supplied values in selectLab, convert labels to
  # NA and then re-set with those in selectLab
  if (!is.null(selectLab)) {
    names.new <- rep(NA, length(toptable$lab))
    indices <- which(toptable$lab %in% selectLab)
    names.new[indices] <- toptable$lab[indices]
    toptable$lab <- names.new
  }

  # create a base theme that will later be modified
  th <- theme_bw(base_size=14) +

    theme(
      legend.background=element_rect(),
      plot.title=element_text(angle=0, size=16, face="bold", vjust=1, hjust=-0.4),

      axis.text.x=element_text(angle=0, size=13, vjust=1),
      axis.text.y=element_text(angle=0, size=13, vjust=1),
      axis.title=element_text(size=axisLabSize),

      legend.position=legendPosition,
      legend.key=element_blank(),
      legend.key.size=unit(0.7, "cm"),
      legend.text=element_text(size=legendLabSize),

      title=element_text(size=legendLabSize),
      legend.title=element_blank())

  # Create the plot object differently based on whether colOverride is
  # NULL or not. This helps to avoid messing up the legend.
  if (!is.null(colOverride)) {
    plot <- ggplot(toptable, aes(x=xvals, y=-log10(yvals))) + th +

      guides(colour = guide_legend(order = 1,
        override.aes=list(size=legendIconSize))) +

      geom_point(aes(color=factor(names(colOverride))),
        alpha=colAlpha,
        size=transcriptPointSize,
        na.rm = TRUE) +

      scale_color_manual(values=colOverride)
  } else {
    plot <- ggplot(toptable, aes(x=xvals, y=-log10(yvals))) + th +

      guides(colour = guide_legend(order = 1,
        override.aes=list(size=legendIconSize))) +

      geom_point(aes(color=factor(Sig)),
        alpha=colAlpha,
        size=transcriptPointSize,
        na.rm = TRUE,
        show.legend = legendVisible) +

      scale_color_manual(values=c(NS=col[1],
        FC_P_DOWN=col[2],
        FC_P_UP = col[3]),
        labels=c(NS=legend[1],
        FC_P_DOWN=paste(legend[2], sep=""),
        FC_P_UP=paste(legend[3], sep="")))
  }

  # add more elements to the plot
  plot <- plot +

    xlab(xlab) +
    ylab(ylab) +

    xlim(xlim[1], xlim[2]) +
    ylim(ylim[1], ylim[2]) +

    ggtitle(title) +

    geom_vline(xintercept=c(-FCcutoff, FCcutoff),
      linetype=cutoffLineType,
      colour=cutoffLineCol,
      size=cutoffLineWidth) +

    geom_hline(yintercept=-log10(pCutoff),
      linetype=cutoffLineType,
      colour=cutoffLineCol,
      size=cutoffLineWidth)

  # Border around plot
  if (border == "full") {
    plot <- plot + theme(panel.border = element_rect(colour = borderColour, fill = NA, size = borderWidth))
  } else if (border == "partial") {
    plot <- plot + theme(axis.line = element_line(size = borderWidth, colour = borderColour),
      panel.border = element_blank(),
      panel.background = element_blank())
  } else {
    stop("Unrecognised value passed to 'border'. Must be 'full' or 'partial'")
  }

  # Gridlines
  if (gridlines.major == TRUE) {
    plot <- plot + theme(panel.grid.major = element_line())
  } else {
    plot <- plot + theme(panel.grid.major = element_blank())
  }
  if (gridlines.minor == TRUE) {
    plot <- plot + theme(panel.grid.minor = element_line())
  } else {
    plot <- plot + theme(panel.grid.minor = element_blank())
  }

  # For labeling with geom_text_repel (connectors) and
  # geom_text(.., check_overlap = TRUE), 4 possible scenarios
  # can arise
  if (drawConnectors == TRUE && is.null(selectLab)) {
    plot <- plot + geom_text_repel(
      data=subset(toptable,
        toptable[,y]<pLabellingCutoff &
          abs(toptable[,x])>FCcutoff),
        aes(label=subset(toptable,
          toptable[,y]<pLabellingCutoff &
            abs(toptable[,x])>FCcutoff)[,"lab"]),
        size = transcriptLabSize,
        segment.color = colConnectors,
        segment.size = widthConnectors,
        hjust = labhjust,
        vjust = labvjust,
        na.rm = TRUE)
  } else if (drawConnectors == TRUE && !is.null(selectLab)) {
    plot <- plot + geom_text_repel(
      data=subset(toptable,
        !is.na(toptable[,"lab"])),
        aes(label=subset(toptable,
          !is.na(toptable[,"lab"]))[,"lab"]),
        size = transcriptLabSize,
        segment.color = colConnectors,
        segment.size = widthConnectors,
        hjust = labhjust,
        vjust = labvjust,
        na.rm = TRUE)
  } else if (drawConnectors == FALSE && !is.null(selectLab)) {
    plot <- plot + geom_text(
      data=subset(toptable,
        !is.na(toptable[,"lab"])),
        aes(label=subset(toptable,
          !is.na(toptable[,"lab"]))[,"lab"]),
        size = transcriptLabSize,
        check_overlap = TRUE,
        hjust = labhjust,
        vjust = labvjust,
        na.rm = TRUE)
  } else if (drawConnectors == FALSE && is.null(selectLab)) {
    plot <- plot + geom_text(
      data=subset(toptable,
        toptable[,y]<pLabellingCutoff &
          abs(toptable[,x])>FCcutoff),
        aes(label=subset(toptable,
          toptable[,y]<pLabellingCutoff &
            abs(toptable[,x])>FCcutoff)[,"lab"]),
        size = transcriptLabSize,
        check_overlap = TRUE,
        hjust = labhjust,
        vjust = labvjust,
        na.rm = TRUE)
  }

  # shading
  if (!is.null(shade)) {
    plot <- plot + 
      stat_density2d(
        data = subset(toptable,
          rownames(toptable) %in% shade),
        fill = shadeFill,
        alpha = shadeAlpha,
        geom = "polygon",
        contour = TRUE,
        size = shadeSize,
        bins = shadeBins,
        show.legend = FALSE,
        na.rm = TRUE) +

      scale_fill_identity(name = shadeLabel,
        labels = shadeLabel,
        guide = "legend")
  }

  return(plot)
}

# Key genes
#P_key_root <- c("g01760","g15774","g21154", "g30123", "g13452", "g27931", "g11645")

#P_key_shoot <- c("g21154", "g21155", "g24495", "g01760", "g13702", "g19325", "g24765", "g21576", "g23102", "g12843", "g07320", "g27931", "g12617", "g11645", "g10414")

P_key_root <- c("FRO1", "bHLH101", "bHLH39", "ORG1", "VTL5", "VTL1", "FER1",  "FER3")

P_key_shoot <- c("bHLH39", "bHLH38", "bHLH100", "NAS4", "BTSL1", "FRO1", "FRO3", "ZIF1", "BTS", "ORG1", "OPT3", "PYE", "NRAMP4", "FER1", "FER4", "FER3", "YSL1")


#####

NS <- EnhancedVolcano_GJL(resNossS, 
                lab = rownames(resNossS), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,20),
                title =  "  (d)               ",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                ) 

PS <- EnhancedVolcano_GJL(resPaisS_v2, 
                lab = rownames(resPaisS_v2), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = P_key_shoot,
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,20),
                title = "  (e)               ",
                colAlpha = 0.4,
                legendLabSize = 14,
                legendIconSize = 3.0,
                legend=c("NS  ",  "Down-regulated ", "Up-regulated "),
                legendPosition = "none",
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = c(ifelse(resPaisS_v2$Ath_name %in% P_key_shoot, 3, 1)),
                transcriptLabSize = 3.0
                )
PS

WS <- EnhancedVolcano_GJL(resWallS_v2, 
                lab = rownames(resWallS_v2), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = P_key_shoot,
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,20),
                title = "  (f)               ",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = c(ifelse(resWallS_v2$Ath_name %in% P_key_shoot, 3, 1)),
                transcriptLabSize = 3.0
                )
WS

NR <- EnhancedVolcano_GJL(resNossR, 
                lab = rownames(resNossR), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,20),
                title = "(a)            Noss", 
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )

PR <- EnhancedVolcano_GJL(resPaisR_v2, 
                lab = rownames(resPaisR_v2), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = P_key_root,
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,20),
                title = "(b)             Pais", 
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = c(ifelse(resPaisR_v2$Ath_name %in% P_key_root, 3, 1)),
                transcriptLabSize = 3.0
                )
PR


WR <- EnhancedVolcano_GJL(resWallR_v2, 
                lab = rownames(resWallR_v2), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = P_key_root,
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,20),
                title = "(c)              Wall",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = c(ifelse(resWallR_v2$Ath_name %in% P_key_root, 3, 1)),
                transcriptLabSize = 3.0
                )

WR

legend_vol <- get_legend(PS + theme(legend.position="bottom"))
prow <- grid.arrange(NR,PR,WR,NS,PS,WS, ncol=3, nrow=2,
                     bottom=textGrob(bquote(~Log[2]~ "(fold change Cd "~italic(vs.)~"-Cd)"), gp = gpar(cex = 1.3), vjust=0),
                     left=textGrob(bquote(~-Log[10]~" (adjusted"~italic(P)~"-value)"), 
                                   rot = 90, vjust=1, gp = gpar(cex = 1.3)))
plot_grid( prow, legend_vol, ncol = 1, rel_heights = c(1, .1))             


###################


#Individuals

resNoss5R <- results(ddsNoss5R, contrast = c("Condition","Cd","Ctr"))
resNoss10R <- results(ddsNoss10R, contrast = c("Condition","Cd","Ctr"))
resPais5R <- results(ddsPais5R, contrast = c("Condition","Cd","Ctr"))
resPais9R <- results(ddsPais9R, contrast = c("Condition","Cd","Ctr"))
resWall7R <- results(ddsWall7R, contrast = c("Condition","Cd","Ctr"))
resWall10R <- results(ddsWall10R, contrast = c("Condition","Cd","Ctr"))
resNoss5S <- results(ddsNoss5S, contrast = c("Condition","Cd","Ctr"))
resNoss10S <- results(ddsNoss10S, contrast = c("Condition","Cd","Ctr"))
resPais5S <- results(ddsPais5S, contrast = c("Condition","Cd","Ctr"))
resPais9S <- results(ddsPais9S, contrast = c("Condition","Cd","Ctr"))
resWall7S <- results(ddsWall7S, contrast = c("Condition","Cd","Ctr"))
resWall10S <- results(ddsWall10S, contrast = c("Condition","Cd","Ctr"))

resNoss5S0 = cbind(genes, resNoss5S)
resNoss10S0 = cbind(genes, resNoss10S)
resPais5S0 = cbind(genes, resPais5S)
resPais9S0 = cbind(genes, resPais9S)
resWall7S0 = cbind(genes, resWall7S)
resWall10S0 = cbind(genes, resWall10S)

resNoss5R0 = cbind(genes, resNoss5R)
resNoss10R0 = cbind(genes, resNoss10R)
resPais5R0 = cbind(genes, resPais5R)
resPais9R0 = cbind(genes, resPais9R)
resWall7R0 = cbind(genes, resWall7R)
resWall10R0 = cbind(genes, resWall10R)

resNoss5S1 <- subset(resNoss5S0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resNoss10S1 <- subset(resNoss10S0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resPais5S1 <- subset(resPais5S0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resPais9S1 <- subset(resPais9S0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resWall7S1 <- subset(resWall7S0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resWall10S1 <- subset(resWall10S0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)

resNoss5R1 <- subset(resNoss5R0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resNoss10R1 <- subset(resNoss10R0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resPais5R1 <- subset(resPais5R0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resPais9R1 <- subset(resPais9R0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resWall7R1 <- subset(resWall7R0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)
resWall10R1 <- subset(resWall10R0, padj<=0.05 & log2FoldChange > 0.5 | padj<=0.05 & log2FoldChange < -0.5)


setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/CdvsCtr/Individual")

write.table(resNoss5S1, "resNoss5S1.txt", sep = "\t", row.names = FALSE)
write.table(resNoss10S1, "resNoss10S1.txt", sep = "\t", row.names = FALSE)
write.table(resPais5S1, "resNoss5S1.txt", sep = "\t", row.names = FALSE)
write.table(resPais9S1, "resPais9S1.txt", sep = "\t", row.names = FALSE)
write.table(resWall7S1, "resWall7S1.txt", sep = "\t", row.names = FALSE)
write.table(resWall10S1, "resWall10S1.txt", sep = "\t", row.names = FALSE)

write.table(resNoss5R1, "resNoss5R1.txt", sep = "\t", row.names = FALSE)
write.table(resNoss10R1, "resNoss10R1.txt", sep = "\t", row.names = FALSE)
write.table(resPais5R1, "resNoss5R1.txt", sep = "\t", row.names = FALSE)
write.table(resPais9R1, "resPais9R1.txt", sep = "\t", row.names = FALSE)
write.table(resWall7R1, "resWall7R1.txt", sep = "\t", row.names = FALSE)
write.table(resWall10R1, "resWall10R1.txt", sep = "\t", row.names = FALSE)


subset(resNoss10R, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1)
subset(resNoss5S, padj<=0.05 & log2FoldChange > 1 | padj<=0.05 & log2FoldChange < -1)

N5S <- EnhancedVolcano_GJL(resNoss5S, 
                lab = rownames(resNoss5S), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,40),
                title = "       Noss_05",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )


N10S <- EnhancedVolcano_GJL(resNoss10S, 
                lab = rownames(resNoss10S), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,40),
                title = "       Noss_10",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )


P5S <-EnhancedVolcano_GJL(resPais5S, 
                lab = rownames(resPais5S), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,40),
                title = "       Pais_05",
                colAlpha = 0.4,
                legendLabSize = 14,
                legendIconSize = 3.0,
                legend=c("NS  ",  "Down-regulated ", "Up-regulated "),
                legendPosition = "none",
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )

P9S <-EnhancedVolcano_GJL(resPais9S, 
                lab = rownames(resPais9S), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,40),
                title = "       Pais_09",
                colAlpha = 0.4,
                legendLabSize = 14,
                legendIconSize = 3.0,
                legend=c("NS  ",  "Down-regulated ", "Up-regulated "),
                legendPosition = "none",
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )

W7S <-EnhancedVolcano_GJL(resWall7S, 
                lab = rownames(resWall7S), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,40),
                title = "       Wall_07",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )

W10S <-EnhancedVolcano_GJL(resWall10S, 
                lab = rownames(resWall10S), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,40),
                title = "       Wall_10",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )

N5R <-EnhancedVolcano_GJL(resNoss5R, 
                lab = rownames(resNoss5R), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,40),
                title = "      Noss_05",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )

N10R <-EnhancedVolcano_GJL(resNoss10R, 
                lab = rownames(resNoss10R), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,40),
                title = "      Noss_10",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )

P5R <-EnhancedVolcano_GJL(resPais5R, 
                lab = rownames(resPais5R), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,40),
                title = "      Pais_05",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )

P9R <-EnhancedVolcano_GJL(resPais9R, 
                lab = rownames(resPais9R), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,40),
                title = "      Pais_09",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )

W7R <-EnhancedVolcano_GJL(resWall7R, 
                lab = rownames(resWall7R), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,40),
                title = "      Wall_07",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )

W10R <-EnhancedVolcano_GJL(resWall10R, 
                lab = rownames(resWall10R), 
                x = "log2FoldChange",
                y = "padj",
                selectLab = "",
                xlab = "",
                ylab = "",
                pCutoff = 0.05,
                FCcutoff = 0.5,
                xlim = c(-10,10),
                ylim = c(0,40),
                title = "      Wall_10",
                colAlpha = 0.4,
                legend=c("NS",  "Down-regulated", "Up-regulated"),
                legendPosition = "none",
                legendLabSize = 10,
                legendIconSize = 3.0,
                gridlines.major = FALSE,
                gridlines.minor = FALSE,  
                border = "full",
                borderWidth = 1.2,
                borderColour = "grey30",
                transcriptPointSize = 1.5,
                transcriptLabSize = 3.0
                )

legend_vol <- get_legend(PS + theme(legend.position="bottom"))
prow_S <- grid.arrange(N5S,N10S,P5S,P9S,W7S,W10S, ncol=2, nrow=3,
                       top=textGrob("    Shoot", gp = gpar(cex = 1.3)),
                     bottom=textGrob(bquote(~Log[2]~ "(fold change Cd "~italic(vs.)~"-Cd)"), gp = gpar(cex = 1.3), vjust=0)
                     )


prow_R <- grid.arrange(N5R,N10R,P5R,P9R,W7R,W10R, ncol=2, nrow=3,
                       top=textGrob("   Root", gp = gpar(cex = 1.3)),
                     bottom=textGrob(bquote(~Log[2]~ "(fold change Cd "~italic(vs.)~"-Cd)"), gp = gpar(cex = 1.3), vjust=0),
                     left=textGrob(bquote(~-Log[10]~" (adjusted"~italic(P)~"-value)"), rot = 90, vjust=1, gp = gpar(cex =1.3)))
 

plot_grid( prow_R, NULL, prow_S, legend_vol, ncol = 3, rel_heights = c(1, .1), labels=c("(a)","", "(b)"), rel_widths = c(1,.1,1), label_size = 18) 


```


#Extract overlapping genes between constitutively more highly expressed gene in Noss and Cd responded genes in Pais and Wall
```{r}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table")

CDG_Cdresp <- read.delim("CDG_Cdresp.txt")
CDG_Cdresp_Ano <- merge(CDG_Cdresp, Predata3[,c(1, 29:52)], by = "genes")

write.table(CDG_Cdresp_Ano, "CDG_Cdresp_Ano.txt", sep = "\t", row.names = FALSE)
```

#Significant GO for CDGE
```{r}
## For cuttoff of log2FC > 1
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table")
New_GO_CDEG_root <- read.delim("New_GO_CDEG_Noss_shoot.txt")
Pre_CDEG <- read.delim("Pre_CDEG.txt")

#require(data.table)
New_GO_CDEG_shoot_ano <- join(New_GO_CDEG_root, Pre_CDEG, by ="AGI")

write.table(New_GO_CDEG_shoot_ano, "New_GO_CDEG_root_ano.txt", sep = "\t", row.names = FALSE)


## For cuttoff of log2FC > 0.5
Old_GO_CDEG <- read.delim("Old_GO_CDEG_Noss.txt")
Predata_Noss <- read.delim("Dataset S1_CDEG_Noss.txt")


Old_GO_CDEG_ano <- join(Old_GO_CDEG, Predata_Noss, by ="AGI")
dim(Old_GO_CDEG_ano)

write.table(Old_GO_CDEG_ano, "Old_GO_CDEG_ano.txt", sep = "\t", row.names = FALSE)

```

#Correction for Pais_ctr_shoot
```{r}
Pais_corr_genes <- read.delim("Correction_forPais/Pais_shoot_genes.txt")
Pais_corr_values <- read.delim("Correction_forPais/Pais_shoot_values.txt")

Corrected_Pais <- join(Pais_corr_genes, Pais_corr_values)

write.table(Corrected_Pais, "Corrected_Pais_ctr_shoot.txt", sep = "\t", row.names = FALSE)

NP_genes <- read.delim("Correction_forPais/NP_ctr_shoot_genes.txt")
Corrected_NP <- join(NP_genes, Pais_corr_values)
write.table(Corrected_NP, "Correction_forPais/Corrected_NP_ctr_shoot.txt", sep = "\t", row.names = FALSE)

PW_genes <- read.delim("Correction_forPais/PW_ctr_shoot_genes.txt")
Corrected_PW <- join(PW_genes, Pais_corr_values)
write.table(Corrected_PW, "Correction_forPais/Corrected_PW_ctr_shoot.txt", sep = "\t", row.names = FALSE)


CdCtr_genes <- read.delim("Correction_forPais/CdCtr_Pais_genes.txt")
CdCtr_values <- read.delim("Correction_forPais/CdCtr_Pais_values.txt")
Corrected_CdCtr <- join(CdCtr_genes, CdCtr_values)
write.table(Corrected_CdCtr, "Correction_forPais/Corrected_CdCtr_ctr_shoot.txt", sep = "\t", row.names = FALSE)

```


#Overlapping DEGs upon Cd in both Pais and Wall for Photosynthesis
```{r}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/Test_GO")
Photosyn_AGI <- read.delim("Photosyn_AGI.txt")
CdvsCtr_AGI <- read.delim("CdvsCtr_AGI.txt")

CdvsCtr_shoot_Photo <- merge(CdvsCtr_AGI, Photosyn_AGI, by = "AGI")
write.table(CdvsCtr_shoot_Photo, "CdvsCtr_shoot_Photo.txt", sep = "\t", row.names = FALSE)


```

#Cd vs Ctr log2FC 0.5
```{r}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/CdvsCtr/Significant_genes")

Pais_Root_CdvsCtr_Sig0.5_Results <- read.delim("Pais_Root_CdvsCtr_Sig0.5_Results.txt")
Wall_Root_CdvsCtr_Sig0.5_Results <- read.delim("Wall_Root_CdvsCtr_Sig0.5_Results.txt")

Pais_Shoot_CdvsCtr_Sig0.5_Results <- read.delim("Pais_Shoot_CdvsCtr_Sig0.5_Results.txt")
Wall_Shoot_CdvsCtr_Sig0.5_Results <- read.delim("Wall_Shoot_CdvsCtr_Sig0.5_Results.txt")

Ano_Pais_Root_CdvsCtr_Sig0.5 <- merge(Predata[,1:21],Pais_Root_CdvsCtr_Sig0.5_Results,  by = "genes")
Ano_Wall_Root_CdvsCtr_Sig0.5 <- merge(Predata[,1:21],Wall_Root_CdvsCtr_Sig0.5_Results, by = "genes")
Ano_Pais_Shoot_CdvsCtr_Sig0.5 <- merge(Predata[,1:21],Pais_Shoot_CdvsCtr_Sig0.5_Results,  by = "genes")
Ano_Wall_Shoot_CdvsCtr_Sig0.5 <- merge(Predata[,1:21],Wall_Shoot_CdvsCtr_Sig0.5_Results,  by = "genes")

write.table(Ano_Pais_Root_CdvsCtr_Sig0.5, "Ano_Pais_Root_CdvsCtr_Sig0.5.txt", sep = "\t", row.names = FALSE)
write.table(Ano_Wall_Root_CdvsCtr_Sig0.5, "Ano_Wall_Root_CdvsCtr_Sig0.5.txt", sep = "\t", row.names = FALSE)
write.table(Ano_Pais_Shoot_CdvsCtr_Sig0.5, "Ano_Pais_Shoot_CdvsCtr_Sig0.5.txt", sep = "\t", row.names = FALSE)
write.table(Ano_Wall_Shoot_CdvsCtr_Sig0.5, "Ano_Wall_Shoot_CdvsCtr_Sig0.5.txt", sep = "\t", row.names = FALSE)

#########################
# Overlapping with CDEG in Noss (|log2fc| > 0.5)
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table")



All_Cdreponse_Root <- rbind(Ano_Pais_Root_CdvsCtr_Sig0.5[,c(1,7,8)],
                       Ano_Wall_Root_CdvsCtr_Sig0.5[,c(1,7,8)])
                       

All_Cdreponse_Shoot <- rbind(Ano_Pais_Shoot_CdvsCtr_Sig0.5[,c(1,7,8)],
                       Ano_Wall_Shoot_CdvsCtr_Sig0.5[,c(1,7,8)])


Overlapping_Cdrespon_CDEGNoss_Root <- merge(Ano_Shared_DEGs_MvsNM_Root_Ctr_NT1, All_Cdreponse_Root, by = "genes")
write.table(Overlapping_Cdrespon_CDEGNoss_Root, "Overlapping_Cdrespon_CDEGNoss_Root.txt", sep = "\t", row.names = FALSE)

Overlapping_Cdrespon_CDEGNoss_Shoot <- merge(Ano_Shared_DEGs_MvsNM_Shoot_Ctr_NT1, All_Cdreponse_Shoot, by = "genes")
write.table(Overlapping_Cdrespon_CDEGNoss_Shoot, "Overlapping_Cdrespon_CDEGNoss_Shoot.txt", sep = "\t", row.names = FALSE)

#With over-represented GO
GO_DEGs_Noss_Cutoff0.5 <- read.delim("GO_DEGs_Noss_Cutoff0.5.txt")
CDEG_GO_Cd_respon_all <- merge(GO_DEGs_Noss_Cutoff0.5, All_Cdreponse, by = "genes")
dim(CDEG_GO_Cd_respon_all)


###########################
#GO
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/CdvsCtr/GO/log2fc_0.5")


GO_Pais_Root_Up_0.5 <- read.delim("GO_Pais_Root_Up_0.5.txt")
GO_Wall_Root_Up_0.5 <- read.delim("GO_Wall_Root_Up_0.5.txt")
GO_Pais_Root_Down_0.5 <- read.delim("GO_Pais_Root_Down_0.5.txt")
GO_Wall_Root_Down_0.5 <- read.delim("GO_Wall_Root_Down_0.5.txt")

GO_Pais_Shoot_Up_0.5 <- read.delim("GO_Pais_Shoot_Up_0.5.txt")
GO_Wall_Shoot_Up_0.5 <- read.delim("GO_Wall_Shoot_Up_0.5.txt")
GO_Pais_Shoot_Down_0.5 <- read.delim("GO_Pais_Shoot_Down_0.5.txt")
GO_Wall_Shoot_Down_0.5 <- read.delim("GO_Wall_Shoot_Down_0.5.txt")

#write.table(GO_Pais_Root_Up_0.5, "GO_Pais_Root_Up_0.5.txt", sep = "\t", row.names = FALSE)
#write.table(GO_Wall_Root_Up_0.5, "GO_Wall_Root_Up_0.5.txt", sep = "\t", row.names = FALSE)
#write.table(GO_Pais_Root_Down_0.5, "GO_Pais_Root_Down_0.5.txt", sep = "\t", row.names = FALSE)
#write.table(GO_Wall_Root_Down_0.5, "GO_Wall_Root_Down_0.5.txt", sep = "\t", row.names = FALSE)

#write.table(GO_Pais_Shoot_Up_0.5, "GO_Pais_Shoot_Up_0.5.txt", sep = "\t", row.names = FALSE)
#write.table(GO_Wall_Shoot_Up_0.5, "GO_Wall_Shoot_Up_0.5.txt", sep = "\t", row.names = FALSE)
#write.table(GO_Pais_Shoot_Down_0.5, "GO_Pais_Shoot_Down_0.5.txt", sep = "\t", row.names = FALSE)
#write.table(GO_Wall_Shoot_Down_0.5, "GO_Wall_Shoot_Down_0.5.txt", sep = "\t", row.names = FALSE)


#Bind all GO results with space inbetween
# create a one-row matrix the same length as data
 
temprow2 <- matrix(c(rep.int(NA,length(GO_Pais_Root_Up_0.5))),nrow=1,ncol=length(GO_Pais_Root_Up_0.5))
 
# make it a data.frame and give cols the same names as data
 
newrow2 <- data.frame(temprow2)
colnames(newrow2) <- colnames(GO_Pais_Root_Up_0.5)
 

# rbind the empty row to data
 
GO_all_0.5 <- rbind(newrow2, GO_Pais_Root_Up_0.5, newrow2, 
              GO_Wall_Root_Up_0.5, newrow2, 
              GO_Pais_Root_Down_0.5, newrow2,
              GO_Wall_Root_Down_0.5, newrow2,
              GO_Pais_Shoot_Up_0.5, newrow2, 
              GO_Wall_Shoot_Up_0.5, newrow2, 
              GO_Pais_Shoot_Down_0.5, newrow2,
              GO_Wall_Shoot_Down_0.5
              )

write.table(GO_all_0.5, "GO_all_0.5.txt", sep = "\t", row.names = FALSE)


### Extract overlapped GO

head(GO_Pais_Root_Up_0.5)[,1:9]

GO_NM_Root_up <- rbind(GO_Pais_Root_Up_0.5[,1:9], GO_Wall_Root_Up_0.5[,1:9])
GO_NM_Root_down <- rbind(GO_Pais_Root_Down_0.5[,1:9], GO_Wall_Root_Down_0.5[,1:9])
GO_NM_Shoot_up <- rbind(GO_Pais_Shoot_Up_0.5[,1:9], GO_Wall_Shoot_Up_0.5[,1:9])
GO_NM_Shoot_down <- rbind(GO_Pais_Shoot_Down_0.5[,1:9], GO_Wall_Shoot_Down_0.5[,1:9])

GO_NM_dupli_Root_up <- GO_NM_Root_up[duplicated(GO_NM_Root_up$term_name)| 
                                       duplicated(GO_NM_Root_up$term_name, fromLast=TRUE),]
GO_NM_dupli_Root_up <- with(GO_NM_dupli_Root_up,  GO_NM_dupli_Root_up[order(term_name) , ])

GO_NM_dupli_Root_down <- GO_NM_Root_down[duplicated(GO_NM_Root_down$term_name)| 
                                                 duplicated(GO_NM_Root_down$term_name, fromLast=TRUE),]
GO_NM_dupli_Root_down <- with(GO_NM_dupli_Root_down,  GO_NM_dupli_Root_down[order(term_name) , ])  
  
GO_NM_dupli_Shoot_up <- GO_NM_Shoot_up[duplicated(GO_NM_Shoot_up$term_name)| 
                                       duplicated(GO_NM_Shoot_up$term_name, fromLast=TRUE),]
GO_NM_dupli_Shoot_up <- with(GO_NM_dupli_Shoot_up,  GO_NM_dupli_Shoot_up[order(term_name) , ])

GO_NM_dupli_Shoot_down <- GO_NM_Shoot_down[duplicated(GO_NM_Shoot_down$term_name)| 
                                                 duplicated(GO_NM_Shoot_down$term_name, fromLast=TRUE),]
GO_NM_dupli_Shoot_down <- with(GO_NM_dupli_Shoot_down,  GO_NM_dupli_Shoot_down[order(term_name) , ])  
  
write.table(GO_NM_dupli_Root_up, "Duplicated/GO_NM_dupli_Root_up.txt", sep = "\t", row.names = FALSE)
write.table(GO_NM_dupli_Root_down, "Duplicated/GO_NM_dupli_Root_down.txt", sep = "\t", row.names = FALSE)
write.table(GO_NM_dupli_Shoot_up, "Duplicated/GO_NM_dupli_Shoot_up.txt", sep = "\t", row.names = FALSE)
write.table(GO_NM_dupli_Shoot_down, "Duplicated/GO_NM_dupli_Shoot_down.txt", sep = "\t", row.names = FALSE)


##############################################################
#GO for overlapping DEGs in Pais and Wall

CdvsCtr_GO_Overlap0 <- read.delim("Duplicated/GO_all_CdvsCtr_0.5.txt", check.names=FALSE)

CdvsCtr_GO_Overlap1 <- split(CdvsCtr_GO_Overlap0, CdvsCtr_GO_Overlap0$Tissue)
CdvsCtr_GO_Overlap_Root <- CdvsCtr_GO_Overlap1$Root
CdvsCtr_GO_Overlap_Root$logPvalue <- ifelse(CdvsCtr_GO_Overlap_Root$Regulation=="Down",CdvsCtr_GO_Overlap_Root$logPvalue*-1, CdvsCtr_GO_Overlap_Root$logPvalue)

CdvsCtr_GO_Overlap_Shoot <- CdvsCtr_GO_Overlap1$Shoot
CdvsCtr_GO_Overlap_Shoot$logPvalue <- ifelse(CdvsCtr_GO_Overlap_Shoot$Regulation=="Down",CdvsCtr_GO_Overlap_Shoot$logPvalue*-1, CdvsCtr_GO_Overlap_Shoot$logPvalue)

CdvsCtr_GO_Overlap_Root$just <- ifelse(CdvsCtr_GO_Overlap_Root$Regulation=="Down",0,1)
CdvsCtr_GO_Overlap_Shoot$just <- ifelse(CdvsCtr_GO_Overlap_Shoot$Regulation=="Down",0,1)

CdvsCtr_GO_Overlap_Root2 <- split(CdvsCtr_GO_Overlap_Root, CdvsCtr_GO_Overlap_Root$Population)
CdvsCtr_GO_Overlap_Root_Pais <- CdvsCtr_GO_Overlap_Root2$`Pais`
CdvsCtr_GO_Overlap_Root_Wall <- CdvsCtr_GO_Overlap_Root2$`Wall`
CdvsCtr_GO_Overlap_Shoot2 <- split(CdvsCtr_GO_Overlap_Shoot, CdvsCtr_GO_Overlap_Shoot$Population)
CdvsCtr_GO_Overlap_Shoot_Pais <- CdvsCtr_GO_Overlap_Shoot2$`Pais`
CdvsCtr_GO_Overlap_Shoot_Wall <- CdvsCtr_GO_Overlap_Shoot2$`Wall`

CdvsCtr_GO_Overlap_Root$Population <- factor(CdvsCtr_GO_Overlap_Root$Population, levels = c("Wall", "Pais"))
CdvsCtr_GO_Overlap_Shoot$Population <- factor(CdvsCtr_GO_Overlap_Shoot$Population, levels = c("Wall", "Pais"))

GO_PS_Root <- ggplot(CdvsCtr_GO_Overlap_Root, aes(y=logPvalue, x=reorder(GO_name, logPvalue), fill= Type, group=Population)) + 
      geom_bar(position=position_dodge(), stat="identity", colour="grey50", size=0.5) +
      facet_grid(Type ~ ., scales="free_y", space="free", labeller = label_bquote("")) +
      scale_y_continuous( limits=c(-15,15), breaks = c(-15,-10,-5,0,5,10,15), labels=c(15,10,5,0,5,10,15)) +
      scale_x_discrete("", labels=NULL) +
      coord_flip()  +
      labs(x= "", fill = "GO category", y=expression(
        paste(bold("Down-regulated"),"            -log(adjusted ", italic("P"), "-value)            ", bold("Up-regulated")))) + 
      theme_minimal() +
      geom_text(aes(x=GO_name, y=0, label=GO_name), hjust=CdvsCtr_GO_Overlap_Root$just, size = 6) +
      geom_text(aes(label=QandT), 
            position = position_stack(vjust = 0.5), size=5, fontface="bold") +
      theme(axis.text.y=element_text(size=15),axis.text.x=element_text(size=10), axis.title=element_text(size=12), strip.text.x =
            element_text(size=20), legend.text = element_text(size=12),legend.title = element_text(size=16)) +
      theme(panel.grid.major = element_blank()) +
      theme(panel.grid.minor = element_blank()) + 
      geom_hline(yintercept = 0, col= "grey40")  + 
      scale_fill_manual(values=c("brown1", "orange1","royalblue4")) 
 
   
GO_PS_Root

GO_PS_Shoot <- ggplot(CdvsCtr_GO_Overlap_Shoot, aes(y=logPvalue, x=reorder(GO_name, logPvalue), fill= Type, group=Population)) + 
      geom_bar(position=position_dodge(), stat="identity", colour="grey50", size=0.5) +
      facet_grid(Type ~ ., scales="free_y", space="free", labeller = label_bquote("")) +
      scale_y_continuous( limits=c(-15,15), breaks = c(-15,-10,-5,0,5,10,15), labels=c(15,10,5,0,5,10,15)) +
      scale_x_discrete("", labels=NULL) +
      coord_flip()  +
      labs(x= "", fill = "GO category", y=expression(
        paste(bold("Down-regulated"),"            -log(adjusted ", italic("P"), "-value)            ", bold("Up-regulated")))) + 
      theme_minimal() +
      geom_text(aes(x=GO_name, y=0, label=GO_name), hjust=CdvsCtr_GO_Overlap_Shoot$just, size = 6) +
      geom_text(aes(label=QandT), 
            position = position_stack(vjust = 0.5), size=5, fontface="bold") +
      theme(axis.text.y=element_text(size=15),axis.text.x=element_text(size=10), axis.title=element_text(size=12), strip.text.x =
            element_text(size=20), legend.text = element_text(size=12),legend.title = element_text(size=16)) +
      theme(panel.grid.major = element_blank()) +
      theme(panel.grid.minor = element_blank()) + 
      geom_hline(yintercept = 0, col= "grey40")  + 
      scale_fill_manual(values=c("brown1", "orange1","royalblue4")) 
 
   
GO_PS_Shoot

GO_legend <- g_legend(GO_PS_Root + theme(legend.position='bottom'))

grid.arrange(GO_PS_Root+theme(legend.position='hidden') + theme(axis.line.x = element_line(color="black", size = 0.5)),
             GO_PS_Shoot+theme(legend.position='hidden')+ theme(axis.line.x = element_line(color="black", size = 0.5)), 
             GO_legend,
             layout_matrix = cbind(c(1,NA,2,NA,3)),
             heights = c(2.5,0.1,1.6,0.1, 0.2)
             )




```


##Add GO to the universal data using thaliana gaf
```{r}
library(devtools)
#install_github("skinnider/flavin")
library(flavin)

setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/Genome/Athaliana/")

Tair <- read_gaf("tair.gaf.gz")
class(Tair)
dim(Tair)
colnames(Tair)[10] <- "AGI"
head(Predata)[,1:10]

Tair2 <- Tair[,c(3,5,10)]

head(as.data.frame(Tair2))

Tair_all <- join(Predata[,c(1,7)], as.data.frame(Tair2))



```

##Statistical test for the genes for pubilcation
```{r}
#For meiotic cell cycle
AGO9_Root.aov <- aov( g12002 ~ Population*Condition, data = NCPK_Root)
summary(AGO9_Root.aov)
TukeyHSD(AGO9_Root.aov)
AGO9_Root_res <- HSD.test(AGO9_Root.aov, c("Population", "Condition"), group=TRUE)
AGO9_Root_res$groups
oder.group_AGO9_Root <- AGO9_Root_res$groups[order(rownames(AGO9_Root_res$groups)),] 

#
plot(AGO9_Root.aov, 1)
leveneTest(g12002 ~ Population*Condition, data = NCPK_Root) #Equal variance?

plot(AGO9_Root.aov, 2)
histogram(AGO9_Root.aov$residuals)
shapiro.test(AGO9_Root.aov$residuals) #Normality?

AGO9_Root.fit = aov(g12002 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(AGO9_Root.fit)

NCPK_Root$log_g12002 <- log(NCPK_Root$g12002 + 0.01)
leveneTest(log_g12002 ~ Population*Condition, data = NCPK_Root)

#Box-cox (https://rstudio-pubs-static.s3.amazonaws.com/296375_65d9076a94314bcba4b690e7504d6cba.html)
library(Matrix)
library(car)
library(stats)
library(MASS)
library(graphics)


Box_AGO9_Root = boxcox(g12002+2 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_AGO9_Root = data.frame(Box_AGO9_Root$x, Box_AGO9_Root$y)            # Create a data frame with the results

Cox2_AGO9_Root = Cox_AGO9_Root[with(Cox_AGO9_Root, order(-Cox_AGO9_Root$Box_AGO9_Root.y)),] # Order the new data frame by decreasing y

Cox2_AGO9_Root[1,]                                  # Display the lambda wi

lambda_AGO9_Root = Cox2_AGO9_Root[1, "Box_AGO9_Root.x"]                 # Extract that lambda

T_box_AGO9_Root = (NCPK_Root$g12002 ^ lambda_AGO9_Root - 1)/lambda_AGO9_Root   # Transform the original data
T_box_AGO9_Root

box_AGO9_Root <- cbind(NCPK_Root[,1:6], T_box_AGO9_Root)

leveneTest(T_box_AGO9_Root ~ Population*Condition, data = box_AGO9_Root) #Equal variance?
AGO9_box_Root.aov <- aov( T_box_AGO9_Root ~ Population*Condition, data = box_AGO9_Root)
shapiro.test(AGO9_box_Root.aov$residuals) #Normality?
TukeyHSD(AGO9_box_Root.aov)  

AGO9_box_Root_res <- HSD.test(AGO9_box_Root.aov, c("Population", "Condition"), group=TRUE)
AGO9_box_Root_res$groups

#log
NCPK_Root$log_g12002 <- log(NCPK_Root$g12002 + 0.01)
leveneTest(log_g12002 ~ Population*Condition, data = NCPK_Root)

AGO9_Root_log.aov <- aov( log_g12002 ~ Population*Condition, data = NCPK_Root)
shapiro.test(AGO9_Root_log.aov$residuals)
summary(AGO9_Root_log.aov)
TukeyHSD(AGO9_Root_log.aov)
AGO9_Root_log_res <- HSD.test(AGO9_Root_log.aov, c("Population", "Condition"), group=TRUE)
AGO9_Root_log_res$groups
#

AGO9_Shoot.aov <- aov( g12002 ~ Population*Condition, data = NCPK_Shoot)
summary(AGO9_Shoot.aov)
TukeyHSD(AGO9_Shoot.aov)  
AGO9_Shoot_res <- HSD.test(AGO9_Shoot.aov, c("Population", "Condition"), group=TRUE)
AGO9_Shoot_res$groups
oder.group_AGO9_Shoot <- AGO9_Shoot_res$groups[order(rownames(AGO9_Shoot_res$groups)),] 

#Stats
plot(AGO9_Shoot.aov, 1)
shapiro.test(AGO9_Shoot.aov$residuals) #Normality?
leveneTest(g12002 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?


AGO9_Shoot.fit = aov(g12002 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(AGO9_Shoot.fit)

NCPK_Shoot$log_g12002 <- log10(NCPK_Shoot$g12002 + 0.01)
leveneTest(log_g12002 ~ Population*Condition, data = NCPK_Shoot)

#Box-cox
Box_AGO9_Shoot = boxcox(g12002+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_AGO9_Shoot = data.frame(Box_AGO9_Shoot$x, Box_AGO9_Shoot$y)            # Create a data frame with the results

Cox2_AGO9_Shoot = Cox_AGO9_Shoot[with(Cox_AGO9_Shoot, order(-Cox_AGO9_Shoot$Box_AGO9_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_AGO9_Shoot[1,]                                  # Display the lambda wi

lambda_AGO9_Shoot = Cox2_AGO9_Shoot[1, "Box_AGO9_Shoot.x"]                 # Extract that lambda

T_box_AGO9_Shoot = (NCPK_Shoot$g12002 ^ lambda_AGO9_Shoot - 1)/lambda_AGO9_Shoot   # Transform the original data
T_box_AGO9_Shoot

box_AGO9_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_AGO9_Shoot)

leveneTest(T_box_AGO9_Shoot ~ Population*Condition, data = box_AGO9_Shoot) #Equal variance?
AGO9_box_Shoot.aov <- aov( T_box_AGO9_Shoot ~ Population*Condition, data = box_AGO9_Shoot)
shapiro.test(AGO9_box_Shoot.aov$residuals) #Normality?
TukeyHSD(AGO9_box_Shoot.aov)  

AGO9_box_Shoot_res <- HSD.test(AGO9_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
AGO9_box_Shoot_res$groups

#log
NCPK_Shoot$log_g12002 <- log(NCPK_Shoot$g12002 + 0.01)
leveneTest(log_g12002 ~ Population*Condition, data = NCPK_Shoot)

AGO9_Shoot_log.aov <- aov( log_g12002 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(AGO9_Shoot_log.aov$residuals)
summary(AGO9_Shoot_log.aov)
TukeyHSD(AGO9_Shoot_log.aov)
AGO9_Shoot_log_res <- HSD.test(AGO9_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
AGO9_Shoot_log_res$groups
#

AGO9_NC <- summarySE(NCPK, measurevar="g12002", groupvars=c("Tissue","Population","Condition"))


AGO9_NC$Condition <- factor(AGO9_NC$Condition, levels = c("Control", "Cd"))
AGO9 <- ggplot(AGO9_NC, aes(x=Population, y=g12002, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="AGO9", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g12002, ymax=g12002+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g12002+sd+5, 
                  label=c(as.character(oder.group_AGO9_Root$groups),as.character(oder.group_AGO9_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

AGO9

ZYP1a_Root.aov <- aov( g13785 ~ Population*Condition, data = NCPK_Root)
summary(ZYP1a_Root.aov)
TukeyHSD(ZYP1a_Root.aov)
ZYP1a_Root_res <- HSD.test(ZYP1a_Root.aov, c("Population", "Condition"), group=TRUE)
ZYP1a_Root_res$groups
oder.group_ZYP1a_Root <- ZYP1a_Root_res$groups[order(rownames(ZYP1a_Root_res$groups)),] 

#Stats
plot(ZYP1a_Root.aov, 1)
leveneTest(g13785 ~ Population*Condition, data = NCPK_Root) #Equal variance?
shapiro.test(ZYP1a_Root.aov$residuals) #Normality?

ZYP1a_Root.fit = aov(g13785 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(ZYP1a_Root.fit)

NCPK_Root$log_g13785 <- log10(NCPK_Root$g13785 + 0.01)
leveneTest(log_g13785 ~ Population*Condition, data = NCPK_Root)

#Box-cox
Box_ZYP1a_Root = boxcox(g13785+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_ZYP1a_Root = data.frame(Box_ZYP1a_Root$x, Box_ZYP1a_Root$y)            # Create a data frame with the results

Cox2_ZYP1a_Root = Cox_ZYP1a_Root[with(Cox_ZYP1a_Root, order(-Cox_ZYP1a_Root$Box_ZYP1a_Root.y)),] # Order the new data frame by decreasing y

Cox2_ZYP1a_Root[1,]                                  # Display the lambda wi

lambda_ZYP1a_Root = Cox2_ZYP1a_Root[1, "Box_ZYP1a_Root.x"]                 # Extract that lambda

T_box_ZYP1a_Root = (NCPK_Root$g13785 ^ lambda_ZYP1a_Root - 1)/lambda_ZYP1a_Root   # Transform the original data
T_box_ZYP1a_Root

box_ZYP1a_Root <- cbind(NCPK_Root[,1:6], T_box_ZYP1a_Root)

leveneTest(T_box_ZYP1a_Root ~ Population*Condition, data = box_ZYP1a_Root) #Equal variance?
ZYP1a_box_Root.aov <- aov( T_box_ZYP1a_Root ~ Population*Condition, data = box_ZYP1a_Root)
shapiro.test(ZYP1a_box_Root.aov$residuals) #Normality?
TukeyHSD(ZYP1a_box_Root.aov)  

ZYP1a_box_Root_res <- HSD.test(ZYP1a_box_Root.aov, c("Population", "Condition"), group=TRUE)
ZYP1a_box_Root_res$groups

#log
NCPK_Root$log_g13785 <- log(NCPK_Root$g13785 + 0.01)
leveneTest(log_g13785 ~ Population*Condition, data = NCPK_Root)

ZYP1a_Root_log.aov <- aov( log_g13785 ~ Population*Condition, data = NCPK_Root)
shapiro.test(ZYP1a_Root_log.aov$residuals)
summary(ZYP1a_Root_log.aov)
TukeyHSD(ZYP1a_Root_log.aov)
ZYP1a_Root_log_res <- HSD.test(ZYP1a_Root_log.aov, c("Population", "Condition"), group=TRUE)
ZYP1a_Root_log_res$groups
#


ZYP1a_Shoot.aov <- aov( g13785 ~ Population*Condition, data = NCPK_Shoot)
summary(ZYP1a_Shoot.aov)
TukeyHSD(ZYP1a_Shoot.aov)  
ZYP1a_Shoot_res <- HSD.test(ZYP1a_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZYP1a_Shoot_res$groups
oder.group_ZYP1a_Shoot <- ZYP1a_Shoot_res$groups[order(rownames(ZYP1a_Shoot_res$groups)),] 

#
plot(ZYP1a_Shoot.aov, 1)
leveneTest(g13785 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

plot(ZYP1a_Shoot.aov, 2)
histogram(ZYP1a_Shoot.aov$residuals)
shapiro.test(ZYP1a_Shoot.aov$residuals) #Normality?

ZYP1a_Shoot.fit = aov(g13785 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(ZYP1a_Shoot.fit)


#Box-cox
Box_ZYP1a_Shoot = boxcox(g13785+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_ZYP1a_Shoot = data.frame(Box_ZYP1a_Shoot$x, Box_ZYP1a_Shoot$y)            # Create a data frame with the results

Cox2_ZYP1a_Shoot = Cox_ZYP1a_Shoot[with(Cox_ZYP1a_Shoot, order(-Cox_ZYP1a_Shoot$Box_ZYP1a_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_ZYP1a_Shoot[1,]                                  # Display the lambda wi

lambda_ZYP1a_Shoot = Cox2_ZYP1a_Shoot[1, "Box_ZYP1a_Shoot.x"]                 # Extract that lambda

T_box_ZYP1a_Shoot = (NCPK_Shoot$g13785 ^ lambda_ZYP1a_Shoot - 1)/lambda_ZYP1a_Shoot   # Transform the original data

box_ZYP1a_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_ZYP1a_Shoot)

leveneTest(T_box_ZYP1a_Shoot ~ Population*Condition, data = box_ZYP1a_Shoot) #Equal variance?
ZYP1a_box_Shoot.aov <- aov( T_box_ZYP1a_Shoot ~ Population*Condition, data = box_ZYP1a_Shoot)
shapiro.test(ZYP1a_box_Shoot.aov$residuals) #Normality?
TukeyHSD(ZYP1a_box_Shoot.aov)  

ZYP1a_box_Shoot_res <- HSD.test(ZYP1a_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZYP1a_box_Shoot_res$groups


#log
NCPK_Shoot$log_g13785 <- log(NCPK_Shoot$g13785 + 0.01)
leveneTest(log_g13785 ~ Population*Condition, data = NCPK_Shoot)

ZYP1a_Shoot_log.aov <- aov( log_g13785 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(ZYP1a_Shoot_log.aov$residuals)
summary(ZYP1a_Shoot_log.aov)
TukeyHSD(ZYP1a_Shoot_log.aov)
ZYP1a_Shoot_log_res <- HSD.test(ZYP1a_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
ZYP1a_Shoot_log_res$groups
#


ZYP1a_NC <- summarySE(NCPK, measurevar="g13785", groupvars=c("Tissue","Population","Condition"))
ZYP1a_NC$Condition <- factor(ZYP1a_NC$Condition, levels = c("Control", "Cd"))
ZYP1a <- ggplot(ZYP1a_NC, aes(x=Population, y=g13785, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZYP1a", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g13785, ymax=g13785+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g13785+sd+1, 
                  label=c(as.character(oder.group_ZYP1a_Root$groups),as.character(oder.group_ZYP1a_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZYP1a
######
ZYP1b_Root.aov <- aov( g13782 ~ Population*Condition, data = NCPK_Root)
summary(ZYP1b_Root.aov)
TukeyHSD(ZYP1b_Root.aov)
ZYP1b_Root_res <- HSD.test(ZYP1b_Root.aov, c("Population", "Condition"), group=TRUE)
ZYP1b_Root_res$groups
oder.group_ZYP1b_Root <- ZYP1b_Root_res$groups[order(rownames(ZYP1b_Root_res$groups)),] 

#Stats
plot(ZYP1b_Root.aov, 1)
leveneTest(g13782 ~ Population*Condition, data = NCPK_Root) #Equal variance?
shapiro.test(ZYP1b_Root.aov$residuals) #Normality?

ZYP1b_Root.fit = aov(g13782 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(ZYP1b_Root.fit)

NCPK_Root$log_g13782 <- log10(NCPK_Root$g13782 + 0.01)
leveneTest(log_g13782 ~ Population*Condition, data = NCPK_Root)

#Box-cox
Box_ZYP1b_Root = boxcox(g13782+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_ZYP1b_Root = data.frame(Box_ZYP1b_Root$x, Box_ZYP1b_Root$y)            # Create a data frame with the results

Cox2_ZYP1b_Root = Cox_ZYP1b_Root[with(Cox_ZYP1b_Root, order(-Cox_ZYP1b_Root$Box_ZYP1b_Root.y)),] # Order the new data frame by decreasing y

Cox2_ZYP1b_Root[1,]                                  # Display the lambda wi

lambda_ZYP1b_Root = Cox2_ZYP1b_Root[1, "Box_ZYP1b_Root.x"]                 # Extract that lambda

T_box_ZYP1b_Root = (NCPK_Root$g13782 ^ lambda_ZYP1b_Root - 1)/lambda_ZYP1b_Root   # Transform the original data
T_box_ZYP1b_Root

box_ZYP1b_Root <- cbind(NCPK_Root[,1:6], T_box_ZYP1b_Root)

leveneTest(T_box_ZYP1b_Root ~ Population*Condition, data = box_ZYP1b_Root) #Equal variance?
ZYP1b_box_Root.aov <- aov( T_box_ZYP1b_Root ~ Population*Condition, data = box_ZYP1b_Root)
shapiro.test(ZYP1b_box_Root.aov$residuals) #Normality?
TukeyHSD(ZYP1b_box_Root.aov)  

ZYP1b_box_Root_res <- HSD.test(ZYP1b_box_Root.aov, c("Population", "Condition"), group=TRUE)
ZYP1b_box_Root_res$groups

#log
NCPK_Root$log_g13782 <- log(NCPK_Root$g13782 + 0.01)
leveneTest(log_g13782 ~ Population*Condition, data = NCPK_Root)

ZYP1b_Root_log.aov <- aov( log_g13782 ~ Population*Condition, data = NCPK_Root)
shapiro.test(ZYP1b_Root_log.aov$residuals)
summary(ZYP1b_Root_log.aov)
TukeyHSD(ZYP1b_Root_log.aov)
ZYP1b_Root_log_res <- HSD.test(ZYP1b_Root_log.aov, c("Population", "Condition"), group=TRUE)
ZYP1b_Root_log_res$groups
#


ZYP1b_Shoot.aov <- aov( g13782 ~ Population*Condition, data = NCPK_Shoot)
summary(ZYP1b_Shoot.aov)
TukeyHSD(ZYP1b_Shoot.aov)  
ZYP1b_Shoot_res <- HSD.test(ZYP1b_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZYP1b_Shoot_res$groups
oder.group_ZYP1b_Shoot <- ZYP1b_Shoot_res$groups[order(rownames(ZYP1b_Shoot_res$groups)),] 

#
plot(ZYP1b_Shoot.aov, 1)
leveneTest(g13782 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

plot(ZYP1b_Shoot.aov, 2)
histogram(ZYP1b_Shoot.aov$residuals)
shapiro.test(ZYP1b_Shoot.aov$residuals) #Normality?

ZYP1b_Shoot.fit = aov(g13782 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(ZYP1b_Shoot.fit)


#Box-cox
Box_ZYP1b_Shoot = boxcox(g13782+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_ZYP1b_Shoot = data.frame(Box_ZYP1b_Shoot$x, Box_ZYP1b_Shoot$y)            # Create a data frame with the results

Cox2_ZYP1b_Shoot = Cox_ZYP1b_Shoot[with(Cox_ZYP1b_Shoot, order(-Cox_ZYP1b_Shoot$Box_ZYP1b_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_ZYP1b_Shoot[1,]                                  # Display the lambda wi

lambda_ZYP1b_Shoot = Cox2_ZYP1b_Shoot[1, "Box_ZYP1b_Shoot.x"]                 # Extract that lambda

T_box_ZYP1b_Shoot = (NCPK_Shoot$g13782 ^ lambda_ZYP1b_Shoot - 1)/lambda_ZYP1b_Shoot   # Transform the original data

box_ZYP1b_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_ZYP1b_Shoot)

leveneTest(T_box_ZYP1b_Shoot ~ Population*Condition, data = box_ZYP1b_Shoot) #Equal variance?
ZYP1b_box_Shoot.aov <- aov( T_box_ZYP1b_Shoot ~ Population*Condition, data = box_ZYP1b_Shoot)
shapiro.test(ZYP1b_box_Shoot.aov$residuals) #Normality?
TukeyHSD(ZYP1b_box_Shoot.aov)  

ZYP1b_box_Shoot_res <- HSD.test(ZYP1b_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZYP1b_box_Shoot_res$groups


#log
NCPK_Shoot$log_g13782 <- log(NCPK_Shoot$g13782 + 0.01)
leveneTest(log_g13782 ~ Population*Condition, data = NCPK_Shoot)

ZYP1b_Shoot_log.aov <- aov( log_g13782 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(ZYP1b_Shoot_log.aov$residuals)
summary(ZYP1b_Shoot_log.aov)
TukeyHSD(ZYP1b_Shoot_log.aov)
ZYP1b_Shoot_log_res <- HSD.test(ZYP1b_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
ZYP1b_Shoot_log_res$groups
#


ZYP1b_NC <- summarySE(NCPK, measurevar="g13782", groupvars=c("Tissue","Population","Condition"))
ZYP1b_NC$Condition <- factor(ZYP1b_NC$Condition, levels = c("Control", "Cd"))
ZYP1b <- ggplot(ZYP1b_NC, aes(x=Population, y=g13782, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="ZYP1b", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g13782, ymax=g13782+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g13782+sd+1, 
                  label=c(as.character(oder.group_ZYP1b_Root$groups),as.character(oder.group_ZYP1b_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

ZYP1b


SMC2_Root.aov <- aov( g12815 ~ Population*Condition, data = NCPK_Root)
summary(SMC2_Root.aov)
TukeyHSD(SMC2_Root.aov)
SMC2_Root_res <- HSD.test(SMC2_Root.aov, c("Population", "Condition"), group=TRUE)
SMC2_Root_res$groups
oder.group_SMC2_Root <- SMC2_Root_res$groups[order(rownames(SMC2_Root_res$groups)),] 

#Stats
leveneTest(g12815 ~ Population*Condition, data = NCPK_Root) #Equal variance?
shapiro.test(SMC2_Root.aov$residuals) #Normality?

SMC2_Root.fit = aov(g12815 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(SMC2_Root.fit)


#Box-cox
Box_SMC2_Root = boxcox(g12815+0.8 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_SMC2_Root = data.frame(Box_SMC2_Root$x, Box_SMC2_Root$y)            # Create a data frame with the results

Cox2_SMC2_Root = Cox_SMC2_Root[with(Cox_SMC2_Root, order(-Cox_SMC2_Root$Box_SMC2_Root.y)),] # Order the new data frame by decreasing y

Cox2_SMC2_Root[1,]                                  # Display the lambda wi

lambda_SMC2_Root = Cox2_SMC2_Root[1, "Box_SMC2_Root.x"]                 # Extract that lambda

T_box_SMC2_Root = (NCPK_Root$g12815 ^ lambda_SMC2_Root - 1)/lambda_SMC2_Root   # Transform the original data
T_box_SMC2_Root

box_SMC2_Root <- cbind(NCPK_Root[,1:6], T_box_SMC2_Root)

leveneTest(T_box_SMC2_Root ~ Population*Condition, data = box_SMC2_Root) #Equal variance?
SMC2_box_Root.aov <- aov( T_box_SMC2_Root ~ Population*Condition, data = box_SMC2_Root)
shapiro.test(SMC2_box_Root.aov$residuals) #Normality?
TukeyHSD(SMC2_box_Root.aov)  

SMC2_box_Root_res <- HSD.test(SMC2_box_Root.aov, c("Population", "Condition"), group=TRUE)
SMC2_box_Root_res$groups

#log
NCPK_Root$log_g12815 <- log(NCPK_Root$g12815 + 0.01)
leveneTest(log_g12815 ~ Population*Condition, data = NCPK_Root)

SMC2_Root_log.aov <- aov( log_g12815 ~ Population*Condition, data = NCPK_Root)
shapiro.test(SMC2_Root_log.aov$residuals)
summary(SMC2_Root_log.aov)
TukeyHSD(SMC2_Root_log.aov)
SMC2_Root_log_res <- HSD.test(SMC2_Root_log.aov, c("Population", "Condition"), group=TRUE)
SMC2_Root_log_res$groups
#



SMC2_Shoot.aov <- aov( g12815 ~ Population*Condition, data = NCPK_Shoot)
summary(SMC2_Shoot.aov)
TukeyHSD(SMC2_Shoot.aov)  
SMC2_Shoot_res <- HSD.test(SMC2_Shoot.aov, c("Population", "Condition"), group=TRUE)
SMC2_Shoot_res$groups
oder.group_SMC2_Shoot <- SMC2_Shoot_res$groups[order(rownames(SMC2_Shoot_res$groups)),] 

#Stats
shapiro.test(SMC2_Shoot.aov$residuals) #Normality?
leveneTest(g12815 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?


SMC2_Shoot.fit = aov(g12815 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(SMC2_Shoot.fit)

NCPK_Shoot$log_g12815 <- log10(NCPK_Shoot$g12815 + 0.01)
leveneTest(log_g12815 ~ Population*Condition, data = NCPK_Shoot)

#Box-cox
Box_SMC2_Shoot = boxcox(g12815+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_SMC2_Shoot = data.frame(Box_SMC2_Shoot$x, Box_SMC2_Shoot$y)            # Create a data frame with the results

Cox2_SMC2_Shoot = Cox_SMC2_Shoot[with(Cox_SMC2_Shoot, order(-Cox_SMC2_Shoot$Box_SMC2_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_SMC2_Shoot[1,]                                  # Display the lambda wi

lambda_SMC2_Shoot = Cox2_SMC2_Shoot[1, "Box_SMC2_Shoot.x"]                 # Extract that lambda

T_box_SMC2_Shoot = (NCPK_Shoot$g12815 ^ lambda_SMC2_Shoot - 1)/lambda_SMC2_Shoot   # Transform the original data
T_box_SMC2_Shoot

box_SMC2_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_SMC2_Shoot)

leveneTest(T_box_SMC2_Shoot ~ Population*Condition, data = box_SMC2_Shoot) #Equal variance?
SMC2_box_Shoot.aov <- aov( T_box_SMC2_Shoot ~ Population*Condition, data = box_SMC2_Shoot)
shapiro.test(SMC2_box_Shoot.aov$residuals) #Normality?
TukeyHSD(SMC2_box_Shoot.aov)  

SMC2_box_Shoot_res <- HSD.test(SMC2_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
SMC2_box_Shoot_res$groups

#log
NCPK_Shoot$log_g12815 <- log(NCPK_Shoot$g12815 + 0.01)
leveneTest(log_g12815 ~ Population*Condition, data = NCPK_Shoot)

SMC2_Shoot_log.aov <- aov( log_g12815 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(SMC2_Shoot_log.aov$residuals)
summary(SMC2_Shoot_log.aov)
TukeyHSD(SMC2_Shoot_log.aov)
SMC2_Shoot_log_res <- HSD.test(SMC2_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
SMC2_Shoot_log_res$groups
#



SMC2_NC <- summarySE(NCPK, measurevar="g12815", groupvars=c("Tissue","Population","Condition"))
SMC2_NC$Condition <- factor(SMC2_NC$Condition, levels = c("Control", "Cd"))
SMC2 <- ggplot(SMC2_NC, aes(x=Population, y=g12815, fill=Condition)) + 
    theme_bw() +
    geom_bar(position=position_dodge(), stat="identity", color="grey20") +
    labs(y="", title="SMC2", x="") +
    theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=13), axis.title=element_text(size=14), strip.text =
          element_text(size=16),legend.text = element_text(size=12),legend.title = element_text(size=13)) +
    geom_errorbar(aes(ymin=g12815, ymax=g12815+sd),
                  size=.3, color="grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
    facet_wrap( ~ Tissue) +
    geom_text(aes(x=Population, y=g12815+sd+3, 
                  label=c(as.character(oder.group_SMC2_Root$groups),as.character(oder.group_SMC2_Shoot$groups))),
                  position=position_dodge(width=0.9), size=4) +
    scale_fill_manual(values=c("grey90", "grey30"),
                      breaks=c("Control","Cd"))+       theme(panel.grid.major = element_blank())

SMC2

#######
## Metal-related genes
ZIP2_Root.aov <- aov( g11007 ~ Population*Condition, data = NCPK_Root)
summary(ZIP2_Root.aov)
TukeyHSD(ZIP2_Root.aov)
ZIP2_Root_res <- HSD.test(ZIP2_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP2_Root_res$groups
oder.group_ZIP2_Root <- ZIP2_Root_res$groups[order(rownames(ZIP2_Root_res$groups)),] 

#Stats
leveneTest(g11007 ~ Population*Condition, data = NCPK_Root) #Equal variance?
shapiro.test(ZIP2_Root.aov$residuals) #Normality?

ZIP2_Root.fit = aov(g11007 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(ZIP2_Root.fit)


#Box-cox
Box_ZIP2_Root = boxcox(g11007+0.6 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_ZIP2_Root = data.frame(Box_ZIP2_Root$x, Box_ZIP2_Root$y)            # Create a data frame with the results

Cox2_ZIP2_Root = Cox_ZIP2_Root[with(Cox_ZIP2_Root, order(-Cox_ZIP2_Root$Box_ZIP2_Root.y)),] # Order the new data frame by decreasing y

Cox2_ZIP2_Root[1,]                                  # Display the lambda wi

lambda_ZIP2_Root = Cox2_ZIP2_Root[1, "Box_ZIP2_Root.x"]                 # Extract that lambda

T_box_ZIP2_Root = (NCPK_Root$g11007 ^ lambda_ZIP2_Root - 1)/lambda_ZIP2_Root   # Transform the original data
T_box_ZIP2_Root

box_ZIP2_Root <- cbind(NCPK_Root[,1:6], T_box_ZIP2_Root)

leveneTest(T_box_ZIP2_Root ~ Population*Condition, data = box_ZIP2_Root) #Equal variance?
ZIP2_box_Root.aov <- aov( T_box_ZIP2_Root ~ Population*Condition, data = box_ZIP2_Root)
shapiro.test(ZIP2_box_Root.aov$residuals) #Normality?
TukeyHSD(ZIP2_box_Root.aov)  

ZIP2_box_Root_res <- HSD.test(ZIP2_box_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP2_box_Root_res$groups

#log
NCPK_Root$log_g11007 <- log10(NCPK_Root$g11007)
leveneTest(log_g11007 ~ Population*Condition, data = NCPK_Root)

ZIP2_Root_log.aov <- aov( log_g11007 ~ Population*Condition, data = NCPK_Root)
shapiro.test(ZIP2_Root_log.aov$residuals)

ZIP2_Root_log_res <- HSD.test(ZIP2_Root_log.aov, c("Population", "Condition"), group=TRUE)
ZIP2_Root_log_res$groups

#
###
#stats
#Original
HMA2_Root.aov <- aov( g04213 ~ Population*Condition, data = NCPK_Root)

HMA2_Root_res <- HSD.test(HMA2_Root.aov, c("Population", "Condition"), group=TRUE)
HMA2_Root_res$groups

shapiro.test(HMA2_Root.aov$residuals) #Normality?
leveneTest(g04213 ~ Population*Condition, data = NCPK_Root) #Equal variance?

HMA2_Root.fit = aov(g04213 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(HMA2_Root.fit)

#log
HMA2_Root_log.aov <- aov( log_g04213 ~ Population*Condition, data = NCPK_Root)
shapiro.test(HMA2_Root_log.aov$residuals)

NCPK_Root$log_g04213 <- log(NCPK_Root$g04213 + 0.01)
leveneTest(log_g04213 ~ Population*Condition, data = NCPK_Root)

HMA2_Root_log_res <- HSD.test(HMA2_Root_log.aov, c("Population", "Condition"), group=TRUE)
HMA2_Root_log_res$groups

#Box-cox
Box_HMA2_Root = boxcox(g04213+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_HMA2_Root = data.frame(Box_HMA2_Root$x, Box_HMA2_Root$y)            # Create a data frame with the results

Cox2_HMA2_Root = Cox_HMA2_Root[with(Cox_HMA2_Root, order(-Cox_HMA2_Root$Box_HMA2_Root.y)),] # Order the new data frame by decreasing y

Cox2_HMA2_Root[1,]                                  # Display the lambda wi

lambda_HMA2_Root = Cox2_HMA2_Root[1, "Box_HMA2_Root.x"]                 # Extract that lambda

T_box_HMA2_Root = (NCPK_Root$g04213 ^ lambda_HMA2_Root - 1)/lambda_HMA2_Root   # Transform the original data
T_box_HMA2_Root

box_HMA2_Root <- cbind(NCPK_Root[,1:6], T_box_HMA2_Root)

HMA2_box_Root.aov <- aov( T_box_HMA2_Root ~ Population*Condition, data = box_HMA2_Root)
shapiro.test(HMA2_box_Root.aov$residuals) #Normality?
leveneTest(T_box_HMA2_Root ~ Population*Condition, data = box_HMA2_Root) #Equal variance?

HMA2_box_Root_res <- HSD.test(HMA2_box_Root.aov, c("Population", "Condition"), group=TRUE)
HMA2_box_Root_res$groups
#
###
#stats
#Original
ZIP6_Root.aov <- aov( g24253 ~ Population*Condition, data = NCPK_Root)

ZIP6_Root_res <- HSD.test(ZIP6_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP6_Root_res$groups

shapiro.test(ZIP6_Root.aov$residuals) #Normality?
leveneTest(g24253 ~ Population*Condition, data = NCPK_Root) #Equal variance?

ZIP6_Root.fit = aov(g24253 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(ZIP6_Root.fit)

#log
ZIP6_Root_log.aov <- aov( log_g24253 ~ Population*Condition, data = NCPK_Root)
shapiro.test(ZIP6_Root_log.aov$residuals)

NCPK_Root$log_g24253 <- log(NCPK_Root$g24253 + 0.01)
leveneTest(log_g24253 ~ Population*Condition, data = NCPK_Root)

ZIP6_Root_log_res <- HSD.test(ZIP6_Root_log.aov, c("Population", "Condition"), group=TRUE)
ZIP6_Root_log_res$groups

#Box-cox
Box_ZIP6_Root = boxcox(g24253+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_ZIP6_Root = data.frame(Box_ZIP6_Root$x, Box_ZIP6_Root$y)            # Create a data frame with the results

Cox2_ZIP6_Root = Cox_ZIP6_Root[with(Cox_ZIP6_Root, order(-Cox_ZIP6_Root$Box_ZIP6_Root.y)),] # Order the new data frame by decreasing y

Cox2_ZIP6_Root[1,]                                  # Display the lambda wi

lambda_ZIP6_Root = Cox2_ZIP6_Root[1, "Box_ZIP6_Root.x"]                 # Extract that lambda

T_box_ZIP6_Root = (NCPK_Root$g24253 ^ lambda_ZIP6_Root - 1)/lambda_ZIP6_Root   # Transform the original data
T_box_ZIP6_Root

box_ZIP6_Root <- cbind(NCPK_Root[,1:6], T_box_ZIP6_Root)

ZIP6_box_Root.aov <- aov( T_box_ZIP6_Root ~ Population*Condition, data = box_ZIP6_Root)
shapiro.test(ZIP6_box_Root.aov$residuals) #Normality?
leveneTest(T_box_ZIP6_Root ~ Population*Condition, data = box_ZIP6_Root) #Equal variance?

ZIP6_box_Root_res <- HSD.test(ZIP6_box_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP6_box_Root_res$groups

###
#stats
#Original
IRT1_Root.aov <- aov( g15474 ~ Population*Condition, data = NCPK_Root)

IRT1_Root_res <- HSD.test(IRT1_Root.aov, c("Population", "Condition"), group=TRUE)
IRT1_Root_res$groups

shapiro.test(IRT1_Root.aov$residuals) #Normality?
leveneTest(g15474 ~ Population*Condition, data = NCPK_Root) #Equal variance?

IRT1_Root.fit = aov(g15474 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(IRT1_Root.fit)

#log
IRT1_Root_log.aov <- aov( log_g15474 ~ Population*Condition, data = NCPK_Root)
shapiro.test(IRT1_Root_log.aov$residuals)

NCPK_Root$log_g15474 <- log(NCPK_Root$g15474 + 0.01)
leveneTest(log_g15474 ~ Population*Condition, data = NCPK_Root)

IRT1_Root_log_res <- HSD.test(IRT1_Root_log.aov, c("Population", "Condition"), group=TRUE)
IRT1_Root_log_res$groups

#Box-cox
Box_IRT1_Root = boxcox(g15474+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_IRT1_Root = data.frame(Box_IRT1_Root$x, Box_IRT1_Root$y)            # Create a data frame with the results

Cox2_IRT1_Root = Cox_IRT1_Root[with(Cox_IRT1_Root, order(-Cox_IRT1_Root$Box_IRT1_Root.y)),] # Order the new data frame by decreasing y

Cox2_IRT1_Root[1,]                                  # Display the lambda wi

lambda_IRT1_Root = Cox2_IRT1_Root[1, "Box_IRT1_Root.x"]                 # Extract that lambda

T_box_IRT1_Root = (NCPK_Root$g15474 ^ lambda_IRT1_Root - 1)/lambda_IRT1_Root   # Transform the original data
T_box_IRT1_Root

box_IRT1_Root <- cbind(NCPK_Root[,1:6], T_box_IRT1_Root)

IRT1_box_Root.aov <- aov( T_box_IRT1_Root ~ Population*Condition, data = box_IRT1_Root)
shapiro.test(IRT1_box_Root.aov$residuals) #Normality?
leveneTest(T_box_IRT1_Root ~ Population*Condition, data = box_IRT1_Root) #Equal variance?

IRT1_box_Root_res <- HSD.test(IRT1_box_Root.aov, c("Population", "Condition"), group=TRUE)
IRT1_box_Root_res$groups

###
#stats
#Original
CAX10_Root.aov <- aov( g32370 ~ Population*Condition, data = NCPK_Root)

CAX10_Root_res <- HSD.test(CAX10_Root.aov, c("Population", "Condition"), group=TRUE)
CAX10_Root_res$groups

shapiro.test(CAX10_Root.aov$residuals) #Normality?
leveneTest(g32370 ~ Population*Condition, data = NCPK_Root) #Equal variance?

CAX10_Root.fit = aov(g32370 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(CAX10_Root.fit)

#log
CAX10_Root_log.aov <- aov( log_g32370 ~ Population*Condition, data = NCPK_Root)
shapiro.test(CAX10_Root_log.aov$residuals)

NCPK_Root$log_g32370 <- log(NCPK_Root$g32370 + 0.01)
leveneTest(log_g32370 ~ Population*Condition, data = NCPK_Root)

CAX10_Root_log_res <- HSD.test(CAX10_Root_log.aov, c("Population", "Condition"), group=TRUE)
CAX10_Root_log_res$groups

#Box-cox
Box_CAX10_Root = boxcox(g32370+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_CAX10_Root = data.frame(Box_CAX10_Root$x, Box_CAX10_Root$y)            # Create a data frame with the results

Cox2_CAX10_Root = Cox_CAX10_Root[with(Cox_CAX10_Root, order(-Cox_CAX10_Root$Box_CAX10_Root.y)),] # Order the new data frame by decreasing y

Cox2_CAX10_Root[1,]                                  # Display the lambda wi

lambda_CAX10_Root = Cox2_CAX10_Root[1, "Box_CAX10_Root.x"]                 # Extract that lambda

T_box_CAX10_Root = (NCPK_Root$g32370 ^ lambda_CAX10_Root - 1)/lambda_CAX10_Root   # Transform the original data
T_box_CAX10_Root

box_CAX10_Root <- cbind(NCPK_Root[,1:6], T_box_CAX10_Root)

CAX10_box_Root.aov <- aov( T_box_CAX10_Root ~ Population*Condition, data = box_CAX10_Root)
shapiro.test(CAX10_box_Root.aov$residuals) #Normality?
leveneTest(T_box_CAX10_Root ~ Population*Condition, data = box_CAX10_Root) #Equal variance?

CAX10_box_Root_res <- HSD.test(CAX10_box_Root.aov, c("Population", "Condition"), group=TRUE)
CAX10_box_Root_res$groups

###
#stats
#Original
PCR9_Root.aov <- aov( g24696 ~ Population*Condition, data = NCPK_Root)

PCR9_Root_res <- HSD.test(PCR9_Root.aov, c("Population", "Condition"), group=TRUE)
PCR9_Root_res$groups

shapiro.test(PCR9_Root.aov$residuals) #Normality?
leveneTest(g24696 ~ Population*Condition, data = NCPK_Root) #Equal variance?

PCR9_Root.fit = aov(g24696 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(PCR9_Root.fit)

#log
PCR9_Root_log.aov <- aov( log_g24696 ~ Population*Condition, data = NCPK_Root)
shapiro.test(PCR9_Root_log.aov$residuals)

NCPK_Root$log_g24696 <- log(NCPK_Root$g24696 + 0.01)
leveneTest(log_g24696 ~ Population*Condition, data = NCPK_Root)

PCR9_Root_log_res <- HSD.test(PCR9_Root_log.aov, c("Population", "Condition"), group=TRUE)
PCR9_Root_log_res$groups

#Box-cox
Box_PCR9_Root = boxcox(g24696+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_PCR9_Root = data.frame(Box_PCR9_Root$x, Box_PCR9_Root$y)            # Create a data frame with the results

Cox2_PCR9_Root = Cox_PCR9_Root[with(Cox_PCR9_Root, order(-Cox_PCR9_Root$Box_PCR9_Root.y)),] # Order the new data frame by decreasing y

Cox2_PCR9_Root[1,]                                  # Display the lambda wi

lambda_PCR9_Root = Cox2_PCR9_Root[1, "Box_PCR9_Root.x"]                 # Extract that lambda

T_box_PCR9_Root = (NCPK_Root$g24696 ^ lambda_PCR9_Root - 1)/lambda_PCR9_Root   # Transform the original data
T_box_PCR9_Root

box_PCR9_Root <- cbind(NCPK_Root[,1:6], T_box_PCR9_Root)

PCR9_box_Root.aov <- aov( T_box_PCR9_Root ~ Population*Condition, data = box_PCR9_Root)
shapiro.test(PCR9_box_Root.aov$residuals) #Normality?
leveneTest(T_box_PCR9_Root ~ Population*Condition, data = box_PCR9_Root) #Equal variance?

PCR9_box_Root_res <- HSD.test(PCR9_box_Root.aov, c("Population", "Condition"), group=TRUE)
PCR9_box_Root_res$groups

#Shoot

###
#stats
#Original
PCR8_Shoot.aov <- aov( g15174 ~ Population*Condition, data = NCPK_Shoot)

PCR8_Shoot_res <- HSD.test(PCR8_Shoot.aov, c("Population", "Condition"), group=TRUE)
PCR8_Shoot_res$groups

shapiro.test(PCR8_Shoot.aov$residuals) #Normality?
leveneTest(g15174 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

PCR8_Shoot.fit = aov(g15174 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(PCR8_Shoot.fit)

#log
PCR8_Shoot_log.aov <- aov( log_g15174 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(PCR8_Shoot_log.aov$residuals)

NCPK_Shoot$log_g15174 <- log(NCPK_Shoot$g15174 + 0.01)
leveneTest(log_g15174 ~ Population*Condition, data = NCPK_Shoot)

PCR8_Shoot_log_res <- HSD.test(PCR8_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
PCR8_Shoot_log_res$groups

#Box-cox
Box_PCR8_Shoot = boxcox(g15174+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_PCR8_Shoot = data.frame(Box_PCR8_Shoot$x, Box_PCR8_Shoot$y)            # Create a data frame with the results

Cox2_PCR8_Shoot = Cox_PCR8_Shoot[with(Cox_PCR8_Shoot, order(-Cox_PCR8_Shoot$Box_PCR8_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_PCR8_Shoot[1,]                                  # Display the lambda wi

lambda_PCR8_Shoot = Cox2_PCR8_Shoot[1, "Box_PCR8_Shoot.x"]                 # Extract that lambda

T_box_PCR8_Shoot = (NCPK_Shoot$g15174 ^ lambda_PCR8_Shoot - 1)/lambda_PCR8_Shoot   # Transform the original data
T_box_PCR8_Shoot

box_PCR8_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_PCR8_Shoot)

PCR8_box_Shoot.aov <- aov( T_box_PCR8_Shoot ~ Population*Condition, data = box_PCR8_Shoot)
shapiro.test(PCR8_box_Shoot.aov$residuals) #Normality?
leveneTest(T_box_PCR8_Shoot ~ Population*Condition, data = box_PCR8_Shoot) #Equal variance?

PCR8_box_Shoot_res <- HSD.test(PCR8_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
PCR8_box_Shoot_res$groups

###
#stats
#Original
HMA2_Shoot.aov <- aov( g04213 ~ Population*Condition, data = NCPK_Shoot)

HMA2_Shoot_res <- HSD.test(HMA2_Shoot.aov, c("Population", "Condition"), group=TRUE)
HMA2_Shoot_res$groups

shapiro.test(HMA2_Shoot.aov$residuals) #Normality?
leveneTest(g04213 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

HMA2_Shoot.fit = aov(g04213 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(HMA2_Shoot.fit)

#log
HMA2_Shoot_log.aov <- aov( log_g04213 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(HMA2_Shoot_log.aov$residuals)

NCPK_Shoot$log_g04213 <- log(NCPK_Shoot$g04213 + 0.01)
leveneTest(log_g04213 ~ Population*Condition, data = NCPK_Shoot)

HMA2_Shoot_log_res <- HSD.test(HMA2_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
HMA2_Shoot_log_res$groups

#Box-cox
Box_HMA2_Shoot = boxcox(g04213+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_HMA2_Shoot = data.frame(Box_HMA2_Shoot$x, Box_HMA2_Shoot$y)            # Create a data frame with the results

Cox2_HMA2_Shoot = Cox_HMA2_Shoot[with(Cox_HMA2_Shoot, order(-Cox_HMA2_Shoot$Box_HMA2_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_HMA2_Shoot[1,]                                  # Display the lambda wi

lambda_HMA2_Shoot = Cox2_HMA2_Shoot[1, "Box_HMA2_Shoot.x"]                 # Extract that lambda

T_box_HMA2_Shoot = (NCPK_Shoot$g04213 ^ lambda_HMA2_Shoot - 1)/lambda_HMA2_Shoot   # Transform the original data
T_box_HMA2_Shoot

box_HMA2_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_HMA2_Shoot)

HMA2_box_Shoot.aov <- aov( T_box_HMA2_Shoot ~ Population*Condition, data = box_HMA2_Shoot)
shapiro.test(HMA2_box_Shoot.aov$residuals) #Normality?
leveneTest(T_box_HMA2_Shoot ~ Population*Condition, data = box_HMA2_Shoot) #Equal variance?

HMA2_box_Shoot_res <- HSD.test(HMA2_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
HMA2_box_Shoot_res$groups
#

#stats
#Original
CHX2_Shoot.aov <- aov( g01507 ~ Population*Condition, data = NCPK_Shoot)

CHX2_Shoot_res <- HSD.test(CHX2_Shoot.aov, c("Population", "Condition"), group=TRUE)
CHX2_Shoot_res$groups

shapiro.test(CHX2_Shoot.aov$residuals) #Normality?
leveneTest(g01507 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

CHX2_Shoot.fit = aov(g01507 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(CHX2_Shoot.fit)

#log
CHX2_Shoot_log.aov <- aov( log_g01507 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(CHX2_Shoot_log.aov$residuals)

NCPK_Shoot$log_g01507 <- log(NCPK_Shoot$g01507 + 0.01)
leveneTest(log_g01507 ~ Population*Condition, data = NCPK_Shoot)

CHX2_Shoot_log_res <- HSD.test(CHX2_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
CHX2_Shoot_log_res$groups

#Box-cox
Box_CHX2_Shoot = boxcox(g01507+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_CHX2_Shoot = data.frame(Box_CHX2_Shoot$x, Box_CHX2_Shoot$y)            # Create a data frame with the results

Cox2_CHX2_Shoot = Cox_CHX2_Shoot[with(Cox_CHX2_Shoot, order(-Cox_CHX2_Shoot$Box_CHX2_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_CHX2_Shoot[1,]                                  # Display the lambda wi

lambda_CHX2_Shoot = Cox2_CHX2_Shoot[1, "Box_CHX2_Shoot.x"]                 # Extract that lambda

T_box_CHX2_Shoot = (NCPK_Shoot$g01507 ^ lambda_CHX2_Shoot - 1)/lambda_CHX2_Shoot   # Transform the original data
T_box_CHX2_Shoot

box_CHX2_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_CHX2_Shoot)

CHX2_box_Shoot.aov <- aov( T_box_CHX2_Shoot ~ Population*Condition, data = box_CHX2_Shoot)
shapiro.test(CHX2_box_Shoot.aov$residuals) #Normality?
leveneTest(T_box_CHX2_Shoot ~ Population*Condition, data = box_CHX2_Shoot) #Equal variance?

CHX2_box_Shoot_res <- HSD.test(CHX2_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
CHX2_box_Shoot_res$groups
#
###
#stats
#Original
CAX10_Shoot.aov <- aov( g32370 ~ Population*Condition, data = NCPK_Shoot)

CAX10_Shoot_res <- HSD.test(CAX10_Shoot.aov, c("Population", "Condition"), group=TRUE)
CAX10_Shoot_res$groups

shapiro.test(CAX10_Shoot.aov$residuals) #Normality?
leveneTest(g32370 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

CAX10_Shoot.fit = aov(g32370 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(CAX10_Shoot.fit)

#log
CAX10_Shoot_log.aov <- aov( log_g32370 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(CAX10_Shoot_log.aov$residuals)

NCPK_Shoot$log_g32370 <- log(NCPK_Shoot$g32370 + 0.01)
leveneTest(log_g32370 ~ Population*Condition, data = NCPK_Shoot)

CAX10_Shoot_log_res <- HSD.test(CAX10_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
CAX10_Shoot_log_res$groups

#Box-cox
Box_CAX10_Shoot = boxcox(g32370+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_CAX10_Shoot = data.frame(Box_CAX10_Shoot$x, Box_CAX10_Shoot$y)            # Create a data frame with the results

Cox2_CAX10_Shoot = Cox_CAX10_Shoot[with(Cox_CAX10_Shoot, order(-Cox_CAX10_Shoot$Box_CAX10_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_CAX10_Shoot[1,]                                  # Display the lambda wi

lambda_CAX10_Shoot = Cox2_CAX10_Shoot[1, "Box_CAX10_Shoot.x"]                 # Extract that lambda

T_box_CAX10_Shoot = (NCPK_Shoot$g32370 ^ lambda_CAX10_Shoot - 1)/lambda_CAX10_Shoot   # Transform the original data
T_box_CAX10_Shoot

box_CAX10_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_CAX10_Shoot)

CAX10_box_Shoot.aov <- aov( T_box_CAX10_Shoot ~ Population*Condition, data = box_CAX10_Shoot)
shapiro.test(CAX10_box_Shoot.aov$residuals) #Normality?
leveneTest(T_box_CAX10_Shoot ~ Population*Condition, data = box_CAX10_Shoot) #Equal variance?

CAX10_box_Shoot_res <- HSD.test(CAX10_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
CAX10_box_Shoot_res$groups



##For Wall

###
#stats
#Original
CHX8_Root.aov <- aov( g07136 ~ Population*Condition, data = NCPK_Root)

CHX8_Root_res <- HSD.test(CHX8_Root.aov, c("Population", "Condition"), group=TRUE)
CHX8_Root_res$groups

shapiro.test(CHX8_Root.aov$residuals) #Normality?
leveneTest(g07136 ~ Population*Condition, data = NCPK_Root) #Equal variance?

CHX8_Root.fit = aov(g07136 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(CHX8_Root.fit)

#log
CHX8_Root_log.aov <- aov( log_g07136 ~ Population*Condition, data = NCPK_Root)
shapiro.test(CHX8_Root_log.aov$residuals)

NCPK_Root$log_g07136 <- log(NCPK_Root$g07136 + 0.01)
leveneTest(log_g07136 ~ Population*Condition, data = NCPK_Root)

CHX8_Root_log_res <- HSD.test(CHX8_Root_log.aov, c("Population", "Condition"), group=TRUE)
CHX8_Root_log_res$groups

#Box-cox
Box_CHX8_Root = boxcox(g07136+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_CHX8_Root = data.frame(Box_CHX8_Root$x, Box_CHX8_Root$y)            # Create a data frame with the results

Cox2_CHX8_Root = Cox_CHX8_Root[with(Cox_CHX8_Root, order(-Cox_CHX8_Root$Box_CHX8_Root.y)),] # Order the new data frame by decreasing y

Cox2_CHX8_Root[1,]                                  # Display the lambda wi

lambda_CHX8_Root = Cox2_CHX8_Root[1, "Box_CHX8_Root.x"]                 # Extract that lambda

T_box_CHX8_Root = (NCPK_Root$g07136 ^ lambda_CHX8_Root - 1)/lambda_CHX8_Root   # Transform the original data
T_box_CHX8_Root

box_CHX8_Root <- cbind(NCPK_Root[,1:6], T_box_CHX8_Root)

CHX8_box_Root.aov <- aov( T_box_CHX8_Root ~ Population*Condition, data = box_CHX8_Root)
shapiro.test(CHX8_box_Root.aov$residuals) #Normality?
leveneTest(T_box_CHX8_Root ~ Population*Condition, data = box_CHX8_Root) #Equal variance?

CHX8_box_Root_res <- HSD.test(CHX8_box_Root.aov, c("Population", "Condition"), group=TRUE)
CHX8_box_Root_res$groups

###
#stats
#Original
CUTA_Root.aov <- aov( g03274 ~ Population*Condition, data = NCPK_Root)

CUTA_Root_res <- HSD.test(CUTA_Root.aov, c("Population", "Condition"), group=TRUE)
CUTA_Root_res$groups

shapiro.test(CUTA_Root.aov$residuals) #Normality?
leveneTest(g03274 ~ Population*Condition, data = NCPK_Root) #Equal variance?

CUTA_Root.fit = aov(g03274 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(CUTA_Root.fit)

#log
CUTA_Root_log.aov <- aov( log_g03274 ~ Population*Condition, data = NCPK_Root)
shapiro.test(CUTA_Root_log.aov$residuals)

NCPK_Root$log_g03274 <- log(NCPK_Root$g03274 + 0.01)
leveneTest(log_g03274 ~ Population*Condition, data = NCPK_Root)

CUTA_Root_log_res <- HSD.test(CUTA_Root_log.aov, c("Population", "Condition"), group=TRUE)
CUTA_Root_log_res$groups

#Box-cox
Box_CUTA_Root = boxcox(g03274+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_CUTA_Root = data.frame(Box_CUTA_Root$x, Box_CUTA_Root$y)            # Create a data frame with the results

Cox2_CUTA_Root = Cox_CUTA_Root[with(Cox_CUTA_Root, order(-Cox_CUTA_Root$Box_CUTA_Root.y)),] # Order the new data frame by decreasing y

Cox2_CUTA_Root[1,]                                  # Display the lambda wi

lambda_CUTA_Root = Cox2_CUTA_Root[1, "Box_CUTA_Root.x"]                 # Extract that lambda

T_box_CUTA_Root = (NCPK_Root$g03274 ^ lambda_CUTA_Root - 1)/lambda_CUTA_Root   # Transform the original data
T_box_CUTA_Root

box_CUTA_Root <- cbind(NCPK_Root[,1:6], T_box_CUTA_Root)

CUTA_box_Root.aov <- aov( T_box_CUTA_Root ~ Population*Condition, data = box_CUTA_Root)
shapiro.test(CUTA_box_Root.aov$residuals) #Normality?
leveneTest(T_box_CUTA_Root ~ Population*Condition, data = box_CUTA_Root) #Equal variance?

CUTA_box_Root_res <- HSD.test(CUTA_box_Root.aov, c("Population", "Condition"), group=TRUE)
CUTA_box_Root_res$groups

###
#stats
#Original
ZIP10_Root.aov <- aov( g27549 ~ Population*Condition, data = NCPK_Root)

ZIP10_Root_res <- HSD.test(ZIP10_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP10_Root_res$groups

shapiro.test(ZIP10_Root.aov$residuals) #Normality?
leveneTest(g27549 ~ Population*Condition, data = NCPK_Root) #Equal variance?

ZIP10_Root.fit = aov(g27549 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(ZIP10_Root.fit)

#log
ZIP10_Root_log.aov <- aov( log_g27549 ~ Population*Condition, data = NCPK_Root)
shapiro.test(ZIP10_Root_log.aov$residuals)

NCPK_Root$log_g27549 <- log(NCPK_Root$g27549 + 0.01)
leveneTest(log_g27549 ~ Population*Condition, data = NCPK_Root)

ZIP10_Root_log_res <- HSD.test(ZIP10_Root_log.aov, c("Population", "Condition"), group=TRUE)
ZIP10_Root_log_res$groups

#Box-cox
Box_ZIP10_Root = boxcox(g27549+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_ZIP10_Root = data.frame(Box_ZIP10_Root$x, Box_ZIP10_Root$y)            # Create a data frame with the results

Cox2_ZIP10_Root = Cox_ZIP10_Root[with(Cox_ZIP10_Root, order(-Cox_ZIP10_Root$Box_ZIP10_Root.y)),] # Order the new data frame by decreasing y

Cox2_ZIP10_Root[1,]                                  # Display the lambda wi

lambda_ZIP10_Root = Cox2_ZIP10_Root[1, "Box_ZIP10_Root.x"]                 # Extract that lambda

T_box_ZIP10_Root = (NCPK_Root$g27549 ^ lambda_ZIP10_Root - 1)/lambda_ZIP10_Root   # Transform the original data
T_box_ZIP10_Root

box_ZIP10_Root <- cbind(NCPK_Root[,1:6], T_box_ZIP10_Root)

ZIP10_box_Root.aov <- aov( T_box_ZIP10_Root ~ Population*Condition, data = box_ZIP10_Root)
shapiro.test(ZIP10_box_Root.aov$residuals) #Normality?
leveneTest(T_box_ZIP10_Root ~ Population*Condition, data = box_ZIP10_Root) #Equal variance?

ZIP10_box_Root_res <- HSD.test(ZIP10_box_Root.aov, c("Population", "Condition"), group=TRUE)
ZIP10_box_Root_res$groups

###
#stats
#Original
CMT1_Root.aov <- aov( g21322 ~ Population*Condition, data = NCPK_Root)

CMT1_Root_res <- HSD.test(CMT1_Root.aov, c("Population", "Condition"), group=TRUE)
CMT1_Root_res$groups

shapiro.test(CMT1_Root.aov$residuals) #Normality?
leveneTest(g21322 ~ Population*Condition, data = NCPK_Root) #Equal variance?

CMT1_Root.fit = aov(g21322 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(CMT1_Root.fit)

#log
CMT1_Root_log.aov <- aov( log_g21322 ~ Population*Condition, data = NCPK_Root)
shapiro.test(CMT1_Root_log.aov$residuals)

NCPK_Root$log_g21322 <- log(NCPK_Root$g21322 + 0.01)
leveneTest(log_g21322 ~ Population*Condition, data = NCPK_Root)

CMT1_Root_log_res <- HSD.test(CMT1_Root_log.aov, c("Population", "Condition"), group=TRUE)
CMT1_Root_log_res$groups

#Box-cox
Box_CMT1_Root = boxcox(g21322+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_CMT1_Root = data.frame(Box_CMT1_Root$x, Box_CMT1_Root$y)            # Create a data frame with the results

Cox2_CMT1_Root = Cox_CMT1_Root[with(Cox_CMT1_Root, order(-Cox_CMT1_Root$Box_CMT1_Root.y)),] # Order the new data frame by decreasing y

Cox2_CMT1_Root[1,]                                  # Display the lambda wi

lambda_CMT1_Root = Cox2_CMT1_Root[1, "Box_CMT1_Root.x"]                 # Extract that lambda

T_box_CMT1_Root = (NCPK_Root$g21322 ^ lambda_CMT1_Root - 1)/lambda_CMT1_Root   # Transform the original data
T_box_CMT1_Root

box_CMT1_Root <- cbind(NCPK_Root[,1:6], T_box_CMT1_Root)

CMT1_box_Root.aov <- aov( T_box_CMT1_Root ~ Population*Condition, data = box_CMT1_Root)
shapiro.test(CMT1_box_Root.aov$residuals) #Normality?
leveneTest(T_box_CMT1_Root ~ Population*Condition, data = box_CMT1_Root) #Equal variance?

CMT1_box_Root_res <- HSD.test(CMT1_box_Root.aov, c("Population", "Condition"), group=TRUE)
CMT1_box_Root_res$groups

###
#stats
#Original
PCR6_Root.aov <- aov( g24315 ~ Population*Condition, data = NCPK_Root)

PCR6_Root_res <- HSD.test(PCR6_Root.aov, c("Population", "Condition"), group=TRUE)
PCR6_Root_res$groups

shapiro.test(PCR6_Root.aov$residuals) #Normality?
leveneTest(g24315 ~ Population*Condition, data = NCPK_Root) #Equal variance?

PCR6_Root.fit = aov(g24315 ~ Population*Condition + Error(Individual), data = NCPK_Root) #Nested Anova
summary(PCR6_Root.fit)

#log
PCR6_Root_log.aov <- aov( log_g24315 ~ Population*Condition, data = NCPK_Root)
shapiro.test(PCR6_Root_log.aov$residuals)

NCPK_Root$log_g24315 <- log(NCPK_Root$g24315 + 0.01)
leveneTest(log_g24315 ~ Population*Condition, data = NCPK_Root)

PCR6_Root_log_res <- HSD.test(PCR6_Root_log.aov, c("Population", "Condition"), group=TRUE)
PCR6_Root_log_res$groups

#Box-cox
Box_PCR6_Root = boxcox(g24315+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Root
             )

Cox_PCR6_Root = data.frame(Box_PCR6_Root$x, Box_PCR6_Root$y)            # Create a data frame with the results

Cox2_PCR6_Root = Cox_PCR6_Root[with(Cox_PCR6_Root, order(-Cox_PCR6_Root$Box_PCR6_Root.y)),] # Order the new data frame by decreasing y

Cox2_PCR6_Root[1,]                                  # Display the lambda wi

lambda_PCR6_Root = Cox2_PCR6_Root[1, "Box_PCR6_Root.x"]                 # Extract that lambda

T_box_PCR6_Root = (NCPK_Root$g24315 ^ lambda_PCR6_Root - 1)/lambda_PCR6_Root   # Transform the original data
T_box_PCR6_Root

box_PCR6_Root <- cbind(NCPK_Root[,1:6], T_box_PCR6_Root)

PCR6_box_Root.aov <- aov( T_box_PCR6_Root ~ Population*Condition, data = box_PCR6_Root)
shapiro.test(PCR6_box_Root.aov$residuals) #Normality?
leveneTest(T_box_PCR6_Root ~ Population*Condition, data = box_PCR6_Root) #Equal variance?

PCR6_box_Root_res <- HSD.test(PCR6_box_Root.aov, c("Population", "Condition"), group=TRUE)
PCR6_box_Root_res$groups

#################
#Shoot

###
#stats
#Original
ZIP7_Shoot.aov <- aov( g23539 ~ Population*Condition, data = NCPK_Shoot)

ZIP7_Shoot_res <- HSD.test(ZIP7_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIP7_Shoot_res$groups

shapiro.test(ZIP7_Shoot.aov$residuals) #Normality?
leveneTest(g23539 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

ZIP7_Shoot.fit = aov(g23539 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(ZIP7_Shoot.fit)

#log
ZIP7_Shoot_log.aov <- aov( log_g23539 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(ZIP7_Shoot_log.aov$residuals)

NCPK_Shoot$log_g23539 <- log(NCPK_Shoot$g23539 + 0.01)
leveneTest(log_g23539 ~ Population*Condition, data = NCPK_Shoot)

ZIP7_Shoot_log_res <- HSD.test(ZIP7_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
ZIP7_Shoot_log_res$groups

#Box-cox
Box_ZIP7_Shoot = boxcox(g23539+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_ZIP7_Shoot = data.frame(Box_ZIP7_Shoot$x, Box_ZIP7_Shoot$y)            # Create a data frame with the results

Cox2_ZIP7_Shoot = Cox_ZIP7_Shoot[with(Cox_ZIP7_Shoot, order(-Cox_ZIP7_Shoot$Box_ZIP7_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_ZIP7_Shoot[1,]                                  # Display the lambda wi

lambda_ZIP7_Shoot = Cox2_ZIP7_Shoot[1, "Box_ZIP7_Shoot.x"]                 # Extract that lambda

T_box_ZIP7_Shoot = (NCPK_Shoot$g23539 ^ lambda_ZIP7_Shoot - 1)/lambda_ZIP7_Shoot   # Transform the original data
T_box_ZIP7_Shoot

box_ZIP7_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_ZIP7_Shoot)

ZIP7_box_Shoot.aov <- aov( T_box_ZIP7_Shoot ~ Population*Condition, data = box_ZIP7_Shoot)
shapiro.test(ZIP7_box_Shoot.aov$residuals) #Normality?
leveneTest(T_box_ZIP7_Shoot ~ Population*Condition, data = box_ZIP7_Shoot) #Equal variance?

ZIP7_box_Shoot_res <- HSD.test(ZIP7_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
ZIP7_box_Shoot_res$groups

###
#stats
#Original
CHX8_Shoot.aov <- aov( g07136 ~ Population*Condition, data = NCPK_Shoot)

CHX8_Shoot_res <- HSD.test(CHX8_Shoot.aov, c("Population", "Condition"), group=TRUE)
CHX8_Shoot_res$groups

shapiro.test(CHX8_Shoot.aov$residuals) #Normality?
leveneTest(g07136 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

CHX8_Shoot.fit = aov(g07136 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(CHX8_Shoot.fit)

#log
CHX8_Shoot_log.aov <- aov( log_g07136 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(CHX8_Shoot_log.aov$residuals)

NCPK_Shoot$log_g07136 <- log(NCPK_Shoot$g07136 + 0.01)
leveneTest(log_g07136 ~ Population*Condition, data = NCPK_Shoot)

CHX8_Shoot_log_res <- HSD.test(CHX8_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
CHX8_Shoot_log_res$groups

#Box-cox
Box_CHX8_Shoot = boxcox(g07136+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_CHX8_Shoot = data.frame(Box_CHX8_Shoot$x, Box_CHX8_Shoot$y)            # Create a data frame with the results

Cox2_CHX8_Shoot = Cox_CHX8_Shoot[with(Cox_CHX8_Shoot, order(-Cox_CHX8_Shoot$Box_CHX8_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_CHX8_Shoot[1,]                                  # Display the lambda wi

lambda_CHX8_Shoot = Cox2_CHX8_Shoot[1, "Box_CHX8_Shoot.x"]                 # Extract that lambda

T_box_CHX8_Shoot = (NCPK_Shoot$g07136 ^ lambda_CHX8_Shoot - 1)/lambda_CHX8_Shoot   # Transform the original data
T_box_CHX8_Shoot

box_CHX8_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_CHX8_Shoot)

CHX8_box_Shoot.aov <- aov( T_box_CHX8_Shoot ~ Population*Condition, data = box_CHX8_Shoot)
shapiro.test(CHX8_box_Shoot.aov$residuals) #Normality?
leveneTest(T_box_CHX8_Shoot ~ Population*Condition, data = box_CHX8_Shoot) #Equal variance?

CHX8_box_Shoot_res <- HSD.test(CHX8_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
CHX8_box_Shoot_res$groups

###
#stats
#Original
CUTA_Shoot.aov <- aov( g03274 ~ Population*Condition, data = NCPK_Shoot)

CUTA_Shoot_res <- HSD.test(CUTA_Shoot.aov, c("Population", "Condition"), group=TRUE)
CUTA_Shoot_res$groups

shapiro.test(CUTA_Shoot.aov$residuals) #Normality?
leveneTest(g03274 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

CUTA_Shoot.fit = aov(g03274 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(CUTA_Shoot.fit)

#log
CUTA_Shoot_log.aov <- aov( log_g03274 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(CUTA_Shoot_log.aov$residuals)

NCPK_Shoot$log_g03274 <- log(NCPK_Shoot$g03274 + 0.01)
leveneTest(log_g03274 ~ Population*Condition, data = NCPK_Shoot)

CUTA_Shoot_log_res <- HSD.test(CUTA_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
CUTA_Shoot_log_res$groups

#Box-cox
Box_CUTA_Shoot = boxcox(g03274+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_CUTA_Shoot = data.frame(Box_CUTA_Shoot$x, Box_CUTA_Shoot$y)            # Create a data frame with the results

Cox2_CUTA_Shoot = Cox_CUTA_Shoot[with(Cox_CUTA_Shoot, order(-Cox_CUTA_Shoot$Box_CUTA_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_CUTA_Shoot[1,]                                  # Display the lambda wi

lambda_CUTA_Shoot = Cox2_CUTA_Shoot[1, "Box_CUTA_Shoot.x"]                 # Extract that lambda

T_box_CUTA_Shoot = (NCPK_Shoot$g03274 ^ lambda_CUTA_Shoot - 1)/lambda_CUTA_Shoot   # Transform the original data
T_box_CUTA_Shoot

box_CUTA_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_CUTA_Shoot)

CUTA_box_Shoot.aov <- aov( T_box_CUTA_Shoot ~ Population*Condition, data = box_CUTA_Shoot)
shapiro.test(CUTA_box_Shoot.aov$residuals) #Normality?
leveneTest(T_box_CUTA_Shoot ~ Population*Condition, data = box_CUTA_Shoot) #Equal variance?

CUTA_box_Shoot_res <- HSD.test(CUTA_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
CUTA_box_Shoot_res$groups

###
#stats
#Original
CMT1_Shoot.aov <- aov( g21322 ~ Population*Condition, data = NCPK_Shoot)

CMT1_Shoot_res <- HSD.test(CMT1_Shoot.aov, c("Population", "Condition"), group=TRUE)
CMT1_Shoot_res$groups

shapiro.test(CMT1_Shoot.aov$residuals) #Normality?
leveneTest(g21322 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

CMT1_Shoot.fit = aov(g21322 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(CMT1_Shoot.fit)

#log
CMT1_Shoot_log.aov <- aov( log_g21322 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(CMT1_Shoot_log.aov$residuals)

NCPK_Shoot$log_g21322 <- log(NCPK_Shoot$g21322 + 0.01)
leveneTest(log_g21322 ~ Population*Condition, data = NCPK_Shoot)

CMT1_Shoot_log_res <- HSD.test(CMT1_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
CMT1_Shoot_log_res$groups

#Box-cox
Box_CMT1_Shoot = boxcox(g21322+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_CMT1_Shoot = data.frame(Box_CMT1_Shoot$x, Box_CMT1_Shoot$y)            # Create a data frame with the results

Cox2_CMT1_Shoot = Cox_CMT1_Shoot[with(Cox_CMT1_Shoot, order(-Cox_CMT1_Shoot$Box_CMT1_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_CMT1_Shoot[1,]                                  # Display the lambda wi

lambda_CMT1_Shoot = Cox2_CMT1_Shoot[1, "Box_CMT1_Shoot.x"]                 # Extract that lambda

T_box_CMT1_Shoot = (NCPK_Shoot$g21322 ^ lambda_CMT1_Shoot - 1)/lambda_CMT1_Shoot   # Transform the original data
T_box_CMT1_Shoot

box_CMT1_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_CMT1_Shoot)

CMT1_box_Shoot.aov <- aov( T_box_CMT1_Shoot ~ Population*Condition, data = box_CMT1_Shoot)
shapiro.test(CMT1_box_Shoot.aov$residuals) #Normality?
leveneTest(T_box_CMT1_Shoot ~ Population*Condition, data = box_CMT1_Shoot) #Equal variance?

CMT1_box_Shoot_res <- HSD.test(CMT1_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
CMT1_box_Shoot_res$groups

###
#stats
#Original
HIPP20_Shoot.aov <- aov( g03852 ~ Population*Condition, data = NCPK_Shoot)

HIPP20_Shoot_res <- HSD.test(HIPP20_Shoot.aov, c("Population", "Condition"), group=TRUE)
HIPP20_Shoot_res$groups

shapiro.test(HIPP20_Shoot.aov$residuals) #Normality?
leveneTest(g03852 ~ Population*Condition, data = NCPK_Shoot) #Equal variance?

HIPP20_Shoot.fit = aov(g03852 ~ Population*Condition + Error(Individual), data = NCPK_Shoot) #Nested Anova
summary(HIPP20_Shoot.fit)

#log
HIPP20_Shoot_log.aov <- aov( log_g03852 ~ Population*Condition, data = NCPK_Shoot)
shapiro.test(HIPP20_Shoot_log.aov$residuals)

NCPK_Shoot$log_g03852 <- log(NCPK_Shoot$g03852 + 0.01)
leveneTest(log_g03852 ~ Population*Condition, data = NCPK_Shoot)

HIPP20_Shoot_log_res <- HSD.test(HIPP20_Shoot_log.aov, c("Population", "Condition"), group=TRUE)
HIPP20_Shoot_log_res$groups

#Box-cox
Box_HIPP20_Shoot = boxcox(g03852+0.01 ~ 1,              # Transform Turbidity as a single vector
             lambda = seq(-6,6,0.1),  # Try values -6 to 6 by 0.1
             data = NCPK_Shoot
             )

Cox_HIPP20_Shoot = data.frame(Box_HIPP20_Shoot$x, Box_HIPP20_Shoot$y)            # Create a data frame with the results

Cox2_HIPP20_Shoot = Cox_HIPP20_Shoot[with(Cox_HIPP20_Shoot, order(-Cox_HIPP20_Shoot$Box_HIPP20_Shoot.y)),] # Order the new data frame by decreasing y

Cox2_HIPP20_Shoot[1,]                                  # Display the lambda wi

lambda_HIPP20_Shoot = Cox2_HIPP20_Shoot[1, "Box_HIPP20_Shoot.x"]                 # Extract that lambda

T_box_HIPP20_Shoot = (NCPK_Shoot$g03852 ^ lambda_HIPP20_Shoot - 1)/lambda_HIPP20_Shoot   # Transform the original data
T_box_HIPP20_Shoot

box_HIPP20_Shoot <- cbind(NCPK_Shoot[,1:6], T_box_HIPP20_Shoot)

HIPP20_box_Shoot.aov <- aov( T_box_HIPP20_Shoot ~ Population*Condition, data = box_HIPP20_Shoot)
shapiro.test(HIPP20_box_Shoot.aov$residuals) #Normality?
leveneTest(T_box_HIPP20_Shoot ~ Population*Condition, data = box_HIPP20_Shoot) #Equal variance?

HIPP20_box_Shoot_res <- HSD.test(HIPP20_box_Shoot.aov, c("Population", "Condition"), group=TRUE)
HIPP20_box_Shoot_res$groups


```

#New GO term analysis
```{r}
# Consitutively differentially exressed DEGs in Noss
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/CDEG_Noss_0.5/")

New_GO_CDEG_Noss0 <- read.delim("GO_Shared_DEGs_NvsPW_Ctr_forFig.txt")
New_GO_CDEG_Noss1 <- split(New_GO_CDEG_Noss0, New_GO_CDEG_Noss0$Tissue)
New_GO_CDEG_Noss_Root <- New_GO_CDEG_Noss1$Root
New_GO_CDEG_Noss_Shoot <- New_GO_CDEG_Noss1$Shoot

New_GO_CDEG_Noss_Root$logPvalue <- ifelse(New_GO_CDEG_Noss_Root$Regulation=="Down",New_GO_CDEG_Noss_Root$logPvalue*-1, New_GO_CDEG_Noss_Root$logPvalue)

New_GO_CDEG_Noss_Shoot$logPvalue <- ifelse(New_GO_CDEG_Noss_Shoot$Regulation=="Down",New_GO_CDEG_Noss_Shoot$logPvalue*-1, New_GO_CDEG_Noss_Shoot$logPvalue)


New_GO_CDEG_Noss_Root$just <- ifelse(New_GO_CDEG_Noss_Root$Regulation=="Down",0,1)
New_GO_CDEG_Noss_Shoot$just <- ifelse(New_GO_CDEG_Noss_Shoot$Regulation=="Down",0,1)


New_GO_CDEG_Noss_Root_gg <- ggplot(New_GO_CDEG_Noss_Root, aes(y=logPvalue, x=reorder(GO_name, logPvalue), fill= Type)) + 
      geom_bar( stat="identity") +  
      facet_grid(Type2 ~ ., scales="free_y", space="free", labeller = label_bquote("") ) + 
      scale_y_continuous( limits= c(-6,6), breaks = c(-6,-4,-2,0,2,4,6), labels=c(6,4,2,0,2,4,6)) +
      scale_x_discrete("", labels=NULL) +
      coord_flip()  +
      labs(x= "", fill = "GO category", y=expression(
        paste(bold("Lowly expressed in Noss"),"              -log(adjusted ", italic("P"), "-value)              ", bold("Highly expressed in Noss")))) + 
      theme_minimal() +
      geom_text(aes(x=GO_name, y=0, label=GO_name), hjust=New_GO_CDEG_Noss_Root$just, size = 6) +
      geom_text(aes(label=QanT), 
            position = position_stack(vjust = 0.5), size=5, fontface="bold") +
      theme(axis.text.y=element_text(size=15),axis.text.x=element_text(size=10), axis.title=element_text(size=12), strip.text.x =
            element_text(size=20), legend.text = element_text(size=12),legend.title = element_text(size=16)) +
      theme(panel.grid.major = element_blank()) +
      theme(panel.grid.minor = element_blank()) + 
      geom_hline(yintercept = 0, col= "grey40")  + 
      scale_fill_manual(values=c("brown1", "orange1"))

New_GO_CDEG_Noss_Root_gg

New_GO_CDEG_Noss_Shoot_gg <- ggplot(New_GO_CDEG_Noss_Shoot, aes(y=logPvalue, x=reorder(GO_name, logPvalue), fill= Type)) + 
      geom_bar( stat="identity") +  
      facet_grid(Type2 ~ ., scales="free_y", space="free", labeller = label_bquote("") ) + 
      scale_y_continuous( limits= c(-3,3), breaks = c(-3,-2,-1,0,1,2,3), labels=c(3,2,1,0,1,2,3)) +
      scale_x_discrete("", labels=NULL) +
      coord_flip()  +
      labs(x= "", fill = "GO category", y=expression(
        paste(bold("Lowly expressed in Noss"),"              -log(adjusted ", italic("P"), "-value)              ", bold("Highly expressed in Noss")))) + 
      theme_minimal() +
      geom_text(aes(x=GO_name, y=0, label=GO_name), hjust=New_GO_CDEG_Noss_Shoot$just, size = 6) +
      geom_text(aes(label=QanT), 
            position = position_stack(vjust = 0.5), size=5, fontface="bold") +
      theme(axis.text.y=element_text(size=15),axis.text.x=element_text(size=10), axis.title=element_text(size=12), strip.text.x =
            element_text(size=20), legend.text = element_text(size=12),legend.title = element_text(size=16)) +
      theme(panel.grid.major = element_blank()) +
      theme(panel.grid.minor = element_blank()) + 
      geom_hline(yintercept = 0, col= "grey40")  + 
      scale_fill_manual(values=c("brown1", "orange1","royalblue4"))
  
New_GO_CDEG_Noss_Shoot_gg  

#install.packages("lemon")
library(lemon)
GO_legend2 <- g_legend(New_GO_CDEG_Noss_Shoot_gg + theme(legend.position='bottom'))

grid.arrange(New_GO_CDEG_Noss_Root_gg+theme(legend.position='hidden') + theme(axis.line.x = element_line(color="black", size = 0.5)),
             New_GO_CDEG_Noss_Shoot_gg+theme(legend.position='hidden')+ theme(axis.line.x = element_line(color="black", size = 0.5)), 
             GO_legend2,
             layout_matrix = cbind(c(1,NA,2,NA,3)),
             heights = c(1.2,0.3,1.2,0.1,0.2)
             )


```

#NEW GO for Noss
```{r}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/CDEG_Noss_0.5")

Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT <- read.delim("Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT.txt")
Ano_Shared_DEGs_NPW_Root_Ctr_NT <- read.delim("Ano_Shared_DEGs_NPW_Root_Ctr_NT.txt")
Ano_Shared_DEGs_NvsP_NvsW_Shoot_Ctr_NT <- read.delim("Ano_Shared_DEGs_NvsP_NvsW_Shoot_Ctr_NT.txt")
Ano_Shared_DEGs_NPW_Shoot_Ctr_NT <- read.delim("Ano_Shared_DEGs_NPW_Shoot_Ctr_NT.txt")

Predata <- read.delim("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result2/Tables/Pre_dataset S1.txt")
Predata2 <- Predata[,1:9]
head(Predata2)

Predata3<- read.delim("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/Predataset3.txt")
head(Predata3)

# Extract DEGs of constitutively high expressed genes in Noss
#Root
Ano_Shared_DEGs_MvsNM_Root_Ctr_NT1 <- rbind(Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT, Ano_Shared_DEGs_NPW_Root_Ctr_NT)
dim(Ano_Shared_DEGs_MvsNM_Root_Ctr_NT1)

Predata3_Glength_NossR <- Predata3[,c(1:3,35:37)]

Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT2 <- merge(Predata3_Glength_NossR, Ano_Shared_DEGs_MvsNM_Root_Ctr_NT1, by = "genes")

Const_Noss_Ano_diff_R <- merge(Predata2[,c(1:2, 9)],Ano_Shared_DEGs_NvsP_NvsW_Root_Ctr_NT2, by = "genes" )
dim(Const_Noss_Ano_diff_R)

resNvsW_Ctr_Root_Sig <- read.delim("Ano_resNvsW_Ctr_Root_NT.txt")
Const_Noss_Ano_diff_R2 <- merge(Const_Noss_Ano_diff_R, resNvsW_Ctr_Root_Sig[,c(1, 13:17)], by = "genes" )
dim(Const_Noss_Ano_diff_R2)
head(Const_Noss_Ano_diff_R2)
colnames(Const_Noss_Ano_diff_R2)[4] <- "AGI"
write.table(as.data.frame(Const_Noss_Ano_diff_R2), file = "Ano_CDEG_Noss0.5_Root.txt", sep = "\t", row.names = FALSE)

#Shoot
Ano_Shared_DEGs_MvsNM_Shoot_Ctr_NT1 <- rbind(Ano_Shared_DEGs_NvsP_NvsW_Shoot_Ctr_NT, Ano_Shared_DEGs_NPW_Shoot_Ctr_NT)

Predata3_Glength_NossS <- Predata3[,c(1:3,29:31)]

Ano_Shared_DEGs_NvsP_NvsW_Shoot_Ctr_NT2 <- merge(Predata3_Glength_NossS,Ano_Shared_DEGs_MvsNM_Shoot_Ctr_NT1, by = "genes")

Const_Noss_Ano_diff_S <- merge(Predata2[,c(1:2, 9)],Ano_Shared_DEGs_NvsP_NvsW_Shoot_Ctr_NT2, by = "genes" )

resNvsW_Ctr_Shoot_Sig <- read.delim("Ano_resNvsW_Ctr_Shoot_NT.txt")
Const_Noss_Ano_diff_S2 <- merge(Const_Noss_Ano_diff_S, resNvsW_Ctr_Shoot_Sig[,c(1, 13:17)], by = "genes" )
dim(Const_Noss_Ano_diff_S2)
head(Const_Noss_Ano_diff_S2)
colnames(Const_Noss_Ano_diff_S2)[4] <- "AGI"
write.table(as.data.frame(Const_Noss_Ano_diff_S2), file = "Ano_CDEG_Noss0.5_Shoot.txt", sep = "\t", row.names = FALSE)


```



```{r}
#Merge with GO
library(plyr)

New_GO_AGI_Root <- read.delim("New_GO/GO_CDEG_Noss_new_AGI_Root.txt")
dim(New_GO_AGI_Root)
New_GO_AGI_Root_Ano <- join(New_GO_AGI_Root, Const_Noss_Ano_diff_R2)
dim(New_GO_AGI_Root_Ano)
write.table(as.data.frame(New_GO_AGI_Root_Ano), file = "New_GO_AGI_all_Root.txt", sep = "\t", row.names = FALSE)

New_GO_AGI_Shoot <- read.delim("New_GO/GO_CDEG_Noss_new_AGI_Shoot.txt")
dim(New_GO_AGI_Shoot)
New_GO_AGI_Shoot_Ano <- join(New_GO_AGI_Shoot, Const_Noss_Ano_diff_S2)
dim(New_GO_AGI_Shoot_Ano)

write.table(as.data.frame(New_GO_AGI_Shoot_Ano), file = "New_GO_AGI_all_Shoot.txt", sep = "\t", row.names = FALSE)


```


#For Table S8
```{r}
#Re-DESeq
ddsShoot <- DESeqDataSetFromMatrix(
countData = countdata[,1:36],
colData = coldata[1:36,],
design = ~ Type3)
ddsShoot

ddsRoot <- DESeqDataSetFromMatrix(
countData = countdata[,37:72],
colData = coldata[37:72,],
design = ~ Type3)
ddsRoot

ddsS <- DESeq(ddsShoot)
ddsR <- DESeq(ddsRoot)

resNvsP_Ctr_Shoot_NT <- results(ddsS, contrast=c("Type3", "NossCtr", "PaisCtr"))
resNvsP_Ctr_Root_NT <- results(ddsR, contrast=c("Type3", "NossCtr", "PaisCtr"))
resNvsW_Ctr_Shoot_NT <- results(ddsS, contrast=c("Type3", "NossCtr", "WallCtr"))
resNvsW_Ctr_Root_NT <- results(ddsR, contrast=c("Type3", "NossCtr", "WallCtr"))


resNvsP_Ctr_Shoot_NT_0 = cbind(genes, resNvsP_Ctr_Shoot_NT)
resNvsP_Ctr_Root_NT_0 = cbind(genes, resNvsP_Ctr_Root_NT)
resNvsW_Ctr_Shoot_NT_0 = cbind(genes, resNvsW_Ctr_Shoot_NT)
resNvsW_Ctr_Root_NT_0 = cbind(genes, resNvsW_Ctr_Root_NT)


setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/ForTableS7")

NP_Root <- read.delim("NP_Root.txt")
NP_Root_GO <- read.delim("NP_Root_GO.txt")

NP_Shoot <- read.delim("NP_Shoot.txt")
NP_Shoot_GO <- read.delim("NP_Shoot_GO.txt")

NW_Root <- read.delim("NW_Root.txt")
NW_Root_GO <- read.delim("NW_Root_GO.txt")

NW_Shoot <- read.delim("NW_Shoot.txt")
NW_Shoot_GO <- read.delim("NW_Shoot_GO.txt")

library(plyr)

NP_Root_list0 <- merge(NP_Root_GO, NP_Root, by = "AGI")
colnames(NP_Root_list0)[2] <- "genes"
NP_Root_list <- join(NP_Root_list0, as.data.frame(resNvsW_Ctr_Root_NT_0), by = "genes" )
 
NP_Shoot_list0 <- merge(NP_Shoot_GO, NP_Shoot, by = "AGI")
colnames(NP_Shoot_list0)[2] <- "genes"
NP_Shoot_list <- join(NP_Shoot_list0, as.data.frame(resNvsW_Ctr_Shoot_NT_0), by = "genes") 
  
NW_Root_list0 <- merge(NW_Root_GO, NW_Root, by = "AGI")
colnames(NW_Root_list0)[2] <- "genes"
NW_Root_list <- join(NW_Root_list0, as.data.frame(resNvsP_Ctr_Root_NT_0), by = "genes")
 
NW_Shoot_list0 <- merge(NW_Shoot_GO, NW_Shoot, by = "AGI")
colnames(NW_Shoot_list0)[2] <- "genes"
NW_Shoot_list <- join(NW_Shoot_list0, as.data.frame(resNvsP_Ctr_Shoot_NT_0), by = "genes") 
  

write.table(NP_Root_list, file = "NP_Root_list.txt", sep = "\t", row.names = FALSE)
write.table(NP_Shoot_list, file = "NP_Shoot_list.txt", sep = "\t", row.names = FALSE)
write.table(NW_Root_list, file = "NW_Root_list.txt", sep = "\t", row.names = FALSE)
write.table(NW_Shoot_list, file = "NW_Shoot_list.txt", sep = "\t", row.names = FALSE)

###Re-organize not from GO but from all Cd responsive gene in NMs
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/ForTableS8/New_analysis")

Root_NvsP_list <- read.delim("Root_NvsP_list.txt")
Root_NvsW_list <- read.delim("Root_NvsW_list.txt")
Shoot_NvsP_list <- read.delim("Shoot_NvsP_list.txt")
Shoot_NvsW_list <- read.delim("Shoot_NvsW_list.txt")

Uni_data <- read.delim("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/ForTableS17/Universal_data.txt")

colnames(Uni_data)[1] <- "genes"

View(head(Uni_data))
Uni_data_8_Root <- Uni_data[, c(1,2,8:10,132:135,116:119)]
Uni_data_8_Shoot <- Uni_data[, c(1,2,8:10,120:123,110:113)]
head(Uni_data_8_Root)

Root_NvsP_Ano <- merge(Root_NvsP_list, Uni_data_8_Root, by = "genes")
Root_NvsW_Ano <- merge(Root_NvsW_list, Uni_data_8_Root, by = "genes")
Shoot_NvsP_Ano <- merge(Shoot_NvsP_list, Uni_data_8_Shoot, by = "genes")
Shoot_NvsW_Ano <- merge(Shoot_NvsW_list, Uni_data_8_Shoot, by = "genes")

write.table(Root_NvsP_Ano, file = "Root_NvsP_Ano.txt", sep = "\t", row.names = FALSE)
write.table(Root_NvsW_Ano, file = "Root_NvsW_Ano.txt", sep = "\t", row.names = FALSE)
write.table(Shoot_NvsP_Ano, file = "Shoot_NvsP_Ano.txt", sep = "\t", row.names = FALSE)
write.table(Shoot_NvsW_Ano, file = "Shoot_NvsW_Ano.txt", sep = "\t", row.names = FALSE)

```


#For the final table for Cd responsive genes in NM
```{r}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/ForTableS12")

Pais_Root_CdvsCtr_0.5 <- read.delim("Pais_Root_CdvsCtr_Sig0.5_Results.txt")
Pais_Shoot_CdvsCtr_0.5 <- read.delim("Pais_Shoot_CdvsCtr_Sig0.5_Results.txt")
Wall_Root_CdvsCtr_0.5 <- read.delim("Wall_Root_CdvsCtr_Sig0.5_Results.txt")
Wall_Shoot_CdvsCtr_0.5 <- read.delim("Wall_Shoot_CdvsCtr_Sig0.5_Results.txt")


Predata <- read.delim("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result2/Tables/Pre_dataset S1.txt")
Predata2 <- Predata[,1:9]
head(Predata2)


Predata3<- read.delim("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/Predataset3.txt")
colnames(Predata3)

#Pais_Root: 36, 39
#Pais_Shoot: 42, 45
#Wall_Root: 49, 52
#Wall_Shoot: 43, 46

Pais_Root_CdvsCtr_0.5_Ano <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "genes"),
        list(Predata2,
             Predata3[,c(1,36,39)],
             Pais_Root_CdvsCtr_0.5
             ))
Pais_Shoot_CdvsCtr_0.5_Ano <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "genes"),
        list(Predata2,
             Predata3[,c(1,42,45)],
             Pais_Shoot_CdvsCtr_0.5
             ))
Wall_Root_CdvsCtr_0.5_Ano <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "genes"),
        list(Predata2,
             Predata3[,c(1,49, 52)],
             Wall_Root_CdvsCtr_0.5
             ))
Wall_Shoot_CdvsCtr_0.5_Ano <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "genes"),
        list(Predata2,
             Predata3[,c(1,43, 46)],
             Wall_Shoot_CdvsCtr_0.5
             ))

write.table(Pais_Root_CdvsCtr_0.5_Ano, file = "Pais_Root_CdvsCtr_0.5_Ano.txt", sep = "\t", row.names = FALSE)
write.table(Pais_Shoot_CdvsCtr_0.5_Ano, file = "Pais_Shoot_CdvsCtr_0.5_Ano.txt", sep = "\t", row.names = FALSE)
write.table(Wall_Root_CdvsCtr_0.5_Ano, file = "Wall_Root_CdvsCtr_0.5_Ano.txt", sep = "\t", row.names = FALSE)
write.table(Wall_Shoot_CdvsCtr_0.5_Ano, file = "Wall_Shoot_CdvsCtr_0.5_Ano.txt", sep = "\t", row.names = FALSE)

```

## Wall-specifically DEGs upon Cd in root (Candidate genes for enhanced Cd root-to-shoot translocation)
```{r}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/Wall_excl")
Wall_excl_Root <- read.delim("Wall_excl_Root.txt")
Wall_excl_Root_Ano <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "genes"),
        list(Predata2,
             Wall_excl_Root,
             Wall_Root_CdvsCtr_0.5
             ))

head(Wall_excl_Root_Ano)

write.table(as.data.frame(Wall_excl_Root_Ano), "Wall_excl_Root_Ano.txt", sep = "\t", row.names = FALSE)

Wall_excl_Shoot <- read.delim("Wall_excl_Shoot.txt")
Wall_excl_Shoot_Ano <- Reduce(function(dtf1, dtf2) merge(dtf1, dtf2, by = "genes"),
        list(Predata2,
             Wall_excl_Shoot,
             Wall_Shoot_CdvsCtr_0.5
             ))

head(Wall_excl_Shoot_Ano)

write.table(as.data.frame(Wall_excl_Shoot_Ano), "Wall_excl_Shoot_Ano.txt", sep = "\t", row.names = FALSE)

```

## Relative expression of metal genes using Pheatmap (NT)
```{r}

#Make a gene list of all possible pairs in sig DEGs

setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result2/Normalized_Tissue/Tables")

Unidata_PM <- read.delim("UniData_forPheatmap.txt")



resNvsP_Ctr_Shoot_NT_Sig2  = subset(resNvsP_Ctr_Shoot_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resNvsP_Cd_Shoot_NT_Sig2  = subset(resNvsP_Cd_Shoot_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resNvsP_Ctr_Root_NT_Sig2  = subset(resNvsP_Ctr_Root_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resNvsP_Cd_Root_NT_Sig2  = subset(resNvsP_Cd_Root_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 

resPvsW_Ctr_Shoot_NT_Sig2  = subset(resPvsW_Ctr_Shoot_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resPvsW_Cd_Shoot_NT_Sig2  = subset(resPvsW_Cd_Shoot_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resPvsW_Ctr_Root_NT_Sig2  = subset(resPvsW_Ctr_Root_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resPvsW_Cd_Root_NT_Sig2  = subset(resPvsW_Cd_Root_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resNvsW_Ctr_Shoot_NT_Sig2  = subset(resNvsW_Ctr_Shoot_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resNvsW_Cd_Shoot_NT_Sig2  = subset(resNvsW_Cd_Shoot_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resNvsW_Ctr_Root_NT_Sig2  = subset(resNvsW_Ctr_Root_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resNvsW_Cd_Root_NT_Sig2  = subset(resNvsW_Cd_Root_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 

resNoss_Shoot_NT_Sig2  = subset(resNoss_Shoot_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resPais_Shoot_NT_Sig2  = subset(resPais_Shoot_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resWall_Shoot_NT_Sig2  = subset(resWall_Shoot_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1)
resNoss_Root_NT_Sig2  = subset(resNoss_Root_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resPais_Root_NT_Sig2  = subset(resPais_Root_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1) 
resWall_Root_NT_Sig2  = subset(resWall_Root_NT_0, padj<=0.05 & baseMean > 2 & log2FoldChange > 1 | padj<=0.05 & baseMean > 2 & log2FoldChange < -1)

All_DEG_Shoot <- rbind(resNvsP_Ctr_Shoot_NT_Sig2,
                      resNvsW_Ctr_Shoot_NT_Sig2,
                      resPvsW_Ctr_Shoot_NT_Sig2,
                      resNvsP_Cd_Shoot_NT_Sig2,
                      resNvsW_Cd_Shoot_NT_Sig2,
                      resPvsW_Cd_Shoot_NT_Sig2,
                      resNoss_Shoot_NT_Sig2,
                      resPais_Shoot_NT_Sig2,
                      resWall_Shoot_NT_Sig2 
                      )
All_DEG_Shoot <- All_DEG_Shoot[,1:2]
dim(All_DEG_Shoot)
All_DEG_Shoot <- All_DEG_Shoot[!duplicated(All_DEG_Shoot$genes),]
dim(All_DEG_Shoot)

All_DEG_Shoot_AGI <- merge(All_DEG_Shoot, Unidata_PM, by = "genes")  
All_DEG_Shoot_AGI_Metal <- merge(All_DEG_Shoot_AGI, Metal_list, by = "AGI")
dim(All_DEG_Shoot_AGI_Metal)
head(All_DEG_Shoot_AGI_Metal)[,1:8]
write.table(as.data.frame(All_DEG_Shoot_AGI_Metal), "All_DEG_Shoot_AGI_Metal.txt", sep = "\t", row.names = FALSE)

# Root

All_DEG_Root <- rbind(resNvsP_Ctr_Root_NT_Sig2,
                      resNvsW_Ctr_Root_NT_Sig2,
                      resPvsW_Ctr_Root_NT_Sig2,
                      resNvsP_Cd_Root_NT_Sig2,
                      resNvsW_Cd_Root_NT_Sig2,
                      resPvsW_Cd_Root_NT_Sig2,
                      resNoss_Root_NT_Sig2,
                      resPais_Root_NT_Sig2,
                      resWall_Root_NT_Sig2 
                      )

All_DEG_Root <- All_DEG_Root[,1:2]
dim(All_DEG_Root)
All_DEG_Root <- All_DEG_Root[!duplicated(All_DEG_Root$genes),]
dim(All_DEG_Root)

All_DEG_Root_AGI <- merge(All_DEG_Root, Unidata_PM, by = "genes")  
All_DEG_Root_AGI_Metal <- merge(All_DEG_Root_AGI, Metal_list, by = "AGI")
dim(All_DEG_Root_AGI_Metal)
head(All_DEG_Root_AGI_Metal)[,1:8]

write.table(as.data.frame(All_DEG_Root_AGI_Metal), "All_DEG_Root_AGI_Metal.txt", sep = "\t", row.names = FALSE)

##########################################################################

## Stricter filtering

# Root

resNvsPW = merge(resNvsP_Ctr_Root_NT_Sig2, resNvsW_Ctr_Root_NT_Sig2[,1:2], by= "genes")
resPvsNW = merge(resNvsP_Ctr_Root_NT_Sig2, resPvsW_Ctr_Root_NT_Sig2[,1:2], by= "genes")
resWvsNP = merge(resNvsW_Ctr_Root_NT_Sig2, resPvsW_Ctr_Root_NT_Sig2[,1:2], by= "genes")
resNvsPW = resNvsPW[,1:7]
colnames(resNvsPW)[2] <- "baseMean"
resPvsNW = resPvsNW[,1:7]
colnames(resPvsNW)[2] <- "baseMean"
resWvsNP = resWvsNP[,1:7]
colnames(resWvsNP)[2] <- "baseMean"


All_DEG_Root <- rbind(resNvsPW,
                      resPvsNW,
                      resWvsNP,
                      resNoss_Root_NT_Sig2,
                      resPais_Root_NT_Sig2,
                      resWall_Root_NT_Sig2 
                      )

All_DEG_Root <- All_DEG_Root[,1:2]
dim(All_DEG_Root)
All_DEG_Root <- All_DEG_Root[!duplicated(All_DEG_Root$genes),]
dim(All_DEG_Root)

All_DEG_Root_AGI <- merge(All_DEG_Root, Unidata_PM, by = "genes")  
All_DEG_Root_AGI_Metal <- merge(All_DEG_Root_AGI, Metal_list, by = "AGI")
dim(All_DEG_Root_AGI_Metal)
head(All_DEG_Root_AGI_Metal)[,1:8]

write.table(as.data.frame(All_DEG_Root_AGI_Metal), "All_DEG_Root_AGI_Metal2.txt", sep = "\t", row.names = FALSE)

# Shoot

resNvsPW = merge(resNvsP_Ctr_Shoot_NT_Sig2, resNvsW_Ctr_Shoot_NT_Sig2[,1:2], by= "genes")
resPvsNW = merge(resNvsP_Ctr_Shoot_NT_Sig2, resPvsW_Ctr_Shoot_NT_Sig2[,1:2], by= "genes")
resWvsNP = merge(resNvsW_Ctr_Shoot_NT_Sig2, resPvsW_Ctr_Shoot_NT_Sig2[,1:2], by= "genes")
resNvsPW = resNvsPW[,1:7]
colnames(resNvsPW)[2] <- "baseMean"
resPvsNW = resPvsNW[,1:7]
colnames(resPvsNW)[2] <- "baseMean"
resWvsNP = resWvsNP[,1:7]
colnames(resWvsNP)[2] <- "baseMean"


All_DEG_Shoot <- rbind(resNvsPW,
                      resPvsNW,
                      resWvsNP,
                      resNoss_Shoot_NT_Sig2,
                      resPais_Shoot_NT_Sig2,
                      resWall_Shoot_NT_Sig2 
                      )

All_DEG_Shoot <- All_DEG_Shoot[,1:2]
dim(All_DEG_Shoot)
All_DEG_Shoot <- All_DEG_Shoot[!duplicated(All_DEG_Shoot$genes),]
dim(All_DEG_Shoot)

All_DEG_Shoot_AGI <- merge(All_DEG_Shoot, Unidata_PM, by = "genes")  
All_DEG_Shoot_AGI_Metal <- merge(All_DEG_Shoot_AGI, Metal_list, by = "AGI")
dim(All_DEG_Shoot_AGI_Metal)
head(All_DEG_Shoot_AGI_Metal)[,1:8]

write.table(as.data.frame(All_DEG_Shoot_AGI_Metal), "All_DEG_Shoot_AGI_Metal2.txt", sep = "\t", row.names = FALSE)

# Calculate mean of each group in excel and load the data


## Load a mean-calculated data
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result2/Normalized_Tissue/Tables")
NC_mean_MetalGenes_R0 <- read.delim("All_DEG_Root_AGI_Metal4.txt", header = TRUE)
NC_mean_MetalGenes_S0 <- read.delim("All_DEG_Shoot_AGI_Metal3.txt", header = TRUE)

NC_mean_MetalGenes_R <- NC_mean_MetalGenes_R0[,5:10]
rownames(NC_mean_MetalGenes_R) <- NC_mean_MetalGenes_R0[,4] 

NC_mean_MetalGenes_S <- NC_mean_MetalGenes_S0[,5:10]
rownames(NC_mean_MetalGenes_S) <- NC_mean_MetalGenes_S0[,4] 

NormailizedNC_R <- t(apply(NC_mean_MetalGenes_R, 1, function(x)(x-min(x))/(max(x)-min(x))))
NormailizedNC_S <- t(apply(NC_mean_MetalGenes_S, 1, function(x)(x-min(x))/(max(x)-min(x))))

## Pheatmap for Shoot and Root
pres_R <- pheatmap(as.matrix(NormailizedNC_R), cluster_rows=TRUE, cluster_cols=FALSE, legend_breaks = c(0,1),legend_labels = c("Row min", "Row max"))
#cluster = cutree(pres_R$tree_row, k =7)[pres_R$tree_row[["order"]]]
#pres.clust_R <- cbind(pres_R, cluster)
#pres.clust_R <- as.data.frame(unlist(pres.clust_R[,2]))
#colnames(pres.clust_R) <- c("Category")
#row_annotations_R = data.frame( Category = factor(rep(c("R1", "R2", "R3", "R4", "R5", "R6", "R7", "R8", "R9", "R10"), 
 #                                                     c(8,13,17,7,5,5,22,12,7,17))))

#rownames(row_annotations_R) = rownames(pres.clust_R)

color.palette1  <- colorRampPalette(c("darkblue", "white", "red2"))(n=100)
color.palette2 = colorRampPalette( rev(brewer.pal(9, "RdYlGn")) )(100)
color.palette3 = colorRampPalette( rev(brewer.pal(9, "RdYlBu")) )(100)

dim(NormailizedNC_R)
pheatmap(as.matrix(NormailizedNC_R), 
         cluster_cols = FALSE, 
         gaps_col = c(2,4),
         main = "Root", 
         labels_col = c("Noss (Ctr)", "Noss (Cd)", "Pais (Ctr)", "Pais (Cd)", "Wall (Ctr)", "Wall (Cd)"),
         fontsize_col = 14,
         fontsize_row = 10,
         fontsize = 11,
         legend_breaks = c(0,1),
         legend_labels = c("Row min", "Row max"),
         legend = TRUE,
         color = color.palette1,
         cutree_rows = 7
          )

pres_S <- pheatmap(as.matrix(NormailizedNC_S), cluster_rows=TRUE, cluster_cols=FALSE, legend_breaks = c(0,1),legend_labels = c("Row min", "Row max"))
#cluster = cutree(pres_S$tree_row, k =9)[pres_S$tree_row[["order"]]]
#pres.clust_S <- cbind(pres_S, cluster)
#pres.clust_S <- as.data.frame(unlist(pres.clust_S[,2]))
#colnames(pres.clust_S) <- c("Category")
#row_annotations_S = data.frame( Category = factor(rep(c("S1", "S2", "S3", "S4", "S5", "S6", "S7","S8"), c(16,12,5,14,20,7,12,9))))

#rownames(row_annotations_S) = rownames(pres.clust_S)

dim(NormailizedNC_S)
pheatmap(as.matrix(NormailizedNC_S), 
         cluster_cols = FALSE, 
         gaps_col = c(2,4),
         main = "Shoot", 
         labels_col = c("Noss (Ctr)", "Noss (Cd)", "Pais (Ctr)", "Pais (Cd)", "Wall (Ctr)", "Wall (Cd)"),
         fontsize_col = 14,
         fontsize_row = 10,
         fontsize = 11,
         legend_breaks = c(0,1),
         legend_labels = c("Row min", "Row max"),
         legend = TRUE,
         color = color.palette1,
         cutree_rows = 7)


###################################
```



#For a S table- annotation for enriched Cd responsive genes both in Pais and Wall
```{r}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/ForTableS17")

AGI_GO_PW <- read.delim("AGI_GO_PW.txt")
dim(AGI_GO_PW)
head(AGI_GO_PW)

Uni_data <- read.delim("Universal_data.txt")
Uni_data2 <- Uni_data[,c(1:2, 8:10, 110:113, 116:119)]
head(Uni_data2)

Predata3<- read.delim("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/Predataset3.txt")
View(head(Predata3))
NCPK_GO_PW <- Predata3[,c(2, 30,33,31,34, 36, 39, 37, 40)]
head(NCPK_GO_PW)

Uni_NCPK_GO_PW <- cbind(Uni_data2, NCPK_GO_PW, by = "AGI")
dim(Uni_NCPK_GO_PW)

library(plyr)
GO_PW_final <- join(AGI_GO_PW, Uni_NCPK_GO_PW)
dim(GO_PW_final)  
head(GO_PW_final)[,1:5]

write.table(as.data.frame(GO_PW_final), "GO_PW_final.txt", sep = "\t", row.names = FALSE)

```


#qPCR validation
```{r}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/qPCR & western blot/qPCR result")
co <- read.delim("Correlation_R.txt")
dim(co)
co_shoot <- co[1:12,]
co_root <- co[13:32,]

cor.test(co$qPCR, co$RNA.seq, method = "pearson")
cor.test(co$qPCR, co$RNA.seq, method = "spearman")

cor.test(co_shoot$qPCR, co_shoot$RNA.seq, method = "pearson")
cor.test(co_root$qPCR, co_root$RNA.seq, method = "pearson")

cor.test(co_shoot$qPCR, co_shoot$RNA.seq, method = "spearman")
cor.test(co_root$qPCR, co_root$RNA.seq, method = "spearman")

library(ggpubr)
library(Rmisc)
library(ggplot2)
library(plyr)

co2 <- read.delim("Correlation_R2.txt")
co_all <- ggscatter(co, x = "RNA.seq", y = "qPCR", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          main = "Correlation between RNA-seq and qPCR validation (All)",
          add.params = list(color = "blue", fill = "grey50"),
          xlab = "Log2FC (RNA-seq)", ylab = "Log2FC (qPCR)")
co_all

co_shoot <- co[c(1:12),]
co_shoot_plot <- ggscatter(co_shoot, x = "RNA.seq", y = "qPCR", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          main = "Shoot without AGO9",
          add.params = list(color = "blue", fill = "grey50"),
          xlab = "Log2FC (RNA-seq)", ylab = "Log2FC (qPCR)")
co_shoot_plot

co_root <- co[c(13:32),]
co_root_plot <- ggscatter(co_root, x = "RNA.seq", y = "qPCR", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          main = "Root without AGO9",
          add.params = list(color = "blue", fill = "grey50"),
          xlab = "Log2FC (RNA-seq)", ylab = "Log2FC (qPCR)")
co_root_plot




co_wo_AGO_shoot <- co[c(5:12),]
co_shoot_AGOX <- ggscatter(co_wo_AGO_shoot, x = "RNA.seq", y = "qPCR", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          main = "Shoot without AGO9",
          add.params = list(color = "blue", fill = "grey50"),
          xlab = "Log2FC (RNA-seq)", ylab = "Log2FC (qPCR)")

co_wo_AGO_root <- co[c(13:16, 21:32),]
co_root_AGOX <- ggscatter(co_wo_AGO_root, x = "RNA.seq", y = "qPCR", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          main = "Root without AGO9",
          add.params = list(color = "blue", fill = "grey50"),
          xlab = "Log2FC (RNA-seq)", ylab = "Log2FC (qPCR)")


cor_plot <- ggarrange(co_all, 
                      ggarrange(co_root, co_shoot, ncol =2),
                      nrow = 2)
cor_plot


#Compare FC

Root_FC <- read.delim("Root_FC.txt")
Shoot_FC <- read.delim("Shoot_FC.txt")
head(Root_FC)

myDataN <- mutate(myData, mean_new = mean*sign, se_new=se+30)
dodge <- position_dodge(width = 0.9)


Root_FC_stat <- summarySE(Root_FC, measurevar = "FC", groupvars = c("Method", "Gene"))
Root_FC_stat$Gene <- factor(Root_FC_stat$Gene, levels = c('IRT1', 'AGO9', 'ZYP1a', 'HMA2', 'ZIP6'))
Root_FC_stat$Tissue <- "Root"


FC_R <- ggplot(Root_FC_stat, aes(x=Gene, y=FC, fill=Method)) +
  geom_bar(position="dodge", stat="identity", color = "grey20") +
   geom_errorbar(limits,
                  size=.3, color = "grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
  scale_fill_manual(values=c("grey95", "grey30"),
                      breaks=c("qPCR", "RNA_seq")) +
  theme_bw()
FC_R
  

Shoot_FC_stat <- summarySE(Shoot_FC, measurevar = "FC", groupvars = c("Method", "Gene"))
Shoot_FC_stat$Gene <- factor(Shoot_FC_stat$Gene, levels = c('AGO9', 'ZYP1a', 'HMA2'))
Shoot_FC_stat$Tissue <- "Shoot"

FC_S <- ggplot(Shoot_FC_stat, aes(x=Gene, y=FC, fill=Method)) +
  geom_bar(position="dodge", stat="identity", color = "grey20") +
   geom_errorbar(aes(ymin=FC, ymax=FC+sd),
                  size=.3, color = "grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
  scale_fill_manual(values=c("grey95", "grey30"),
                      breaks=c("qPCR", "RNA_seq")) +
  theme_bw() 
FC_S


FC_all <- ggarrange(FC_R, FC_S, ncol = 2, nrow = 1, 
                    common.legend = TRUE, legend = "bottom",
                    align = "v")
FC_all

#Merged plot
All_FC_stat <- rbind(Root_FC_stat, Shoot_FC_stat)
All_FC_stat

limits <- aes(ymax= All_FC_stat$FC + (All_FC_stat$FC > 0)*All_FC_stat$sd,
              ymin= All_FC_stat$FC - (All_FC_stat$FC < 0)*All_FC_stat$sd)

FC_merged <- ggplot(All_FC_stat, aes(x=Gene, y=FC, fill=Method)) +
  geom_bar(position="dodge", stat="identity", color = "grey20") +
   geom_errorbar(limits,
                  size=.3, color = "grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
  scale_fill_manual(values=c("grey95", "grey30"),
                      breaks=c("qPCR", "RNA_seq")) +
  facet_wrap(~Tissue, drop =TRUE) +
  theme_classic() 

FC_merged

########New plots###################

Cor_all <- read.delim("Co_all.txt")
head(Cor_all)


Cor_all_P <- ggscatter(Cor_all, x = "qPCR_TL", y = "NCPK", 
                       shape = "Population", color ="Gene", bg = "Tissue",
                       size = "Tissue",
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          main = "Correlation between RNA-seq and qPCR validation",
          add.params = list(color = "black", fill = "grey50"),
          xlab = "qPCR-based relative transcript level", ylab = "RNA-seq based transcript levels (NCPK)",
          palette = c("springgreen4", "magenta2", "navyblue", "orangered1")
          ) +
          theme_bw() +
          theme(panel.grid.major = element_blank()) +
          theme(panel.grid.minor = element_blank())
Cor_all_P




#For RTL
Cor_shoot_stat <- summarySE(Cor_all[1:12,], measurevar = "qPCR_TL", groupvars = c("Population", "Gene", "Tissue"))
Cor_shoot_stat$Gene <- factor(Cor_shoot_stat$Gene, levels = c('AGO9', 'ZYP1a', 'HMA2'))
Cor_root_stat <- summarySE(Cor_all[13:28,], measurevar = "qPCR_TL", groupvars = c("Population", "Gene", "Tissue"))
Cor_root_stat$Gene <- factor(Cor_root_stat$Gene, levels = c('AGO9', 'ZYP1a', 'HMA2', 'IRT1'))
Cor_all_stat <- rbind(Cor_shoot_stat, Cor_root_stat)
head(Cor_all_stat)

RTL_all <- ggplot(Cor_all_stat, aes(x=Gene, y=qPCR_TL, fill=Population)) +
  geom_bar(position="dodge", stat="identity", color = "grey20") +
   geom_errorbar(aes(ymin=qPCR_TL, ymax=qPCR_TL+sd),
                  size=.3, color = "grey20",
                  width=.2,                    
                  position=position_dodge(.9)) +
  ylab("Relative transcript level") +
  scale_fill_manual(values=c("red", "blue"),
                      breaks=c("Noss", "Pais")) +
  facet_wrap(~Tissue, drop =TRUE) +
  theme_classic() 
RTL_all


#For each gene
#Root
#AGO9
AGO9_Root <- ggplot(Cor_all_stat[c(7,11),], aes(x=Population, y=qPCR_TL, fill=Population)) +
  geom_bar(position="dodge", stat="identity", color = "black") +
   geom_errorbar(aes(ymin=qPCR_TL, ymax=qPCR_TL+sd),
                  size=c(0.5,0.8), color = "black",
                  width=.2,                    
                  position=position_dodge(.9)) +
   ylab("") +
  xlab("") + ggtitle("AGO9") +
  scale_fill_manual(values=c("#EE3B3B", "#0072B2"),
                      breaks=c("Noss", "Pais")) +
  theme_bw() + theme(panel.grid.major = element_blank()) +
  theme(legend.position="none") +
  theme(plot.title = element_text(color="black", size=18, face = "bold"),
    axis.title = element_text( size=15, color = "black"),
  axis.text.y = element_text(size=13, color = "black"),
  axis.text.x = element_text(size=15, color = "black"))

ZYP1a_Root <- ggplot(Cor_all_stat[c(10,14),], aes(x=Population, y=qPCR_TL, fill=Population)) +
  geom_bar(position="dodge", stat="identity", color = "black") +
   geom_errorbar(aes(ymin=qPCR_TL, ymax=qPCR_TL+sd),
                  size=c(0.5,0.8), color = "black",
                  width=.2,                    
                  position=position_dodge(.9)) +
   ylab("") +
  xlab("") + ggtitle("ZYP1a") +
  scale_fill_manual(values=c("#EE3B3B", "#0072B2"),
                      breaks=c("Noss", "Pais")) +
  theme_bw() + theme(panel.grid.major = element_blank()) +
  theme(legend.position="none") +
  theme(plot.title = element_text(color="black", size=18, face = "bold"),
    axis.title = element_text( size=15, color = "black"),
  axis.text.y = element_text(size=13, color = "black"),
  axis.text.x = element_text(size=15, color = "black"))

HMA2_Root <- ggplot(Cor_all_stat[c(8,12),], aes(x=Population, y=qPCR_TL, fill=Population)) +
  geom_bar(position="dodge", stat="identity", color = "black") +
   geom_errorbar(aes(ymin=qPCR_TL, ymax=qPCR_TL+sd),
                  size=.5, color = "black",
                  width=.2,                    
                  position=position_dodge(.9)) +
   ylab("") +
  xlab("") + ggtitle("HMA2") +
  scale_fill_manual(values=c("#EE3B3B", "#0072B2"),
                      breaks=c("Noss", "Pais")) +
  theme_bw() + theme(panel.grid.major = element_blank()) +
  theme(legend.position="none") +
  theme(plot.title = element_text(color="black", size=18, face = "bold"),
    axis.title = element_text( size=15, color = "black"),
  axis.text.y = element_text(size=13, color = "black"),
  axis.text.x = element_text(size=15, color = "black"))


IRT1_Root <- ggplot(Cor_all_stat[c(9,13),], aes(x=Population, y=qPCR_TL, fill=Population)) +
  geom_bar(position="dodge", stat="identity", color = "black") +
   geom_errorbar(aes(ymin=qPCR_TL, ymax=qPCR_TL+sd),
                  size=.5, color = "black",
                  width=.2,                    
                  position=position_dodge(.9)) +
   ylab("") +
  xlab("") + ggtitle("IRT1") +
  scale_fill_manual(values=c("#EE3B3B", "#0072B2"),
                      breaks=c("Noss", "Pais")) +
  theme_bw() + theme(panel.grid.major = element_blank()) +
  theme(legend.position="none") +
  theme(plot.title = element_text(color="black", size=18, face = "bold"),
    axis.title = element_text( size=15, color = "black"),
  axis.text.y = element_text(size=13, color = "black"),
  axis.text.x = element_text(size=15, color = "black"))

#Shoot
AGO9_Shoot <- ggplot(Cor_all_stat[c(1,4),], aes(x=Population, y=qPCR_TL, fill=Population)) +
geom_bar(position="dodge", stat="identity", color = "black") +
   geom_errorbar(aes(ymin=qPCR_TL, ymax=qPCR_TL+sd),
                  size=.5, color = "black",
                  width=.2,                    
                  position=position_dodge(.9)) +
   ylab("") +
  xlab("") + ggtitle("AGO9") +
  scale_fill_manual(values=c("#EE3B3B", "#0072B2"),
                      breaks=c("Noss", "Pais")) +
  theme_bw() + theme(panel.grid.major = element_blank()) +
  theme(legend.position="none") +
  theme(plot.title = element_text(color="black", size=18, face = "bold"),
    axis.title = element_text( size=15, color = "black"),
  axis.text.y = element_text(size=13, color = "black"),
  axis.text.x = element_text(size=15, color = "black"))

ZYP1a_Shoot <- ggplot(Cor_all_stat[c(3,6),], aes(x=Population, y=qPCR_TL, fill=Population)) +
geom_bar(position="dodge", stat="identity", color = "black") +
   geom_errorbar(aes(ymin=qPCR_TL, ymax=qPCR_TL+sd),
                  size=.5, color = "black",
                  width=.2,                    
                  position=position_dodge(.9)) +
   ylab("") +
  xlab("") + ggtitle("ZYP1a") +
  scale_fill_manual(values=c("#EE3B3B", "#0072B2"),
                      breaks=c("Noss", "Pais")) +
  theme_bw() + theme(panel.grid.major = element_blank()) +
  theme(legend.position="none") +
  theme(plot.title = element_text(color="black", size=18, face = "bold"),
    axis.title = element_text( size=15, color = "black"),
  axis.text.y = element_text(size=13, color = "black"),
  axis.text.x = element_text(size=15, color = "black"))

HMA2_Shoot <- ggplot(Cor_all_stat[c(2,5),], aes(x=Population, y=qPCR_TL, fill=Population)) +
geom_bar(position="dodge", stat="identity", color = "black") +
   geom_errorbar(aes(ymin=qPCR_TL, ymax=qPCR_TL+sd),
                  size=.5, color = "black",
                  width=.2,                    
                  position=position_dodge(.9)) +
   ylab("") +
  xlab("") + ggtitle("HMA2") +
  scale_fill_manual(values=c("#EE3B3B", "#0072B2"),
                      breaks=c("Noss", "Pais")) +
  theme_bw() + theme(panel.grid.major = element_blank()) +
  theme(legend.position="none") +
  theme(plot.title = element_text(color="black", size=18, face = "bold"),
    axis.title = element_text( size=15, color = "black"),
  axis.text.y = element_text(size=13, color = "black"),
  axis.text.x = element_text(size=15, color = "black"))



qPCR_RTL <- ggarrange(AGO9_Root, ZYP1a_Root, HMA2_Root, IRT1_Root, 
          AGO9_Shoot, ZYP1a_Shoot, HMA2_Shoot, 
          nrow = 2, ncol = 4,
          align = "hv")

annotate_figure(qPCR_RTL,
                left = text_grob("Relative transcript level (qPCR)", color = "black", rot = 90, size = 17))




```

#Hypergeometric p-value for venn diagram
```{r}

install.packages("gmp")
library(gmp)

enrich_pvalue <- function(N, A, B, k)
{
    m <- A + k
    n <- B + k
    i <- k:min(m,n)

    as.numeric( sum(chooseZ(m,i)*chooseZ(N-m,n-i))/chooseZ(N,n) )
}

#Cd responsive genes
#Root
enrich_pvalue(N=26154, A=166, B=153, k=44)
#Shoot
enrich_pvalue(N=26345, A=344, B=159, k=84)

#Comparison between populations
#Example
#NvsP + NvsW (-Cd) root
enrich_pvalue(N=26154, A=629, B=1724, k=1125)
#NvsP + NvsW (-Cd) shoot
enrich_pvalue(N=26154, A=872, B=1100, k=1130)


```

## New list for meiosis genes
```{r}

setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/New_meiosis")
New_meiosis_AGI <- read.delim("Meiosis_gene_AGI.txt")
dim(New_meiosis_AGI)

Predata3<- read.delim("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/Predataset3.txt")
Predata3_Glength <- Predata3[,c(1:3,28:52)]

Predata_all <- read.delim("Predata_all.txt")
View(head(Predata_all))

New_meiosis_Ano0 <- join(New_meiosis_AGI, Predata_all)
dim(New_meiosis_Ano0)

New_meiosis_Ano <- join(New_meiosis_Ano0, Predata3_Glength)
dim(New_meiosis_Ano)

write.table(as.data.frame(New_meiosis_Ano), "New_meiosis.txt", sep = "\t", row.names = FALSE)

```

# Additional list for CDEG under Cd (Dataset S3)
```{r}
setwd("C:/Users/Justin/OneDrive - rub.de/PhD/Gwonjin/Experiment/Result/RNA seq/DESeq2/Result/New result3/Table/CDEG_Noss_0.5_Cd")
CDEG_Cd_Root <- read.delim("CDEG_Cd_Root.txt")
dim(CDEG_Cd_Root)
CDEG_Cd_Shoot <- read.delim("CDEG_Cd_Shoot.txt")
dim(CDEG_Cd_Shoot)
resNvsP_Cd_Root_ano <- read.delim("Ano_resNvsP_Cd_Root_NT.txt")
resNvsW_Cd_Root_ano <- read.delim("Ano_resNvsW_Cd_Root_NT.txt")
resNvsP_Cd_Shoot_ano <- read.delim("Ano_resNvsP_Cd_Shoot_NT.txt")
resNvsW_Cd_Shoot_ano <- read.delim("Ano_resNvsW_Cd_Shoot_NT.txt")

colnames(Predata_all)[1] <- "genes"
head(Predata_all)[,1:3]

CDEG_Cd_Root_Ano0 <- join(CDEG_Cd_Root, resNvsP_Cd_Root_ano)
CDEG_Cd_Root_Ano1 <- join(CDEG_Cd_Root_Ano0, resNvsW_Cd_Root_ano[,c(1,12:17)], by = "genes")
dim(CDEG_Cd_Root_Ano1)
CDEG_Cd_Root_Ano2 <- join(CDEG_Cd_Root_Ano1, Predata_all[,c(1,11)], by = "genes")
CDEG_Cd_Root_Ano <- join(CDEG_Cd_Root_Ano2, Predata3_Glength, by = "genes")
dim(CDEG_Cd_Root_Ano)
write.table(as.data.frame(CDEG_Cd_Root_Ano), "CDEG_Cd_Root_Ano.txt", sep = "\t", row.names = FALSE)


CDEG_Cd_Shoot_Ano0 <- join(CDEG_Cd_Shoot, resNvsP_Cd_Shoot_ano)
CDEG_Cd_Shoot_Ano1 <- join(CDEG_Cd_Shoot_Ano0, resNvsW_Cd_Shoot_ano[,c(1,12:17)], by = "genes")
dim(CDEG_Cd_Shoot_Ano1)
CDEG_Cd_Shoot_Ano2 <- join(CDEG_Cd_Shoot_Ano1, Predata_all[,c(1,11)], by = "genes")
CDEG_Cd_Shoot_Ano <- join(CDEG_Cd_Shoot_Ano2, Predata3_Glength, by = "genes")
dim(CDEG_Cd_Shoot_Ano)
write.table(as.data.frame(CDEG_Cd_Shoot_Ano), "CDEG_Cd_Shoot_Ano.txt", sep = "\t", row.names = FALSE)

```





